---
title: "Penetrance_github"
output: html_document
---

# set up and packages

```{r library or install.packages}
#setwd()

#my_packages <- c("data.table", "ggplot2", "dplyr", "plyr") #e.g.
#not_installed <- my_packages[!(my_packages %in% installed.packages()[ , "Package"])]
#if(length(not_installed)) install.packages(not_installed)   

library(data.table)
library(ggplot2)
library(dplyr)
library(plyr)
library(tidyr)
library(ggpubr)
set.seed(28061971)
```

# penetrance function

```{r}
#### as a function
penetrance <- function(x_D, n_D, x_AgD, n_AgD, x_A, n_A){
set.seed(28061971)
digits <- 6   
alpha <- 0.05  
p_D <- x_D / n_D
p_AgD <- x_AgD / n_AgD
p_A <- x_A / n_A
d <- 0.5   
p_D_wod <- (x_D + d) / (n_D + d)
p_AgD_wod <- (x_AgD + d) / (n_AgD + d)
p_A_wod <- (x_A + d) / (n_A + d)
log_AR <- log(p_D_wod * p_AgD_wod / p_A_wod) + 1/2 * ((1 / p_A_wod) * (1 - p_A_wod) / (n_A + d) - 
                                                       (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) - 
                                                       (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d))
Var_log_AR <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
              (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
              (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_LCI <- log_AR - qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
log_UCI <- log_AR + qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
penetrance <- pmin(1,pmax(0,exp(log_AR)))
log_lCI <- pmin(1,pmax(0,exp(log_LCI)))
log_uCI <- pmin(1,pmax(0,exp(log_UCI)))
my_list <- list("penetrance" = penetrance, "lci" = log_lCI, "uci" = log_uCI)
return(my_list)
}
```

# select definitive- and strong-evidence HCM and DCM associated genes

```{r}
vep <- data.frame(fread("vepresultsR", header=T)) #Load VEP105 results without comments at start of file.

#MANE == Canonical in VEP105
vep2 <- vep %>%
filter(CANONICAL == "YES")

#which genes included?
# HCM
#
# *ACTC1 - nTV      # ACTC1	altered_gene_product_structure
# *MYBPC3 - PAV     # MYBPC3	decreased_gene_product_level,altered_gene_product_structure
# *MYH7 - nTV       # MYH7	altered_gene_product_structure
# *MYL2 - nTV       # MYL2	altered_gene_product_structure
# *MYL3 - nTV       # MYL3	altered_gene_product_structure
# *TNNI3 - nTV      # TNNI3	altered_gene_product_structure
# *TNNT2 - nTV      # TNNT2	altered_gene_product_structure
# *TPM1 - nTV       # TPM1	altered_gene_product_structure

# DCM
# 
# BAG3 - PAV                  #BAG3	decreased_gene_product_level,altered_gene_product_structure
# DES - nTV                   #DES	altered_gene_product_structure
# FLNC - TV #not included     #FLNC	decreased_gene_product_level
# LMNA - PAV                  #LMNA	decreased_gene_product_level,altered_gene_product_structure
# MYH7 - nTV                  #MYH7	altered_gene_product_structure
# PLN - PAV                   #PLN	decreased_gene_product_level,altered_gene_product_structure
# RBM20 - PAV                 #RBM20	decreased_gene_product_level,altered_gene_product_structure
# SCN5A - PAV                 #SCN5A	decreased_gene_product_level,altered_gene_product_structure
# TNNC1 - nTV                 #TNNC1	altered_gene_product_structure
# TNNT2 - nTV                 #TNNT2	altered_gene_product_structure
# TTN - TV                    #TTN	decreased_gene_product_level
# DSP - strong - PAV          #DSP	decreased_gene_product_level,altered_gene_product_structure

vep3 <- vep2[
vep2$SYMBOL == "BAG3" |
vep2$SYMBOL == "DSP"  |
vep2$SYMBOL == "TTN" |
vep2$SYMBOL == "MYH7"  |
vep2$SYMBOL == "TNNC1"  |
vep2$SYMBOL == "TNNT2"  |
vep2$SYMBOL == "TPM1"  |
vep2$SYMBOL == "LMNA"  |
vep2$SYMBOL == "ACTC1"  |
vep2$SYMBOL == "PLN"  |
vep2$SYMBOL == "RBM20"|
vep2$SYMBOL == "MYBPC3"|
vep2$SYMBOL == "TNNI3" |
vep2$SYMBOL == "MYL2" |
vep2$SYMBOL == "MYL3" |
vep2$SYMBOL == "DES" |
vep2$SYMBOL == "SCN5A" ,
]
```

# manual assessment of pathogenic non-coding variants (splice variants that are mis predicted)
# filter for PAVs
# add .bim file for variant info and case counts for each variant

```{r}
test <- vep2[which(vep2$ClinVar_CLNSIG == "Likely_pathogenic" | vep2$ClinVar_CLNSIG == "Pathogenic/Likely_pathogenic" | vep2$ClinVar_CLNSIG == "Pathogenic"),]
table(test$Consequence)
test$ClinVar[test$Consequence == "intron_variant"]
#"138326" LP splice confirmed
#"188544" P/LP splice confirmed
#"219660" P/LP splice confirmed
test$ClinVar[test$Consequence == "splice_polypyrimidine_tract_variant,intron_variant"]
#"42807"  P/LP splice confirmed
#"42805"  LP splice confirmed
#"987382" P splice confirmed
test$ClinVar[test$Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant"]
#"66856" P/LP splice confirmed
#splice_donor_region_variant,intron_variant
#"42556" P splice confirmed
#"66419" P splice confirmed

vep2$Consequence[(vep2$ClinVar == "138326" |
                     vep2$ClinVar == "188544" |
                     vep2$ClinVar == "219660" |
                     vep2$ClinVar == "42807" |
                     vep2$ClinVar == "42805" |
                     vep2$ClinVar == "987382" |
                     vep2$ClinVar == "66856" |
                     vep2$ClinVar == "42556" |
                     vep2$ClinVar == "66419")] <- "splice_confirmed"

vepx32 <- vep2 %>% filter(Consequence == "frameshift_variant" | 
                           Consequence == "frameshift_variant,splice_region_variant" |
                           Consequence == "frameshift_variant,start_lost" |
                           Consequence == "frameshift_variant,stop_lost" |
                           Consequence == "inframe_deletion" |
                           Consequence == "inframe_deletion,splice_region_variant" |
                           Consequence == "inframe_insertion" |
                           Consequence == "missense_variant" |
                           Consequence == "missense_variant,splice_region_variant" |
                           Consequence == "protein_altering_variant" |
                           Consequence == "splice_acceptor_variant" |
                           Consequence == "splice_confirmed" |
                           Consequence == "splice_donor_5th_base_variant,intron_variant" |
                           Consequence == "splice_donor_region_variant,intron_variant" |
                           Consequence == "splice_donor_variant" |
                           Consequence == "splice_donor_variant,coding_sequence_variant" |
                           Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                           Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                           Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                           Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                           Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                           Consequence == "splice_region_variant,intron_variant" |
                           Consequence == "splice_region_variant,synonymous_variant" |
                           Consequence == "start_lost" |
                           Consequence == "stop_gained" |
                           Consequence == "stop_gained,frameshift_variant" |
                           Consequence == "stop_gained,splice_region_variant" |
                           Consequence == "stop_lost")

# Add bim variant info to dataset
# Add HCM and DCM counts to dataset (on DECIPHER)
```

# Add UKB 200k exome data counts
### Identify the 200,000 participants in UKB with WES available and withdrawn participants removed
### Identify subgroups by ancestry, sex, age at recruitment
### UKB included in PCA flag to remove related individuals and other mismatches
### Use plink --keep, --freq, --freq counts

```{r}
#3. remove relateds
#4. create --frqs and frq counts

#followed https://biobank.ndph.ox.ac.uk/ukb/ukb/docs/ukbgene_instruct.html
#22020 column:
#https://www.rdocumentation.org/packages/ukbtools/versions/0.11.3/topics/ukb_gen_samples_to_remove
#(...) you could use the list of samples which we used to calculate the PCs, which is a (maximal) subset of unrelated participants after applying some QC filtering. Please read supplementary Section S3.3.2 for details. You can find the list of samples using the "used.in.pca.calculation" column in the sample-QC file (ukb_sqc_v2.txt) (...). Note that this set contains diverse ancestries. If you take the intersection with the white British ancestry subset you get ~337,500 unrelated samples.
#Missing rate on autosomes > 0.02.
#Not in a set of unrelated individuals (see Sections S 3.7.1 and S 3.7.4).
#In the list of outliers based on heterozygosity and missing rates (see Section S3.5).
#Mismatch between inferred sex and self-reported sex.
#https://www.nature.com/articles/s41586-018-0579-z#Sec28
#Bycroft et al. 2018

#assess reported ancestry
#agrees with population genetics in ensembl (e.g. https://www.ensembl.org/Homo_sapiens/Variation/Population?db=core;r=1:46404589-46405589;v=rs324420;vdb=variation;vf=158611)
#AFR <- African + Caribbean
#SAS <- The region consists of the countries of Afghanistan, Bangladesh, Bhutan, India, Maldives, Nepal, Pakistan, and Sri Lanka.[7] wikipedia
   #Indian, Pakistani, Bangladeshi
#EAS <- The modern states of East Asia include China, Japan, Mongolia, North Korea, South Korea, and Taiwan.
  #Chinese
#EUR is from previous estimate of european individuals https://doi.org/10.1038/s41586-020-2635-8
#White British is UKB data field 22006 

#join UKB datasets together
# add .frq = MAF + NCHROBS(AN)
# add .frq.counts = C1 = AC

#join UKB counts with HCM/DCM variants = caseukb.txt
```

# filter splice synonymous

```{r filter splice region and TTN PSI 90}
caseukb <- read.table("caseukb.txt", header=T, sep="\t")

#check 1 PAV
caseukb[which(caseukb$Consequence == "protein_altering_variant"),] #Clinvar = 174799
#This sequence change deletes 1 nucleotide and inserts 4 nucleotides in exon 2 of the MYBPC3 mRNA (c.239delinsGAGG). This leads to the deletion of one amino acid and insertion of 2 amino acid residues in the MYBPC3 protein (p.Ala80delinsGlyGly) but otherwise preserves the integrity of the reading frame.
caseukb$Consequence[caseukb$Consequence == "protein_altering_variant"] <- "inframe_insertion"
 
table(caseukb$Consequence)
df <- caseukb %>%
  filter(Consequence == "splice_donor_region_variant,intron_variant" |
           Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
           Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,synonymous_variant") %>% 
  filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
dim(df)
#366 for removal
  
caseukb$found <- caseukb$ID %in% df$ID
table(caseukb$found)
#TRUE = 366 woohoo

caseukb3 <- caseukb[which(caseukb$found == "FALSE"),]
dim(caseukb3)
#4325
caseukb3$found <- NULL
```

# filter out TTN missense and assess LOFTEE flags

```{r TTN missense}
#There are two TTN P/LP missense variants with strong evidence=
#https://www.biorxiv.org/content/10.1101/2020.09.05.282913v1.full.pdf = Chr2(GRCh38):g.178741559A>T; p.Cys3575Ser - not in dataset.
caseukb3[which(caseukb3$SYMBOL == "TTN" & caseukb3$POS == 178741559),]
#Hinson et al PMID: 26315439 An iPS-derived cardiomyocyte model of the missense variant NM_001267550: c.2926T>C, p.(Trp976Arg) from Gerull et al PMID: 11788824 showed that this variant exhibited less than half the contractile force generated by WT, and was comparable to the effect of an A-band truncating variant also tested. Note: This variant has also been detected by a diagnostic laboratory in a DCM proband (https://www.ncbi.nlm.nih.gov/clinvar/variation/12651/), and scores PP1_strong, PM2, PS3_mod, PP3 – pathogenic by ACMG guidelines - not in dataset. #  2: 178782980 
caseukb3[which(caseukb3$ClinVar == "12651"),]
caseukb3[which(caseukb3$SYMBOL == "TTN" & caseukb3$POS == 178782980),]
#Both not in dataset

hmm <- caseukb3[caseukb3$SYMBOL == "TTN",]
hmm <- caseukb3[(caseukb3$SYMBOL == "TTN" & (caseukb3$Consequence == "missense_variant" | caseukb3$Consequence == "inframe_deletion" | caseukb3$Consequence == "inframe_insertion")),]
dim(hmm)
#1896
#none are P/LP

caseukb4 <- caseukb3[!(caseukb3$SYMBOL == "TTN" & (caseukb3$Consequence == "missense_variant" | caseukb3$Consequence == "inframe_deletion" | caseukb3$Consequence == "inframe_insertion")),]
dim(caseukb4)
#2429

#19 "missense_variant,splice_region_variant" in TTN
hmm <- caseukb4 %>%
  filter(SYMBOL == "TTN") %>%
  filter(Consequence == "missense_variant,splice_region_variant") %>%
    filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
dim(hmm)
#38 - all to go
caseukb4$found <- caseukb4$ID %in% hmm$ID
table(caseukb4$found)
doneant2 <- caseukb4[which(caseukb4$found == "FALSE"),]
dim(doneant2)
#2391
doneant2$found <- NULL

#Test LOFTEE flags for each gene
table(doneant2$LoF_filter)
#3UTR_SPLICE 5UTR_SPLICE 
table(doneant2$LoF_flags)
#NAGNAG_SITE PHYLOCSF_WEAK 

doneant2$Consequence <- as.character(doneant2$Consequence)

#flag LC LoFs and ID PAV that are mislabelled.
doneant2$Consequence[doneant2$LoF_flags == "NAGNAG_SITE"] <- "inframe_insertion"
doneant2$Consequence[doneant2$LoF_filter == "3UTR_SPLICE"] <- "splice_donor_variant_LC"
doneant2$Consequence[doneant2$LoF_filter == "5UTR_SPLICE"] <- "splice_donor_variant_LC"
#NAGNAG site = splice acceptor site rescued by inframe acceptor site
#3UTR SPLICE = splice donor with pLof low confidence
#5UTR SPLICE = splice donor with pLof low confidence

#create a label for each variant = SYMBOL:HGVSc
df3 <- doneant2 %>% separate(HGVSc, c("HGVSc.a", "HGVSc.b"), sep=":")
df3$label <- paste0(df3$SYMBOL, ":", df3$HGVSc.b)
```

# split datasets by HCM and DCM-associated genes

```{r}
HCM <- df3 %>% filter(SYMBOL == "ACTC1" | SYMBOL == "MYBPC3" | SYMBOL == "MYH7" | SYMBOL == "MYL2" | SYMBOL == "MYL3" | SYMBOL == "TNNI3" | SYMBOL == "TNNT2" | SYMBOL == "TPM1")
HCM <- HCM[which(HCM$AC_ALL_HCM>0),]
dim(HCM)
#1384 

DCM <- df3 %>% filter(SYMBOL == "BAG3" | SYMBOL == "DES" | SYMBOL == "FLNC" | SYMBOL == "LMNA" | SYMBOL == "MYH7" | SYMBOL == "PLN" | SYMBOL == "RBM20" | SYMBOL == "SCN5A" | SYMBOL == "TNNC1" | SYMBOL == "TNNT2" | SYMBOL == "TTN" | SYMBOL == "DSP")
DCM <- DCM[which(DCM$AC_ALL_DCM>0),]
dim(DCM)
#764 
```

# HCM specific filtering (assess variants, NMDi, filter for disease-associated variant consequences)

```{r specific genes only - HCM}
#check LC variants
HCM$SYMBOL[HCM$Consequence == "splice_donor_variant_LC"] #MYBPC3 - can remove - in intron.
HCM1 <- HCM[!(HCM$Consequence == "splice_donor_variant_LC"),]
dim(HCM1)
# 1383

table(HCM1$Consequence)
#go through variant consequences that require manual assessment:
#start lost - below
#stop lost = inframe insertion
#inframe_deletion,splice_region_variant - below.

read <- HCM1[which(HCM1$Consequence == "inframe_deletion,splice_region_variant"),]
read$SpliceAI_cutoff # no info - no splice prediction, no databases - assume inframe deletion. Its in MYBPC3 so low risk.

read <- HCM1[which(HCM1$Consequence == "start_lost"),]
data.frame(read$ClinVar, read$SYMBOL, read$Consequence, read$ID, read$REF, read$ALT, read$chr,read$POS)
#both MYBPC3 - no obvious rescues and they are missense vars. leave as is.

read <- HCM1[which(HCM1$Consequence == "stop_lost"),]
data.frame(read$ClinVar, read$SYMBOL, read$Consequence, read$ID, read$REF, read$ALT, read$chr,read$POS)
#MYBPC3 - no obvious rescue - assume indel. 
#MYL2 - no obvious rescue - assume indel. 

table(HCM1$NMD)
#NMDi:
#  - NMD_escaping_variant 
# 1292                   91 

table(HCM1$Consequence)

PAV_List=c("frameshift_variant",
           "stop_gained",
           "splice_donor_variant",
           "splice_acceptor_variant",
           "splice_region_variant",
           "splice_polypyrimidine_tract_variant",
           "splice_donor_5th_base_variant",
           "splice_donor_region_variant",
           "splice_confirmed")

hmmx <- HCM1[grepl(paste(PAV_List, collapse="|"), HCM1$Consequence),]
dim(hmmx) 
#388

head(hmmx$POS,10)
hmmx$hgvk <- hmmx$HGVSp
hmmx$hgvk <- gsub("\\?", 0, hmmx$hgvk)
hmmx <- hmmx %>% separate(hgvk, c("hgvka", "hgvkb", "fs"), sep="\\*")
hmmx$fs <- as.integer(as.character(hmmx$fs)) 
hmmx$fs <- (hmmx$fs*3)

for(i in (1:nrow(hmmx))){
if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") &
    (hmmx$SYMBOL[i] == "MYH7" | hmmx$SYMBOL[i] == "MYBPC3" | hmmx$SYMBOL[i] == "ACTC1" | 
       hmmx$SYMBOL[i] == "TNNT2" | hmmx$SYMBOL[i] == "TNNI3" | hmmx$SYMBOL[i] == "MYL2" | hmmx$SYMBOL[i] == "MYL3")) {
    hmmx$POS[i] <- hmmx$POS[i] - hmmx$fs[i] 
        }
    else if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") & 
            (hmmx$SYMBOL[i] == "TPM1")){
            hmmx$POS[i] <- hmmx$POS[i] + hmmx$fs[i] 
    }
   }

hmm <- hmmx[which(hmmx$SYMBOL == "MYH7"),]
#MYH7 <	23,413,759 + 55
hmm2 <- hmm[which(hmm$POS < 23413814),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 14:23882988:TC:T frameshift_variant 23413749 ENSP00000347507.3:p.K1923*fs*10        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "MYBPC3"),]
#MYBPC3 <47,331,881 - second last exon is short
hmm2 <- hmm[which(hmm$POS < 47331881),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

hmm <- hmmx[which(hmmx$SYMBOL == "TNNT2"),]
#TNNT2 < 	201,359,663 - second last exon is short
hmm2 <- hmm[which(hmm$POS < 201359663),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 1:201328345:C:T                                  stop_gained 201359217 ENSP00000499593.1:p.W297* NMD_escaping_variant                 FAIL
#2 1:201328746:C:T splice_donor_5th_base_variant,intron_variant 201359618                         -                    -                 FAIL
#3 1:201328750:C:A                         splice_donor_variant 201359622                         -                    -                 PASS
#4 1:201328750:C:G                         splice_donor_variant 201359622                         -                    -                 PASS

hmm <- hmmx[which(hmmx$SYMBOL == "TNNI3"),]
#TNNI3 < 	55,154,030 + 55
hmm2 <- hmm[which(hmm$POS < 55154085),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1  19:55663262:C:T                            stop_gained 55151894      ENSP00000341838.5:p.W191* NMD_escaping_variant                 FAIL
#2  19:55665398:C:G missense_variant,splice_region_variant 55154030      ENSP00000341838.5:p.K183N                    -                 FAIL
#3 19:55665403:CG:C                     frameshift_variant 55153981 ENSP00000341838.5:p.T181*fs*18                    -                 FAIL
#4 19:55665408:TC:T                     frameshift_variant 55153983 ENSP00000341838.5:p.D180*fs*19                    -                 FAIL
#5 19:55665440:CA:C                     frameshift_variant 55154048  ENSP00000341838.5:p.L169*fs*8                    -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "TPM1"),]
#TPM1 > 63064142-55
hmm2 <- hmm[which(hmm$POS > 63064087),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 15:63356340:A:G missense_variant,splice_region_variant 63064141 ENSP00000385107.4:p.I284V        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "MYL2"),]
#MYL2 < 110,913,144  - second last exon is short
hmm2 <- hmm[which(hmm$POS <  110913144),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 12:111350899:C:A                   splice_donor_variant 110913095                         -        -                 PASS
#2 12:111350900:C:G missense_variant,splice_region_variant 110913096 ENSP00000228841.8:p.E134D        -                 FAIL
#3 12:111350901:T:G missense_variant,splice_region_variant 110913097 ENSP00000228841.8:p.E134A        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "MYL3"),]
#MYL3 <	46,858,272 - second last exon is short
hmm2 <- hmm[which(hmm$POS < 46858272),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 3:46899716:C:G splice_donor_5th_base_variant,intron_variant 46858226          -        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "ACTC1"),]
#ACTC1 < 	34,791,114 + 55
hmm2 <- hmm[which(hmm$POS < 34791169),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

HCM1$NMD[HCM1$Consequence == "missense_variant,splice_region_variant" & (HCM1$SpliceAI_cutoff == "FAIL" | HCM1$SpliceAI_cutoff == "-")] <- "-"

# *ACTC1 - nTV
# *MYBPC3 - PAV
# *MYH7 - nTV
# *MYL2 - nTV
# *MYL3 - nTV
# *TNNI3 - nTV
# *TNNT2 - nTV
# *TPM1 - nTV

table(HCM1$Consequence)

#MYBPC3 (usually NMD escaping only for LoF mechanisms but MYBPC3 = PAV)
pav <- HCM1 %>%
  filter(SYMBOL == "MYBPC3") 
dim(pav)
#639
  
miss_nmd <- HCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "TNNT2" |
        SYMBOL == "TNNI3" |
        SYMBOL == "TPM1" |
        SYMBOL == "MYL2" |
        SYMBOL == "MYL3" |
        SYMBOL == "ACTC1") %>%
  filter(NMD == "NMD_escaping_variant")
dim(miss_nmd)
#  20

miss <- HCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "TNNT2" |
        SYMBOL == "TNNI3" |
        SYMBOL == "TPM1" |
        SYMBOL == "MYL2" |
        SYMBOL == "MYL3" |
        SYMBOL == "ACTC1") %>%
  filter(Consequence == "inframe_deletion" |
         Consequence == "inframe_deletion,splice_region_variant" |
         Consequence == "inframe_insertion" |
         Consequence == "missense_variant" |
         Consequence == "missense_variant,splice_region_variant" |
         Consequence == "start_lost" |
         Consequence == "stop_lost")
dim(miss)
#711

HCM1$filter <- ifelse((HCM1$ID %in% pav$ID | HCM1$ID %in% miss_nmd$ID | HCM1$ID %in% miss$ID), "TRUE", "FALSE")
table(HCM1$filter)
#FALSE  TRUE 
#    13  1370 

HCMfilt <- HCM1[which(HCM1$filter == "TRUE"),]
dim(HCMfilt)
#1370
```

# DCM specific filtering (assess variants, NMDi, filter for TTNPSI>90 exons, filter for disease-associated variant consequences)

```{r specific genes only - DCM}
#check LC variants
table(DCM$Consequence)
#no LC LoF variants remain [not seen in >0 cases]

#no variant consequences present that require additional curation

#NMD
table(DCM$NMD)
#86 = NMD_escaping_variant

doneant2 <- DCM

PAV_List=c("frameshift_variant",
           "stop_gained",
           "splice_donor_variant",
           "splice_acceptor_variant",
           "splice_region_variant",
           "splice_polypyrimidine_tract_variant",
           "splice_donor_5th_base_variant",
           "splice_donor_region_variant",
           "splice_confirmed")

hmmx <- doneant2[grepl(paste(PAV_List, collapse="|"), doneant2$Consequence),]
dim(hmmx)
#295

#frameshift use HGVSp and ID frameshifts upstream to this that could have a PTC at 55 bp into penultimate exon ***
hmmx$hgvk <- hmmx$HGVSp
hmmx$hgvk <- gsub("\\?", 0, hmmx$hgvk)
hmmx <- hmmx %>% separate(hgvk, c("hgvka", "hgvkb", "fs"), sep="\\*")
hmmx$fs <- as.integer(as.character(hmmx$fs)) 
hmmx$fs <- (hmmx$fs*3)

hmmx$POS <- as.integer(as.character(hmmx$POS))
hmmx$pos <- hmmx$POS

for(i in (1:nrow(hmmx))){
   if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") & (is.na(hmmx$fs[i]))) {
    hmmx$POS[i] <- hmmx$pos[i]
  }
   else if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") &
    (hmmx$SYMBOL[i] == "MYH7" | hmmx$SYMBOL[i] == "SCN5A" | hmmx$SYMBOL[i] == "TNNC1" | 
       hmmx$SYMBOL[i] == "TNNT2" | hmmx$SYMBOL[i] == "TTN")) {
    hmmx$POS[i] <- hmmx$POS[i] - hmmx$fs[i] 
        }
    else if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") & 
            (hmmx$SYMBOL[i] == "BAG3" | hmmx$SYMBOL[i] == "DES" | hmmx$SYMBOL[i] == "DSP"
            | hmmx$SYMBOL[i] == "FLNC" | hmmx$SYMBOL[i] == "LMNA" | hmmx$SYMBOL[i] == "PLN"
            | hmmx$SYMBOL[i] == "RBM20")){
            hmmx$POS[i] <- hmmx$POS[i] + hmmx$fs[i] 
    }
 
}

tail(data.frame(hmmx$Consequence, hmmx$POS),10)

hmm <- hmmx[which(hmmx$SYMBOL == "BAG3"),]
hmm2 <- hmm[which(hmm$POS > (119672656 - 55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1         10:121435985:ACC:A frameshift_variant 119676506   ENSP00000358081.4:p.TH307T*fs*11        -
#2 10:121436365:G:GGGGCTGGAGC frameshift_variant 119676883 ENSP00000358081.4:p.Q437RAGA*fs*10        -

hmm <- hmmx[which(hmmx$SYMBOL == "DES"),]
hmm2 <- hmm[which(hmm$POS >(219425745-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

hmm <- hmmx[which(hmmx$SYMBOL == "DSP"),]
hmm2 <- hmm[which(hmm$POS >(7581569-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 6:7583992:G:GAGATAGTCAT frameshift_variant  7583759 ENSP00000369129.3:p.Q2169HR*S*fs*3        -
#2   6:7584586:GTATTCGCT:G frameshift_variant  7584386  ENSP00000369129.3:p.RLL2366*fs*11        -
#3       6:7585569:CAGAA:C frameshift_variant  7585348    ENSP00000369129.3:p.QR2692*fs*4        -
#4          6:7585681:TC:T frameshift_variant  7585496    ENSP00000369129.3:p.Q2730*fs*16        -

#hmm <- hmmx[which(hmmx$SYMBOL == "FLNC"),]
#hmm2 <- hmm[which(hmm$POS > (128858217-55)),]
#data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$ClinVar)
#doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"

hmm <- hmmx[which(hmmx$SYMBOL == "LMNA"),]
hmm2 <- hmm[which(hmm$POS > (156138757-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

hmm <- hmmx[which(hmmx$SYMBOL == "MYH7"),]
hmm2 <- hmm[which(hmm$POS < (23413759+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

hmm <- hmmx[which(hmmx$SYMBOL == "PLN"),]
hmm2 <- hmm[which(hmm$POS >(118548392-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 6:118880119:T:TA frameshift_variant 118558977 ENSP00000350132.5:p.R13K*fs*7                    -
#2  6:118880200:T:G        stop_gained 118559037      ENSP00000350132.5:p.L39* NMD_escaping_variant

hmm <- hmmx[which(hmmx$SYMBOL == "RBM20"),]
hmm2 <- hmm[which(hmm$POS >(110831182-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

hmm <- hmmx[which(hmmx$SYMBOL == "SCN5A"),]
hmm2 <- hmm[which(hmm$POS <(38554279+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 3:38591991:G:A      stop_gained 38550500 ENSP00000398266.2:p.R1957* NMD_escaping_variant

hmm <- hmmx[which(hmmx$SYMBOL == "TNNC1"),]
hmm2 <- hmm[which(hmm$POS < (52451391+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 3:52485409:T:A missense_variant,splice_region_variant 52451393 ENSP00000232975.3:p.D151V        -

hmm <- hmmx[which(hmmx$SYMBOL == "TNNT2"),]
hmm2 <- hmm[which(hmm$POS < (201359623+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

hmm <- hmmx[which(hmmx$SYMBOL == "TTN"),]
hmm2 <- hmm[which(hmm$POS < (178527446+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$ClinVar)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

doneant2$NMD[doneant2$Consequence == "missense_variant,splice_region_variant" & (doneant2$SpliceAI_cutoff == "FAIL" | doneant2$SpliceAI_cutoff == "-")] <- "-"

DCM1 <- doneant2

#TTN PSI 90 regions only
psi <- read.table("TTNpsi90.txt", header=F) #a file containing BP information on exons that have high expression in the heart
DCM1$ttnfound <- ifelse(sapply(DCM1$POS, function(p) any (psi$V1 > p & psi$V2 < p)), "YES", "NO")
dcm <- DCM1[!(DCM1$ttnfound == "NO" & DCM1$SYMBOL == "TTN"),]
dim(dcm)
#744
dcm$ttnfound <- NULL

DCM1 <- dcm

# dcma$SYMBOL == "BAG3" | #PAV
# dcma$SYMBOL == "DES" | #nTV
# dcma$SYMBOL == "TTN" | #TV
# dcma$SYMBOL == "MYH7"  | #nTV
# dcma$SYMBOL == "TNNC1"  | #nTV (missense only)
# dcma$SYMBOL == "TNNT2"  | #nTV (missense only)
# dcma$SYMBOL == "LMNA"  | #PAV
# dcma$SYMBOL == "FLNC"  | #TV
# dcma$SYMBOL == "PLN"  | #PAV
# dcma$SYMBOL == "SCN5A"  | #PAV
# dcma$SYMBOL == "RBM20"| #PAV
# dcma$SYMBOL == "DSP", #PAV

table(DCM1$Consequence)

#PAV
pav <- DCM1 %>%
  filter(SYMBOL == "BAG3"|
         SYMBOL == "LMNA"|
         SYMBOL == "PLN"|
         SYMBOL == "SCN5A"|
         SYMBOL == "RBM20"|
         SYMBOL == "DSP")
dim(pav)
#372
  
miss_nmd <- DCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "DES" |
        SYMBOL == "TNNC1" |
        SYMBOL == "TNNT2") %>%
  filter(NMD == "NMD_escaping_variant")
dim(miss_nmd)
# 2

miss <- DCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "DES" |
        SYMBOL == "TNNC1" |
        SYMBOL == "TNNT2") %>%
  filter(Consequence == "inframe_deletion" |
         Consequence == "inframe_insertion" |
         Consequence == "missense_variant" |
         Consequence == "missense_variant,splice_region_variant")
dim(miss)
#171

ttn <- DCM1 %>%
  filter(SYMBOL == "TTN") %>%
  filter(Consequence == "frameshift_variant" |
         Consequence == "frameshift_variant,splice_region_variant" |
         Consequence == "missense_variant,splice_region_variant" | #curated for splicing [there are none]
         Consequence == "splice_acceptor_variant" |
         Consequence == "splice_confirmed" |
         Consequence == "splice_donor_5th_base_variant,intron_variant" |
         Consequence == "splice_donor_variant" |
         Consequence == "splice_donor_variant,coding_sequence_variant" |
         Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
         Consequence == "splice_polypyrimidine_tract_variant,intron_variant" | 
         Consequence == "stop_gained" |
         Consequence == "stop_gained,frameshift_variant" |
         Consequence == "stop_gained,splice_region_variant")
dim(ttn)
#193

DCM1$filter <- ifelse((DCM1$ID %in% pav$ID | DCM1$ID %in% miss_nmd$ID | DCM1$ID %in% miss$ID | DCM1$ID %in% ttn$ID), "TRUE", "FALSE")
DCMfilt <- DCM1[which(DCM1$filter == "TRUE"),]
dim(DCMfilt)
#738
```

# filter HCM by max AF in case cohort, gnomAD, and UKB

```{r}
dim(HCM)
# 1370

####### common, intermediate, rare PAVs
hmm <- HCM[which(HCM$LMM_AC>0),]
max(hmm$gnomADg_AF, na.rm=T)
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.00100626 gnomad 0.002314 UKB n=612
hmm <- HCM[which(HCM$OMG_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.000804405 gnomAD 0.00103 UKB n=586
hmm <- HCM[which(HCM$BLF_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.00149709 gnomAD 0.002314 UKB n=104
hmm <- HCM[which(HCM$GDX_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.00036598 gnoMAD 0.0007344 UKB n=410
hmm <- HCM[which(HCM$SNG_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.0964262 gnomAD 0.131 UKB n=35
hmm <- HCM[which(HCM$EGY_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.0964262 gnomAD 0.131 UKB n=145
hmm <- HCM[which(HCM$RBH_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.0964262 gnomAD 0.131 UKB n=167
hmm <- HCM[which(HCM$SLD_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.0964262 gnomAD 0.148522 faf 0.131 UKB n=84

#min = GDX
#0.00036598 gnoMAD 0.0007344 UKB n=410

full5 <- HCM[which(HCM$gnomADg_AF <= 0.00036598 | is.na(HCM$gnomADg_AF)),]
dim(full5)
#1341

full5a <- full5[which(full5$ukbMAF_qced <= 0.0007344 | is.na(full5$ukbMAF_qced)),]
dim(full5a)
#1340

full5a$gnomADg_AC0[full5a$gnomADg_AC == 0] <- "AC0" #call those flagged as bad by gnomAD

which( colnames(full5a)=="gnomADg" )
which( colnames(full5a)=="gnomADg_AN_nfe_female" )

for (i in 1:nrow(full5a)) {
 if (full5a$gnomADg_AC0[i] == "AC0") {
   full5a[i,50:96] <- NA #all gnomAD data - call it NA
 }
}

write.table(full5a, "HCMx.txt", quote=F, row.names=F, sep="\t")

table(full5a$SYMBOL)
#ACTC1 MYBPC3   MYH7   MYL2   MYL3  TNNI3  TNNT2   TPM1 
#   17    616    459     42     33     64     62     47 
```

# filter DCM by max AF in case cohort gnomAD and UKB (QCed) + add counts [dcmx.txt]

```{r}
### DCM
DCM <- read.table("DCMfilt.txt", header=T, sep="\t")
dim(DCM)
# 738 

####### common, intermediate, rare PAVs
hmm <- DCM[which(DCM$LMM_AC>0),]
max(hmm$gnomADg_AF, na.rm=T)
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.00927996 gnomad 0.004344 UKB n=156
hmm <- DCM[which(DCM$OMG_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.000552987 gnomAD 0.0006031 UKB n=111
hmm <- DCM[which(DCM$SNG_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.803171 gnomAD 0.8533 UKB n=74
hmm <- DCM[which(DCM$EGY_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.803171 gnomAD 0.8533 UKB n=54
hmm <- DCM[which(DCM$RBH_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.803171 gnomAD 0.8533 UKB n=3
hmm <- DCM[which(DCM$LM2_AC>0),]
max(hmm$gnomADg_AF, na.rm=T) 
max(hmm$ukbMAF_qced, na.rm=T)
dim(hmm)
#0.00141774 gnomAD 0.00266 UKB n=85

#min = OMG
#0.000552987 gnomAD 0.0006031 UKB n=111

full5 <- DCM[which(DCM$gnomADg_AF <= 0.000552987 | is.na(DCM$gnomADg_AF)),]
dim(full5)
#671

full5a <- full5[which(full5$ukbMAF_qced <= 0.0006031 | is.na(full5$ukbMAF_qced)),]
dim(full5a)
#665

full5a$gnomADg_AC0 <- as.character(full5a$gnomADg_AC0)
full5a$gnomADg_AC0[full5a$gnomADg_AC == 0] <- "AC0" #call those flagged as bad by gnomAD
full5a$gnomADg_AN <- as.character(full5a$gnomADg_AN) 

which( colnames(full5a)=="gnomADg" )
which( colnames(full5a)=="gnomADg_AN_nfe_female" )

for (i in 1:nrow(full5a)) {
 if (full5a$gnomADg_AC0[i] == "AC0") {
   full5a[i,50:96] <- NA #all gnomAD data - call it NA
 }
}

write.table(full5a, "DCMx.txt", quote=F, row.names=F, sep="\t")

table(full5a$SYMBOL)
#BAG3   DES   DSP  LMNA  MYH7   PLN RBM20 SCN5A TNNC1 TNNT2   TTN 
# 22    25   113    60   109     5    54    54     9    23   191
```

# variants to be included in the analysis

```{r}
HCMx <- read.table("HCMx.txt", header=T, sep="\t")

HCMxfilt <- HCMx %>%
  filter(AC_ALL_HCM > 1) %>%
  filter(gnukb_AC > 1)

dim(HCMxfilt)
# 257

write.table(HCMxfilt, "HCMpen.txt", quote=F, row.names=F, sep="\t")

####DCM

DCMx <- read.table("DCMx.txt", header=T, sep="\t")

DCMxfilt <- DCMx %>%
  filter(AC_ALL_DCM > 1) %>%
  filter(gnukb_AC > 1)

dim(DCMxfilt)
# 59

write.table(DCMxfilt, "DCMpen.txt", quote=F, row.names=F, sep="\t")
```

# Sex prevalences

```{r}
# ref malecase malepop femalecase femalepop
# "Zou et al. 2004" 9 4064 4 4016
# "Maron et al. 2004" 4 1316 4 2185
# "Maron et al. 1995" 5 1913 2 2198
# "de Marvao et al. 2021" 58 18820 14 20454

# DCM
# ref malecase malepop femalecase femalepop
# UKB 117 18687 60 20316
#=~ 1 in 160 for men and 1 in 340 for women
60/20316 *100 #women
#0.295%
binom.test(60,20316)
#0.002254441 0.003799918

117/18687 *100
#0.626%
binom.test(117,18687)
#0.005180719 0.007499021

library(tidyverse)
library(meta)
library(metafor)

#https://rpubs.com/pekong/532068
#https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/metareg.html#metareg-R
#https://www.rdocumentation.org/packages/meta/versions/4.18-2/topics/forest.meta

#### MEN
meta <- metaprop(as.integer(malecase), malepop, studlab=ref, sm="PFT", data=hcm, method="Inverse", method.tau="DL")
summary(meta)
#                       proportion           95%-CI %W(common) %W(random)
# Zou et al. 2004           0.0022 [0.0010; 0.0042]       15.6       15.6
# Maron et al. 2004         0.0030 [0.0008; 0.0078]        5.0        5.0
# Maron et al. 1995         0.0026 [0.0008; 0.0061]        7.3        7.3
# de Marvao et al. 2021     0.0031 [0.0023; 0.0040]       72.1       72.1
# 
# Number of studies combined: k = 4
# Number of observations: o = 26113
# Number of events: e = 76
# 
#                      proportion           95%-CI
# Common effect model      0.0028 [0.0022; 0.0035]
# Random effects model     0.0028 [0.0022; 0.0035]
# 
# Quantifying heterogeneity:
#  tau^2 = 0 [0.0000; 0.0002]; tau = 0 [0.0000; 0.0123]
#  I^2 = 0.0% [0.0%; 84.7%]; H = 1.00 [1.00; 2.56]
# 
# Test of heterogeneity:
#     Q d.f. p-value
#  0.82    3  0.8435
# 
# Details on meta-analytical method:
# - Inverse variance method
# - DerSimonian-Laird estimator for tau^2
# - Jackson method for confidence interval of tau^2 and tau
# - Freeman-Tukey double arcsine transformation
# - Clopper-Pearson confidence interval for individual studies

forest.meta(meta, layout="meta", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.008), fontsize=10, digits=6, col.square="black", col.square.lines="black", col.inside="black")

######

rm(list = ls())
set.seed(28061971)
digits <- 6   # ***
alpha <- 0.05  # ***
z <- qnorm(1 - alpha /2)
# Input from meta-analysis #
#0.002784 [0.002161; 0.003481]
phat <- 0.002784   # From meta-analysis results
qhat <- 1 - phat
CI_phat_LL <- 0.002161   # From meta-analysis results
CI_phat_UL <- 0.003481   # From meta-analysis results

# Agresti–Coull method #

nStore <- NULL
xStore <- NULL

dif <- 1
C0 <- 1e-6   # ***
iter <- 1

while (dif > 0)
{
    ptilda <- phat + C0 * (iter - 1)
    qtilda <- 1 - ptilda
    
    n <- ceiling(ptilda * qtilda / ((ptilda - CI_phat_LL) / z) ^2 - z ^2)
    ntilda <- n + z ^2
    xtilda <- ntilda * ptilda
    x <- ceiling(xtilda - z ^2 /2)
    pest <- x / n
    
    nStore <- c(nStore, n)
    xStore <- c(xStore, x)
    
    dif <- phat - pest
    
    print(c("Iter:", iter))
    print(c("phat:", round(phat, digits = digits), "pest:", round(pest, digits = digits)))
    print(c("ptilda:", round((x + z ^2 /2) / (n + z ^2), digits = digits), "ptildaest:", round((phat + z ^2 / (2 * n)) / (1 + z ^2 / n), digits = digits)))
    
    print(c("Agresti–Coull method estimated n:", n, "Agresti–Coull method estimated x:", x))
    print(c("Agresti–Coull method estimated ntilda:", round(ntilda, digits = digits), "Agresti–Coull method estimated xtilda:", round(xtilda, digits = digits)))
    
    iter <- iter + 1
}

c("Agresti–Coull method n:", n, "Agresti–Coull method x:", x)
#"Agresti–Coull method n:" "24411"                   "Agresti–Coull method x:" "68"                
c("Agresti–Coull method ntilda:", round(ntilda, digits = digits), "Agresti–Coull method xtilda:", round(xtilda, digits = digits))
#"Agresti–Coull method ntilda:" "24414.841459"                 "Agresti–Coull method xtilda:""69.020757"      

#68 in 24411 == HCM Men

#### Women
meta <- metaprop(as.integer(femalecase), femalepop, studlab=ref, sm="PFT", data=hcm, method="Inverse", method.tau="DL")
summary(meta)
#                       proportion           95%-CI %W(common) %W(random)
# Zou et al. 2004           0.0010 [0.0003; 0.0025]       13.9       16.5
# Maron et al. 2004         0.0018 [0.0005; 0.0047]        7.6        9.3
# Maron et al. 1995         0.0009 [0.0001; 0.0033]        7.6        9.3
# de Marvao et al. 2021     0.0007 [0.0004; 0.0011]       70.9       65.0
# 
# Number of studies combined: k = 4
# Number of observations: o = 28853
# Number of events: e = 24
# 
#                      proportion           95%-CI
# Common effect model      0.0007 [0.0004; 0.0011]
# Random effects model     0.0008 [0.0004; 0.0012]
# 
# Quantifying heterogeneity:
#  tau^2 < 0.0001 [0.0000; 0.0008]; tau = 0.0022 [0.0000; 0.0291]
#  I^2 = 7.9% [0.0%; 85.9%]; H = 1.04 [1.00; 2.66]
# 
# Test of heterogeneity:
#     Q d.f. p-value
#  3.26    3  0.3539
# 
# Details on meta-analytical method:
# - Inverse variance method
# - DerSimonian-Laird estimator for tau^2
# - Jackson method for confidence interval of tau^2 and tau
# - Freeman-Tukey double arcsine transformation
# - Clopper-Pearson confidence interval for individual studies

forest.meta(meta, layout="meta", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.005), fontsize=10, digits=6, col.square="black", col.square.lines="black", col.inside="black")

######

rm(list = ls())
set.seed(28061971)
digits <- 6   # ***
alpha <- 0.05  # ***
z <- qnorm(1 - alpha /2)
# Input from meta-analysis #
#0.000762 [0.000416; 0.001194] #random effects
phat <- 0.000762   # From meta-analysis results
qhat <- 1 - phat
CI_phat_LL <- 0.000416   # From meta-analysis results
CI_phat_UL <- 0.001194   # From meta-analysis results

# Agresti–Coull method #

nStore <- NULL
xStore <- NULL

dif <- 1
C0 <- 1e-6   # ***
iter <- 1

while (dif > 0)
{
    ptilda <- phat + C0 * (iter - 1)
    qtilda <- 1 - ptilda
    
    n <- ceiling(ptilda * qtilda / ((ptilda - CI_phat_LL) / z) ^2 - z ^2)
    ntilda <- n + z ^2
    xtilda <- ntilda * ptilda
    x <- ceiling(xtilda - z ^2 /2)
    pest <- x / n
    
    nStore <- c(nStore, n)
    xStore <- c(xStore, x)
    
    dif <- phat - pest
    
    print(c("Iter:", iter))
    print(c("phat:", round(phat, digits = digits), "pest:", round(pest, digits = digits)))
    print(c("ptilda:", round((x + z ^2 /2) / (n + z ^2), digits = digits), "ptildaest:", round((phat + z ^2 / (2 * n)) / (1 + z ^2 / n), digits = digits)))
    
    print(c("Agresti–Coull method estimated n:", n, "Agresti–Coull method estimated x:", x))
    print(c("Agresti–Coull method estimated ntilda:", round(ntilda, digits = digits), "Agresti–Coull method estimated xtilda:", round(xtilda, digits = digits)))
    
    iter <- iter + 1
}

c("Agresti–Coull method n:", n, "Agresti–Coull method x:", x)
#"Agresti–Coull method n:" "19646"                   "Agresti–Coull method x:" "15"                
c("Agresti–Coull method ntilda:", round(ntilda, digits = digits), "Agresti–Coull method xtilda:", round(xtilda, digits = digits))
#"Agresti–Coull method ntilda:"  "19649.841459"                  "Agresti–Coull method xtilda:"16.014621"      

#15 in 19646 == HCM Women
```

# Penetrance - specific variants

# HCM estimate penetrance

```{r}
df <- read.table("HCMpen.txt", header=T, sep="\t")
dim(df)
#256

#### as a function
penetrance <- function(x_D, n_D, x_AgD, n_AgD, x_A, n_A){
set.seed(28061971)
digits <- 6   
alpha <- 0.05  
p_D <- x_D / n_D
p_AgD <- x_AgD / n_AgD
p_A <- x_A / n_A
d <- 0.5   
p_D_wod <- (x_D + d) / (n_D + d)
p_AgD_wod <- (x_AgD + d) / (n_AgD + d)
p_A_wod <- (x_A + d) / (n_A + d)
log_AR <- log(p_D_wod * p_AgD_wod / p_A_wod) + 1/2 * ((1 / p_A_wod) * (1 - p_A_wod) / (n_A + d) - 
                                                       (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) - 
                                                       (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d))
Var_log_AR <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
              (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
              (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_LCI <- log_AR - qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
log_UCI <- log_AR + qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
penetrance <- pmin(1,pmax(0,exp(log_AR)))
log_lCI <- pmin(1,pmax(0,exp(log_LCI)))
log_uCI <- pmin(1,pmax(0,exp(log_UCI)))
my_list <- list("penetrance" = penetrance, "lci" = log_lCI, "uci" = log_uCI)
return(my_list)
}

names(df)

#total
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_ALL_HCM, n_AgD=df$AN_ALL_HCM, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance", "lci", "uci")

#EUR
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_EUR, n_AgD=df$AN_HCM_EUR, x_A=df$gnukb_AC_eu, n_A=df$gnukb_AN_eu)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_eu", "lci_eu", "uci_eu")

#nwe
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_EUR, n_AgD=df$AN_HCM_EUR, x_A=df$gnukb_AC_nwe, n_A=df$gnukb_AN_nwe)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_nwe", "lci_nwe", "uci_nwe")

#gnomad only
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_ALL_HCM, n_AgD=df$AN_ALL_HCM, x_A=df$gnomADg_AC, n_A=df$gnomADg_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_gnomad", "lci_gnomad", "uci_gnomad")

#ukb only
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_ALL_HCM, n_AgD=df$AN_ALL_HCM, x_A=df$ukbAC_qced, n_A=df$ukbAN_qced)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_ukb", "lci_ukb", "uci_ukb")

#NWE - gnomAD and UKB separate
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_EUR, n_AgD=df$AN_HCM_EUR, x_A=df$ukbAC_brit, n_A=df$ukbAN_brit)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_ukb_brit", "lci_ukb_brit", "uci_ukb_brit")

df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_EUR, n_AgD=df$AN_HCM_EUR, x_A=df$gnomADg_AC_nfe_nwe, n_A=df$gnomADg_AN_nfe_nwe)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_gnomad_nwe", "lci_gnomad_nwe", "uci_gnomad_nwe")

#EU - gnomAD and UKB separate
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_EUR, n_AgD=df$AN_HCM_EUR, x_A=df$ukbAC_eu, n_A=df$ukbAN_eu)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_ukb_eu", "lci_ukb_eu", "uci_ukb_eu")

df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_EUR, n_AgD=df$AN_HCM_EUR, x_A=df$gnomADg_AC_nfe, n_A=df$gnomADg_AN_nfe)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_gnomad_eu", "lci_gnomad_eu", "uci_gnomad_eu")

#afr
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_AFR, n_AgD=df$AN_HCM_AFR, x_A=df$gnukb_AC_afr, n_A=df$gnukb_AN_afr)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_afr", "lci_afr", "uci_afr")

#eas
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_EAS, n_AgD=df$AN_HCM_EAS, x_A=df$gnukb_AC_eas, n_A=df$gnukb_AN_eas)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_eas", "lci_eas", "uci_eas")

#sas
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=97, n_D=52660, x_AgD=df$AC_HCM_SAS, n_AgD=df$AN_HCM_SAS, x_A=df$gnukb_AC_sas, n_A=df$gnukb_AN_sas)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_sas", "lci_sas", "uci_sas")

#male
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=68, n_D=24411, x_AgD=df$AC_HCM_male, n_AgD=df$AN_HCM_male, x_A=df$gnukb_AC_male, n_A=df$gnukb_AN_male)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_male", "lci_male", "uci_male")

#female
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=15, n_D=19646, x_AgD=df$AC_HCM_female, n_AgD=df$AN_HCM_female, x_A=df$gnukb_AC_female, n_A=df$gnukb_AN_female)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_female", "lci_female", "uci_female")

#0s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_0s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_0s, n_AgD=df$AN_HCM_0s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_0s", "lci_0s", "uci_0s")

#10s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_10s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_10s, n_AgD=df$AN_HCM_10s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_10s", "lci_10s", "uci_10s")

#20s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_20s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_20s, n_AgD=df$AN_HCM_20s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_20s", "lci_20s", "uci_20s")

#30s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_30s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_30s, n_AgD=df$AN_HCM_30s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_30s", "lci_30s", "uci_30s")

#40s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_40s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_40s, n_AgD=df$AN_HCM_40s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_40s", "lci_40s", "uci_40s")

#50s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_50s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_50s, n_AgD=df$AN_HCM_50s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_50s", "lci_50s", "uci_50s")

#60s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_60s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_60s, n_AgD=df$AN_HCM_60s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_60s", "lci_60s", "uci_60s")

#70s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_70s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_70s, n_AgD=df$AN_HCM_70s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_70s", "lci_70s", "uci_70s")

#80s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(97*(df$AN_HCM_80s/df$AN_HCM_80s)), n_D=(52660), x_AgD=df$AC_HCM_80s, n_AgD=df$AN_HCM_80s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_80s", "lci_80s", "uci_80s")

write.table(df, "HCM_penetrance.txt", quote=F, row.names=F, sep="\t")
```

# DCM estimate penetrance

```{r}
df <- read.table("DCMpen.txt", header=T, sep="\t")
dim(df)
#59

#### as a function
penetrance <- function(x_D, n_D, x_AgD, n_AgD, x_A, n_A){
set.seed(28061971)
digits <- 6   
alpha <- 0.05  
p_D <- x_D / n_D
p_AgD <- x_AgD / n_AgD
p_A <- x_A / n_A
d <- 0.5   
p_D_wod <- (x_D + d) / (n_D + d)
p_AgD_wod <- (x_AgD + d) / (n_AgD + d)
p_A_wod <- (x_A + d) / (n_A + d)
log_AR <- log(p_D_wod * p_AgD_wod / p_A_wod) + 1/2 * ((1 / p_A_wod) * (1 - p_A_wod) / (n_A + d) - 
                                                       (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) - 
                                                       (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d))
Var_log_AR <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
              (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
              (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_LCI <- log_AR - qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
log_UCI <- log_AR + qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
penetrance <- pmin(1,pmax(0,exp(log_AR)))
log_lCI <- pmin(1,pmax(0,exp(log_LCI)))
log_uCI <- pmin(1,pmax(0,exp(log_UCI)))
my_list <- list("penetrance" = penetrance, "lci" = log_lCI, "uci" = log_uCI)
return(my_list)
}

names(df)
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_ALL_DCM, n_AgD=df$AN_ALL_DCM, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance", "lci", "uci")

#EUR
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_EUR, n_AgD=df$AN_DCM_EUR, x_A=df$gnukb_AC_eu, n_A=df$gnukb_AN_eu)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_eu", "lci_eu", "uci_eu")

#nwe
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_EUR, n_AgD=df$AN_DCM_EUR, x_A=df$gnukb_AC_nwe, n_A=df$gnukb_AN_nwe)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_nwe", "lci_nwe", "uci_nwe")

#gnomad only
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_ALL_DCM, n_AgD=df$AN_ALL_DCM, x_A=df$gnomADg_AC, n_A=df$gnomADg_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_gnomad", "lci_gnomad", "uci_gnomad")

#ukb only
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_ALL_DCM, n_AgD=df$AN_ALL_DCM, x_A=df$ukbAC_qced, n_A=df$ukbAN_qced)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_ukb", "lci_ukb", "uci_ukb")

#NWE - gnomAD and UKB separate
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_EUR, n_AgD=df$AN_DCM_EUR, x_A=df$ukbAC_brit, n_A=df$ukbAN_brit)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_ukb_brit", "lci_ukb_brit", "uci_ukb_brit")

df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_EUR, n_AgD=df$AN_DCM_EUR, x_A=df$gnomADg_AC_nfe_nwe, n_A=df$gnomADg_AN_nfe_nwe)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_gnomad_nwe", "lci_gnomad_nwe", "uci_gnomad_nwe")

#EU - gnomAD and UKB separate
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_EUR, n_AgD=df$AN_DCM_EUR, x_A=df$ukbAC_eu, n_A=df$ukbAN_eu)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_ukb_eu", "lci_ukb_eu", "uci_ukb_eu")

df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_EUR, n_AgD=df$AN_DCM_EUR, x_A=df$gnomADg_AC_nfe, n_A=df$gnomADg_AN_nfe)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_gnomad_eu", "lci_gnomad_eu", "uci_gnomad_eu")

#afr
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_AFR, n_AgD=df$AN_DCM_AFR, x_A=df$gnukb_AC_afr, n_A=df$gnukb_AN_afr)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_afr", "lci_afr", "uci_afr")

#eas
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_EAS, n_AgD=df$AN_DCM_EAS, x_A=df$gnukb_AC_eas, n_A=df$gnukb_AN_eas)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_eas", "lci_eas", "uci_eas")

#sas
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=177, n_D=39003, x_AgD=df$AC_DCM_SAS, n_AgD=df$AN_DCM_SAS, x_A=df$gnukb_AC_sas, n_A=df$gnukb_AN_sas)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_sas", "lci_sas", "uci_sas")

#male
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=116, n_D=18687, x_AgD=df$AC_DCM_male, n_AgD=df$AN_DCM_male, x_A=df$gnukb_AC_male, n_A=df$gnukb_AN_male)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_male", "lci_male", "uci_male")

#female
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=60, n_D=20316, x_AgD=df$AC_DCM_female, n_AgD=df$AN_DCM_female, x_A=df$gnukb_AC_female, n_A=df$gnukb_AN_female)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_female", "lci_female", "uci_female")

#0s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_0s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_0s, n_AgD=df$AN_DCM_0s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_0s", "lci_0s", "uci_0s")

#10s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_10s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_10s, n_AgD=df$AN_DCM_10s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_10s", "lci_10s", "uci_10s")

#20s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_20s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_20s, n_AgD=df$AN_DCM_20s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_20s", "lci_20s", "uci_20s")

#30s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_30s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_30s, n_AgD=df$AN_DCM_30s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_30s", "lci_30s", "uci_30s")

#40s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_40s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_40s, n_AgD=df$AN_DCM_40s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_40s", "lci_40s", "uci_40s")

#50s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_50s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_50s, n_AgD=df$AN_DCM_50s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_50s", "lci_50s", "uci_50s")

#60s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_60s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_60s, n_AgD=df$AN_DCM_60s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_60s", "lci_60s", "uci_60s")

#70s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_70s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_70s, n_AgD=df$AN_DCM_70s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_70s", "lci_70s", "uci_70s")

#80s
df[(ncol(df)+1):(ncol(df)+3)] <- penetrance(x_D=(177*(df$AN_DCM_80s/df$AN_DCM_80s)), n_D=(39003), x_AgD=df$AC_DCM_80s, n_AgD=df$AN_DCM_80s, x_A=df$gnukb_AC, n_A=df$gnukb_AN)
colnames(df)[(ncol(df)-2):(ncol(df))] <- c("penetrance_80s", "lci_80s", "uci_80s")

write.table(df, "DCM_penetrance.txt", quote=F, row.names=F, sep="\t")
```

# ancestry

```{r}
##HCM
df2 <- read.table("HCM_penetrance.txt", header=T, sep="\t")
names(df2)

df2$AF_HCM_AFR <- df2$AC_HCM_AFR/df2$AN_HCM_AFR
df2$AF_HCM_EUR <- df2$AC_HCM_EUR/df2$AN_HCM_EUR
df2$AF_HCM_SAS <- df2$AC_HCM_SAS/df2$AN_HCM_SAS
df2$AF_HCM_EAS <- df2$AC_HCM_EAS/df2$AN_HCM_EAS

df2$popmax <- ""

for(i in (1:nrow(df2))){
   if((df2$AF_HCM_AFR[i] > df2$AF_HCM_EUR[i]) & df2$gnukb_AC_afr[i] > 1 & df2$AC_HCM_AFR[i] > 1) {
    df2$popmax[i] <- "AFR"
  }
   else if((df2$AF_HCM_EAS[i] > df2$AF_HCM_EUR[i]) & df2$gnukb_AC_eas[i] > 1 & df2$AC_HCM_EAS[i] > 1) {
    df2$popmax[i] <- "EAS"
   }
    else if((df2$AF_HCM_SAS[i] > df2$AF_HCM_EUR[i]) & df2$gnukb_AC_sas[i] > 1 & df2$AC_HCM_SAS[i] > 1) {
    df2$popmax[i] <- "SAS"
    }
  else if(df2$gnukb_AC_eu[i] > 1 & df2$AC_HCM_EUR[i] > 1) {
    df2$popmax[i] <- "EUR"
  }
}

df2$popmax[df2$popmax == ""] <- "ALL"
table(df2$popmax)
# AFR ALL EAS EUR 
#   8  53   3 193 

### DCM
df2 <- read.table("DCM_penetrance.txt", header=T, sep="\t")
names(df2)

df2$AF_DCM_AFR <- df2$AC_DCM_AFR/df2$AN_DCM_AFR
df2$AF_DCM_EUR <- df2$AC_DCM_EUR/df2$AN_DCM_EUR
df2$AF_DCM_SAS <- df2$AC_DCM_SAS/df2$AN_DCM_SAS
df2$AF_DCM_EAS <- df2$AC_DCM_EAS/df2$AN_DCM_EAS

df2$popmax <- ""

for(i in (1:nrow(df2))){
   if((df2$AF_DCM_AFR[i] > df2$AF_DCM_EUR[i]) & df2$gnukb_AC_afr[i] > 1 & df2$AC_DCM_AFR[i] > 1) {
    df2$popmax[i] <- "AFR"
  }
   else if((df2$AF_DCM_EAS[i] > df2$AF_DCM_EUR[i]) & df2$gnukb_AC_eas[i] > 1 & df2$AC_DCM_EAS[i] > 1) {
    df2$popmax[i] <- "EAS"
   }
    else if((df2$AF_DCM_SAS[i] > df2$AF_DCM_EUR[i]) & df2$gnukb_AC_sas[i] > 1 & df2$AC_DCM_SAS[i] > 1) {
    df2$popmax[i] <- "SAS"
    }
  else if(df2$gnukb_AC_eu[i] > 1 & df2$AC_DCM_EUR[i] > 1) {
    df2$popmax[i] <- "EUR"
  }
}

df2$popmax[df2$popmax == ""] <- "ALL"
table(df2$popmax)
# AFR ALL EAS EUR 
#  6  15   3  35 
```

# age

```{r}
# diag_group  age_mean    age_min age_max
# EGY_DCM 38.925  2   72
# EGY_HCM 36.84682713347921   0   77
# RBH_DCM 54.91467576791809   0   90
# RBH_HCM 54.5    7   89
# SLD_HCM 56.96078431372549   18  85

##HCM
df2 <- read.table("HCM_penetrance.txt", header=T, sep="\t")

df2$AF_HCM_0s <- df2$AC_HCM_0s/df2$AN_HCM_0s
df2$AF_HCM_10s <- df2$AC_HCM_10s/df2$AN_HCM_10s
df2$AF_HCM_20s <- df2$AC_HCM_20s/df2$AN_HCM_20s
df2$AF_HCM_30s <- df2$AC_HCM_30s/df2$AN_HCM_30s
df2$AF_HCM_40s <- df2$AC_HCM_40s/df2$AN_HCM_40s
df2$AF_HCM_50s <- df2$AC_HCM_50s/df2$AN_HCM_50s
df2$AF_HCM_60s <- df2$AC_HCM_60s/df2$AN_HCM_60s
df2$AF_HCM_70s <- df2$AC_HCM_70s/df2$AN_HCM_70s
df2$AF_HCM_80s <- df2$AC_HCM_80s/df2$AN_HCM_80s

df2$popmaxage <- ""

for(i in (1:nrow(df2))){
   if((df2$AF_HCM_0s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_0s[i] > 1) {
    df2$popmaxage[i] <- "0s"
   }
  else if((df2$AF_HCM_10s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_10s[i] > 1) {
    df2$popmaxage[i] <- "10s"
  }
  else if((df2$AF_HCM_20s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_20s[i] > 1) {
    df2$popmaxage[i] <- "20s"
  }
  else if((df2$AF_HCM_30s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_30s[i] > 1) {
    df2$popmaxage[i] <- "30s"
   }
   else if((df2$AF_HCM_40s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_40s[i] > 1) {
    df2$popmaxage[i] <- "40s"
   }
    else if((df2$AF_HCM_50s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_50s[i] > 1) {
    df2$popmaxage[i] <- "50s"
    }
   else if((df2$AF_HCM_60s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_60s[i] > 1) {
    df2$popmaxage[i] <- "60s"
   }
  else if((df2$AF_HCM_70s[i] < df2$AF_HCM_80s[i]) & df2$AC_HCM_70s[i] > 1) {
    df2$popmaxage[i] <- "70s"
    }
  else if(df2$AC_HCM_80s[i] > 1) {
    df2$popmaxage[i] <- "80s"
  }
}

df2$popmaxage[df2$popmaxage == ""] <- "ALL"
table(df2$popmaxage)
#30s 40s 50s 60s 70s ALL
# 5   5   5  18   8  77 139

#### DCM
df2 <- read.table("DCM_penetrance.txt", header=T, sep="\t")
names(df2)

df2$AF_DCM_0s <- df2$AC_DCM_0s/df2$AN_DCM_0s
df2$AF_DCM_10s <- df2$AC_DCM_10s/df2$AN_DCM_10s
df2$AF_DCM_20s <- df2$AC_DCM_20s/df2$AN_DCM_20s
df2$AF_DCM_30s <- df2$AC_DCM_30s/df2$AN_DCM_30s
df2$AF_DCM_40s <- df2$AC_DCM_40s/df2$AN_DCM_40s
df2$AF_DCM_50s <- df2$AC_DCM_50s/df2$AN_DCM_50s
df2$AF_DCM_60s <- df2$AC_DCM_60s/df2$AN_DCM_60s
df2$AF_DCM_70s <- df2$AC_DCM_70s/df2$AN_DCM_70s
df2$AF_DCM_80s <- df2$AC_DCM_80s/df2$AN_DCM_80s

df2$popmaxage <- ""

for(i in (1:nrow(df2))){
   if((df2$AF_DCM_0s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_0s[i] > 1) {
    df2$popmaxage[i] <- "0s"
   }
  else if((df2$AF_DCM_10s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_10s[i] > 1) {
    df2$popmaxage[i] <- "10s"
  }
  else if((df2$AF_DCM_20s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_20s[i] > 1) {
    df2$popmaxage[i] <- "20s"
  }
  else if((df2$AF_DCM_30s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_30s[i] > 1) {
    df2$popmaxage[i] <- "30s"
   }
   else if((df2$AF_DCM_40s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_40s[i] > 1) {
    df2$popmaxage[i] <- "40s"
   }
    else if((df2$AF_DCM_50s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_50s[i] > 1) {
    df2$popmaxage[i] <- "50s"
    }
   else if((df2$AF_DCM_60s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_60s[i] > 1) {
    df2$popmaxage[i] <- "60s"
   }
  else if((df2$AF_DCM_70s[i] < df2$AF_DCM_80s[i]) & df2$AC_DCM_70s[i] > 1) {
    df2$popmaxage[i] <- "70s"
    }
  else if(df2$AC_DCM_80s[i] > 1) {
    df2$popmaxage[i] <- "80s"
  }
}

df2$popmaxage[df2$popmaxage == ""] <- "ALL"
table(df2$popmaxage)
# 60s 80s ALL 
#   2  24  33  
```

# sex

```{r}
##HCM
df2 <- read.table("HCM_penetrance.txt", header=T, sep="\t")
names(df2)

df2$AF_HCM_male <- df2$AC_HCM_male/df2$AN_HCM_male
df2$AF_HCM_female <- df2$AC_HCM_female/df2$AN_HCM_female

df2$popmaxsex <- ""

for(i in (1:nrow(df2))){
   if((df2$AF_HCM_male[i] > df2$AF_HCM_female[i]) & df2$gnukb_AC_male[i] > 1 & df2$AC_HCM_male[i] > 1) {
    df2$popmaxsex[i] <- "male"
  }
   else if((df2$AF_HCM_female[i] > df2$AF_HCM_male[i]) & df2$gnukb_AC_female[i] > 1 & df2$AC_HCM_female[i] > 1) {
    df2$popmaxsex[i] <- "female"
   }
}

df2$popmaxsex[df2$popmaxsex == ""] <- "both"
table(df2$popmaxsex)
#  both female   male 
#   171     38     48 

##DCM
df2 <- read.table("DCM_penetrance.txt", header=T, sep="\t")

df2$AF_DCM_male <- df2$AC_DCM_male/df2$AN_DCM_male
df2$AF_DCM_female <- df2$AC_DCM_female/df2$AN_DCM_female

df2$popmaxsex <- ""

for(i in (1:nrow(df2))){
   if((df2$AF_DCM_male[i] > df2$AF_DCM_female[i]) & df2$gnukb_AC_male[i] > 1 & df2$AC_DCM_male[i] > 1) {
    df2$popmaxsex[i] <- "male"
  }
   else if((df2$AF_DCM_female[i] > df2$AF_DCM_male[i]) & df2$gnukb_AC_female[i] > 1 & df2$AC_DCM_female[i] > 1) {
    df2$popmaxsex[i] <- "female"
   }
}

df2$popmaxsex[df2$popmaxsex == ""] <- "both"
table(df2$popmaxsex)
#  both female   male 
#   39      6     14 
```

# OR

```{r}
HCM$Var <- HCM$AC_ALL_HCM
HCM$NoVar <- HCM$AN_ALL_HCM - HCM$AC_ALL_HCM
HCM$Var2p <- HCM$gnukb_AC
HCM$NoVarp <- HCM$gnukb_AN - HCM$gnukb_AC

HCM[(ncol(HCM)+1)] <- apply(HCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(HCM)-3):(ncol(HCM))]), ncol=2, byrow=T)
          fisher.test(tbl, alternative="two.sided")$estimate
              })

HCM[(ncol(HCM)+1)] <- apply(HCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(HCM)-3-1):(ncol(HCM)-1)]), ncol=2, byrow=T)
          fisher.test(tbl, alternative="two.sided")$p.value
              })

HCM[(ncol(HCM)+1)] <- apply(HCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(HCM)-3-2):(ncol(HCM)-2)]), ncol=2, byrow=T)
            fisher.test(tbl, alternative="two.sided")$conf.int[1]
              })

HCM[(ncol(HCM)+1)] <- apply(HCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(HCM)-3-3):(ncol(HCM)-3)]), ncol=2, byrow=T)
          fisher.test(tbl, alternative="two.sided")$conf.int[2]
              })

HCM <- data.frame(HCM)
colnames(HCM)[(ncol(HCM)-3):(ncol(HCM))] <- c("OR", "pvalOR", "lciOR", "uciOR")
HCM$pvalOR_BF <- p.adjust(HCM$pvalOR, method="bonferroni")
HCM$pvalOR_FDR <- p.adjust(HCM$pvalOR, method="fdr")
HCM$sigBF <- (HCM$pvalOR_BF <0.05)
HCM$sigFDR <- (HCM$pvalOR_FDR <0.05)
table(HCM$sigBF, HCM$sigFDR)
#        FALSE TRUE
#  FALSE    82   86 #FDR
#   TRUE      0   89 #BF

###DCM
DCM$Var <- DCM$AC_ALL_DCM
DCM$NoVar <- DCM$AN_ALL_DCM - DCM$AC_ALL_DCM
DCM$Var2p <- DCM$gnukb_AC
DCM$NoVarp <- DCM$gnukb_AN - DCM$gnukb_AC

DCM[(ncol(DCM)+1)] <- apply(DCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(DCM)-3):(ncol(DCM))]), ncol=2, byrow=T)
          fisher.test(tbl, alternative="two.sided")$estimate
              })

DCM[(ncol(DCM)+1)] <- apply(DCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(DCM)-3-1):(ncol(DCM)-1)]), ncol=2, byrow=T)
          fisher.test(tbl, alternative="two.sided")$p.value
              })

DCM[(ncol(DCM)+1)] <- apply(DCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(DCM)-3-2):(ncol(DCM)-2)]), ncol=2, byrow=T)
            fisher.test(tbl, alternative="two.sided")$conf.int[1]
              })

DCM[(ncol(DCM)+1)] <- apply(DCM, 1, 
      function(x) {
          tbl <- matrix(as.numeric(x[(ncol(DCM)-3-3):(ncol(DCM)-3)]), ncol=2, byrow=T)
          fisher.test(tbl, alternative="two.sided")$conf.int[2]
              })

DCM <- data.frame(DCM)
colnames(DCM)[(ncol(DCM)-3):(ncol(DCM))] <- c("OR", "pvalOR", "lciOR", "uciOR")
DCM$pvalOR_BF <- p.adjust(DCM$pvalOR, method="bonferroni")
DCM$pvalOR_FDR <- p.adjust(DCM$pvalOR, method="fdr")
DCM$sigBF <- (DCM$pvalOR_BF <0.05)
DCM$sigFDR <- (DCM$pvalOR_FDR <0.05)
table(DCM$sigBF, DCM$sigFDR)
#        FALSE TRUE
#  FALSE    12   29 #FDR
#  TRUE      0   18 #BF
```

# plot HCM + DCM individual variants - Figure 4

```{r}
test <- hcm2
test$Curation[test$Curation == "Pathogenic"] <- "P"
test$Curation[test$Curation == "LB"] <- "B/LB"
test$Curation[test$Curation == "B"] <- "B/LB"
test$Curation <- factor(test$Curation, levels = c("B/LB", "VUS", "LP", "P"))

#All by Penetrance and correlation with classification - Manual
a <- ggplot(test, aes(x=AC_ALL_HCM, y=penetrance, col=Curation)) + 
   theme_minimal() +
  ylab("Penetrance") +   
  xlab("HCM AC")  +
    scale_x_log10(breaks = c(2,3,4,5,10,25,50,100,150)) +
  scale_color_manual(values=c("darkgrey", "lightgoldenrod4", "orange", "red"), labels=c("B/LB", "VUS", "LP", "P"), drop = FALSE)+
  scale_linetype_manual(values = c(2,1), guide="none") +
  theme(text = element_text(size=16)) +
  theme(legend.position="top", legend.box="vertical", legend.margin=margin()) + 
   geom_pointrange(aes(ymin = lci, ymax = uci), position=position_jitter(width=0.1))

test2 <- DCM2
test2$Curation[test2$Curation == "Benign"] <- "B"
test2$Curation[test2$Curation == "B"] <- "B/LB"
test2$Curation[test2$Curation == "LB"] <- "B/LB"
test2$Curation <- factor(test2$Curation, levels = c("B/LB", "VUS", "LP", "P"))

#All by Penetrance and correlation with classification - Manual
b <- ggplot(test2, aes(x=AC_ALL_DCM, y=penetrance, col=Curation)) + 
  xlim(1.5,5.5) + 
  theme_minimal() +
  ylab("Penetrance") +   
  xlab("DCM AC")  +
  scale_color_manual(values=c("darkgrey", "lightgoldenrod4", "orange", "red"), labels=c("B/LB", "VUS", "LP", "P"), drop = FALSE)+
    theme(text = element_text(size=16),
        panel.grid.major.x = element_blank()) +
    scale_linetype_manual(values = c(2,1), guide="none") +
  theme(legend.position="top", legend.box="vertical", legend.margin=margin()) + 
  geom_pointrange(aes(ymin = lci, ymax = uci), position=position_dodge2(width=0.8))

library(ggpubr)
ggarrange(a, b, ncol = 1, nrow = 2, widths=c(0.1,0.1), align='v', common.legend = TRUE, legend = "top")
```

# useful to clinic

```{r}
hmm <- HCM[which(HCM$uci <=0.1),]
dim(hmm)
#162 559
table(hmm$Curation)
#        LB         LP Pathogenic        VUS 
#        13         25          2        122 
eh <- hmm[which(hmm$Curation == "Pathogenic" | hmm$Curation == "LP"),]
 median(eh$penetrance)*100
# 2%
  sd(eh$penetrance)*100
#0.7837093
sum(hmm$AC_ALL_HCM)
#745
#745/10400 = 7% of cases; 745/4314 = 17% of geno positives.
162/257 *100
#63%

hmm <- DCM[which(DCM$uci <=0.1),]
dim(hmm)
#17
 table(hmm$Curation)
#Benign     LB     LP    VUS 
#     2      7      0      8 
 sum(hmm$AC_ALL_DCM)
#45
#45/2564 = 2% of cases; 45/833 = 5% of geno positives.
17/59 *100
#29%
```

# world map of MYBPC3 variant

```{r}
#https://sarahpenir.github.io/r/making-maps/
library(dplyr)
library(stringr)
library(ggplot2)
library(maps)

options(scipen = 999)

world <- map_data("world")
head(world)

worldplot <- ggplot() +
  geom_polygon(data = world, aes(x=long, y = lat, group = group)) + 
  coord_fixed(1.3)

worldplot

country <- fread("country.txt", header=T)
country <- data.frame(country)
colnames(country)[1] <- "region" 

done <- join(world, country, type="left", by="region")
table(done$fill)
#    1     2 
#20605  6743

plain <- theme(
  axis.text = element_blank(),
  axis.line = element_blank(),
  axis.ticks = element_blank(),
  panel.border = element_blank(),
  panel.grid = element_blank(),
  axis.title = element_blank(),
  panel.background = element_rect(fill = "white"),
  plot.title = element_text(hjust = 0.5)
)

hmm <- done[!done$region == "Antarctica",]

worldHDI <- ggplot(data = hmm, mapping = aes(x = long, y = lat, group = group)) + 
  coord_fixed(1.3) +
  geom_polygon(aes(fill = fill)) +
  scale_fill_distiller(palette ="RdBu", direction = -1) + # or direction=1
  ggtitle("MYBPC3 R502W in HCM cases") +
  plain +
  theme(legend.position = "none")

worldHDI
```

# Penetrance - aggregate analyses

# UKB aggregate analyses HCM set up

```{r}
#aim get counts for genes, consequences, 
#filtering
#- 8 genes only
#- same filtering thresholds
#all HCM-associated genes
#genes separately
#variant consequences - MYBPC3 
#Curation
#allele frequency

#HCM
vep <- data.frame(fread("vepresults_hcmnormukbR", header=T, sep="\t"))

vep2 <- vep %>%
filter(CANONICAL == "YES")

vep3 <- vep2[
#vep2$SYMBOL == "BAG3" |
#vep2$SYMBOL == "DSP"  |
#vep2$SYMBOL == "TTN" |
vep2$SYMBOL == "MYH7"  |
#vep2$SYMBOL == "TNNC1"  |
vep2$SYMBOL == "TNNT2"  |
vep2$SYMBOL == "TPM1"  |
#vep2$SYMBOL == "LMNA"  |
vep2$SYMBOL == "ACTC1"  |
#vep2$SYMBOL == "PLN"  |
#vep2$SYMBOL == "RBM20"|
vep2$SYMBOL == "MYBPC3"|
vep2$SYMBOL == "TNNI3" |
vep2$SYMBOL == "MYL2" |
vep2$SYMBOL == "MYL3" ,
#vep2$SYMBOL == "DES" |
#vep2$SYMBOL == "SCN5A" ,
]

test <- vep3[which(vep3$ClinVar_CLNSIG == "Likely_pathogenic" | vep3$ClinVar_CLNSIG == "Pathogenic/Likely_pathogenic" | vep3$ClinVar_CLNSIG == "Pathogenic"),]
table(test$Consequence)
test$ClinVar[test$Consequence == "intron_variant"]
#splice_polypyrimidine_tract_variant,intron_variant
#splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant
#splice_donor_region_variant,intron_variant

vep3$Consequence[(   vep3$ClinVar == "188544" |
                     vep3$ClinVar == "42807" |
                     vep3$ClinVar == "693982" |
                     vep3$ClinVar == "42556")] <- "splice_confirmed"

table(vep3$Consequence)

vepx32 <- vep3 %>% filter(Consequence == "frameshift_variant" | 
                           Consequence == "frameshift_variant,splice_region_variant" |
                           Consequence == "frameshift_variant,start_lost" |
                           Consequence == "frameshift_variant,stop_lost" |
                           Consequence == "inframe_deletion" |
                           Consequence == "inframe_deletion,splice_region_variant" |
                           Consequence == "inframe_insertion" |
                           Consequence == "missense_variant" |
                           Consequence == "missense_variant,splice_region_variant" |
                           Consequence == "protein_altering_variant" |
                           Consequence == "splice_acceptor_variant" |
                           Consequence == "splice_confirmed" |
                           Consequence == "splice_donor_5th_base_variant,intron_variant" |
                           Consequence == "splice_donor_region_variant,intron_variant" |
                           Consequence == "splice_donor_variant" |
                           Consequence == "splice_donor_variant,coding_sequence_variant" |
                           Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                           Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                           Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                           Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                           Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                           Consequence == "splice_region_variant,intron_variant" |
                           Consequence == "splice_region_variant,synonymous_variant" |
                           Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                           Consequence == "start_lost" |
                           Consequence == "start_lost,splice_region_variant" |
                           Consequence == "stop_gained" |
                           Consequence == "stop_gained,frameshift_variant" |
                           Consequence == "stop_gained,splice_region_variant" |
                           Consequence == "stop_lost")

colnames(hcmfrq)[1:2] <- c("CHR", "ID")
colnames(vepx32)[1] <- "ID"

frq <- left_join(vepx32, hcmfrq, by="ID")

# caseukb$Consequence[caseukb$Consequence == "protein_altering_variant"] <- "inframe_insertion"
 
caseukb <- frq
#Check for duplicated IDs before removal
df <- caseukb %>%
  filter(Consequence == "splice_donor_region_variant,intron_variant" |
           Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
           Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,synonymous_variant") %>% 
  filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
dim(df)
#445 for removal
  
caseukb$found <- caseukb$ID %in% df$ID
table(caseukb$found)
#TRUE = 445 woohoo

caseukb2 <- caseukb[which(caseukb$found == "FALSE"),]
dim(caseukb2)
#2113
caseukb2$found <- NULL

doneant2 <- caseukb2

#Test LOFTEE flags for each gene
table(doneant2$LoF_filter)
#3UTR_SPLICE 5UTR_SPLICE 
table(doneant2$LoF_flags)
#NAGNAG_SITE PHYLOCSF_WEAK 

doneant2$Consequence <- as.character(doneant2$Consequence)

#remove LC LoFs and ID PAV that are mislabelled.
doneant2$Consequence[doneant2$LoF_flags == "NAGNAG_SITE"] <- "inframe_insertion"
doneant2$Consequence[doneant2$LoF_filter == "3UTR_SPLICE"] <- "splice_donor_variant_LC"
doneant2$Consequence[doneant2$LoF_filter == "5UTR_SPLICE"] <- "splice_donor_variant_LC"
#NAGNAG site = splice acceptor site rescued by inframe acceptor site
#3UTR SPLICE = splice donor with pLof low confidence
#5UTR SPLICE = splice donor with pLof low confidence

#add labels
df3 <- doneant2 %>% separate(HGVSc, c("HGVSc.a", "HGVSc.b"), sep=":")
df3$label <- paste0(df3$SYMBOL, ":", df3$HGVSc.b)
head(df3$label)
table(duplicated(df3$label))
table(is.na(df3$label))

HCM <- df3[which(df3$ukbAC_qced>0),]
dim(HCM)
#1975 

#check LC variants

table(HCM$Consequence)

HCM1 <- HCM[!(HCM$Consequence == "splice_donor_variant_LC"),]
dim(HCM1)
# 1974

table(HCM1$Consequence)
#go through all manual variants:
#start lost - below
#stop lost = inframe insertion
#inframe_deletion,splice_region_variant - below.

table(HCM1$NMD)
#NMDi:
#  - NMD_escaping_variant 
# 1936                   38 

PAV_List=c("frameshift_variant",
           "stop_gained",
           "splice_donor_variant",
           "splice_acceptor_variant",
           "splice_region_variant",
           "splice_polypyrimidine_tract_variant",
           "splice_donor_5th_base_variant",
           "splice_donor_region_variant",
           "splice_confirmed")

hmmx <- HCM1[grepl(paste(PAV_List, collapse="|"), HCM1$Consequence),]
dim(hmmx) 
#398

hmmx$POS <- hmmx$ID
hmmx2 <- hmmx %>% separate(POS, c("chr", "POS", "ref", "alt"), sep=":")
hmmx2$chr <- NULL
hmmx2$ref <- NULL
hmmx2$alt <- NULL
hmmx2$POS <- as.integer(hmmx2$POS)
hmmx <- hmmx2

head(hmmx$POS,10)
hmmx$hgvk <- hmmx$HGVSp
hmmx$hgvk <- gsub("\\?", 0, hmmx$hgvk)
hmmx <- hmmx %>% separate(hgvk, c("hgvka", "hgvkb", "fs"), sep="\\*")
hmmx$fs <- as.integer(as.character(hmmx$fs)) 
hmmx$fs <- (hmmx$fs*3)

for(i in (1:nrow(hmmx))){
if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") &
    (hmmx$SYMBOL[i] == "MYH7" | hmmx$SYMBOL[i] == "MYBPC3" | hmmx$SYMBOL[i] == "ACTC1" | 
       hmmx$SYMBOL[i] == "TNNT2" | hmmx$SYMBOL[i] == "TNNI3" | hmmx$SYMBOL[i] == "MYL2" | hmmx$SYMBOL[i] == "MYL3")) {
    hmmx$POS[i] <- hmmx$POS[i] - hmmx$fs[i] 
        }
    else if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") & 
            (hmmx$SYMBOL[i] == "TPM1")){
            hmmx$POS[i] <- hmmx$POS[i] + hmmx$fs[i] 
    }
   }

hmm <- hmmx[which(hmmx$SYMBOL == "MYH7"),]
#MYH7 <	23,413,759 + 55
hmm2 <- hmm[which(hmm$POS < 23413814),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 14:23412872:C:A splice_acceptor_variant 23412872          -        -                 PASS
#2 14:23413757:A:G    splice_donor_variant 23413757          -        -                 PASS

hmm <- hmmx[which(hmmx$SYMBOL == "MYBPC3"),]
#MYBPC3 <47,331,881 - second last exon is short
hmm2 <- hmm[which(hmm$POS < 47331881),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 11:47331879:G:A missense_variant,splice_region_variant 47331879 ENSP00000442795.1:p.P1273S        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "TNNT2"),]
#TNNT2 < 	201,359,663 - second last exon is short
hmm2 <- hmm[which(hmm$POS < 201359663),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 1:201359217:C:T                                                              stop_gained 201359217
#2 1:201359258:G:T splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant 201359258
#3 1:201359651:G:A                                                              stop_gained 201359651

hmm <- hmmx[which(hmmx$SYMBOL == "TNNI3"),]
#TNNI3 < 	55,154,030 + 55
hmm2 <- hmm[which(hmm$POS < 55154085),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#none

hmm <- hmmx[which(hmmx$SYMBOL == "TPM1"),]
#TPM1 > 63064142-55
hmm2 <- hmm[which(hmm$POS > 63064087),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 15:63064141:A:G                                   missense_variant,splice_region_variant 63064141 ENSP00000385107.4:p.I284V
#2 15:63064142:T:C                                   missense_variant,splice_region_variant 63064142 ENSP00000385107.4:p.I284T
#3 15:63064143:G:A                                                     splice_donor_variant 63064143                         -
#4 15:63065889:C:G splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant 63065889                         -
#5 15:63065891:C:T splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant 63065891                         -
#6 15:63065892:G:A splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant 63065892                         -

hmm <- hmmx[which(hmmx$SYMBOL == "MYL2"),]
#MYL2 < 110,913,144  - second last exon is short
hmm2 <- hmm[which(hmm$POS <  110913144),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
# 1 12:110911146:D:1                                                       frameshift_variant 110911137
# 2 12:110911176:C:G                                                  splice_acceptor_variant 110911176
# 3 12:110911176:C:T                                                  splice_acceptor_variant 110911176
# 4 12:110911181:A:G splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant 110911181
# 5 12:110913094:D:3                             splice_donor_variant,coding_sequence_variant 110913094
# 6 12:110913095:C:A                                                     splice_donor_variant 110913095
# 7 12:110913097:T:G                                   missense_variant,splice_region_variant 110913097
# 8 12:110913143:C:T                                   missense_variant,splice_region_variant 110913143

hmm <- hmmx[which(hmmx$SYMBOL == "MYL3"),]
#MYL3 <	46,858,272 - second last exon is short
hmm2 <- hmm[which(hmm$POS < 46858272),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 3:46858226:C:G splice_donor_5th_base_variant,intron_variant 46858226          -        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "ACTC1"),]
#ACTC1 < 	34,791,114 + 55
hmm2 <- hmm[which(hmm$POS < 34791169),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
HCM1$NMD[which(HCM1$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 15:34790493:D:2 frameshift_variant 34790466 ENSP00000290378.4:p.L351*fs*9        -                 FAIL

HCM1$NMD[HCM1$Consequence == "missense_variant,splice_region_variant" & (HCM1$SpliceAI_cutoff == "FAIL" | HCM1$SpliceAI_cutoff == "-")] <- "-"

# *ACTC1 - nTV
# *MYBPC3 - PAV
# *MYH7 - nTV
# *MYL2 - nTV
# *MYL3 - nTV
# *TNNI3 - nTV
# *TNNT2 - nTV
# *TPM1 - nTV

#MYBPC3 (usually NMD escaping only for LoF mechanisms but MYBPC3 = PAV)
pav <- HCM1 %>%
  filter(SYMBOL == "MYBPC3") 
dim(pav)
#716
  
miss_nmd <- HCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "TNNT2" |
        SYMBOL == "TNNI3" |
        SYMBOL == "TPM1" |
        SYMBOL == "MYL2" |
        SYMBOL == "MYL3" |
        SYMBOL == "ACTC1") %>%
  filter(NMD == "NMD_escaping_variant")
dim(miss_nmd)
#  45

miss <- HCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "TNNT2" |
        SYMBOL == "TNNI3" |
        SYMBOL == "TPM1" |
        SYMBOL == "MYL2" |
        SYMBOL == "MYL3" |
        SYMBOL == "ACTC1") %>%
  filter(Consequence == "inframe_deletion" |
         Consequence == "inframe_deletion,splice_region_variant" |
         Consequence == "inframe_insertion" |
         Consequence == "missense_variant" |
         Consequence == "missense_variant,splice_region_variant" |
         Consequence == "start_lost" |
         Consequence == "start_lost,splice_region_variant" |
         Consequence == "stop_lost")
dim(miss)
#1009

HCM1$filter <- ifelse((HCM1$ID %in% pav$ID | HCM1$ID %in% miss_nmd$ID | HCM1$ID %in% miss$ID), "TRUE", "FALSE")
table(HCM1$filter)
#FALSE  TRUE 
#     204  1770 

HCMfilt <- HCM1[which(HCM1$filter == "TRUE"),]
dim(HCMfilt)
#1770

#min = GDX
#0.00036598 gnoMAD 0.0007344 UKB n=410

full5 <- HCMfilt[which(HCMfilt$gnomADg_AF <= 0.00036598 | is.na(HCMfilt$gnomADg_AF)),]
dim(full5)
#1736

full5a <- full5[which(full5$ukbMAF_qced <= 0.0007344 | is.na(full5$ukbMAF_qced)),]
dim(full5a)
#1735

full5a$gnomADg_AC0 <- as.character(full5a$gnomADg_AC0)
full5a$gnomADg_AC0[full5a$gnomADg_AC == 0] <- "AC0" #call those flagged as bad by gnomAD
full5a$gnomADg_AN <- as.character(full5a$gnomADg_AN) 

which( colnames(full5a)=="gnomADg" )
which( colnames(full5a)=="gnomADg_AN_nfe_female" )

for (i in 1:nrow(full5a)) {
 if (full5a$gnomADg_AC0[i] == "AC0") {
   full5a[i,50:96] <- NA #all gnomAD data - call it NA
 }
}
```

# UKB all aggregate analyses DCM set up

```{r}
#aim get counts for genes, consequences, 
#filtering
#- dcm genes only
#- same filtering thresholds
#all DCM-associated genes
#genes separately
#variant consequences - TTN 
#Curation
#allele frequency

#DCM
vep <- data.frame(fread("vepresults_dcmnormukbR", header=T, sep="\t"))

vep2 <- vep %>%
filter(CANONICAL == "YES")

vep3 <- vep2[
vep2$SYMBOL == "BAG3" |
vep2$SYMBOL == "DSP"  |
vep2$SYMBOL == "TTN" |
vep2$SYMBOL == "MYH7"  |
vep2$SYMBOL == "TNNC1"  |
vep2$SYMBOL == "TNNT2"  |
#vep2$SYMBOL == "TPM1"  |
vep2$SYMBOL == "LMNA"  |
#vep2$SYMBOL == "ACTC1"  |
vep2$SYMBOL == "PLN"  |
vep2$SYMBOL == "RBM20"|
#vep2$SYMBOL == "MYBPC3"|
#vep2$SYMBOL == "TNNI3" |
#vep2$SYMBOL == "MYL2" |
#vep2$SYMBOL == "MYL3" ,
vep2$SYMBOL == "DES" |
vep2$SYMBOL == "SCN5A" ,
]

vep <- data.frame(fread("vepresults_ttnnormukbR", header=T, sep="\t")) #kept TTN separately to reduce compute time

vep2 <- vep %>%
filter(CANONICAL == "YES")

vep4 <- vep2[
vep2$SYMBOL == "TTN",
]

done <- rbind(vep3,vep4)
vep3 <- done

test <- vep3[which(vep3$ClinVar_CLNSIG == "Likely_pathogenic" | vep3$ClinVar_CLNSIG == "Pathogenic/Likely_pathogenic" | vep3$ClinVar_CLNSIG == "Pathogenic"),]
table(test$Consequence)
test$ClinVar[test$Consequence == "intron_variant"]

#splice_polypyrimidine_tract_variant,intron_variant
test$ClinVar[test$Consequence == "splice_polypyrimidine_tract_variant,intron_variant"]
#splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant
#splice_donor_region_variant,intron_variant
#splice_region_variant,synonymous_variant
#vep3$Consequence[( vep3$ClinVar == "638303" | vep3$ClinVar == "987382")] <- "splice_confirmed" ###not enough evidence

vepx32 <- vep3 %>% filter(Consequence == "frameshift_variant" | 
                           Consequence == "frameshift_variant,splice_region_variant" |
                           Consequence == "frameshift_variant,start_lost" |
                           Consequence == "frameshift_variant,stop_lost" |
                           Consequence == "inframe_deletion" |
                           Consequence == "inframe_deletion,splice_region_variant" |
                           Consequence == "inframe_insertion" |
                           Consequence == "missense_variant" |
                           Consequence == "missense_variant,splice_region_variant" |
                           Consequence == "protein_altering_variant" |
                           Consequence == "splice_acceptor_variant" |
                           Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                           Consequence == "splice_confirmed" |
                           Consequence == "splice_donor_5th_base_variant,intron_variant" |
                           Consequence == "splice_donor_region_variant,intron_variant" |
                           Consequence == "splice_donor_variant" |
                           Consequence == "splice_donor_variant,coding_sequence_variant" |
                           Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                           Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                           Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                           Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                           Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                           Consequence == "splice_region_variant,5_prime_UTR_variant" |
                           Consequence == "splice_region_variant,intron_variant" |
                           Consequence == "splice_region_variant,synonymous_variant" |
                           Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                           Consequence == "start_lost" |
                           Consequence == "start_lost,splice_region_variant" |
                           Consequence == "stop_gained" |
                           Consequence == "stop_gained,frameshift_variant" |
                           Consequence == "stop_gained,splice_region_variant" |
                           Consequence == "stop_lost")

colnames(dcmfrq)[1:2] <- c("CHR", "ID")
colnames(vepx32)[1] <- "ID"

frq <- left_join(vepx32, dcmfrq, by="ID")
dim(frq)
#22643

table(frq$Consequence)

caseukb <- frq
caseukb$Consequence[caseukb$Consequence == "protein_altering_variant"] <- "inframe_insertion"

#Check for duplicated IDs before removal
df <- caseukb %>%
  filter(Consequence == "splice_donor_region_variant,intron_variant" |
           Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
           Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,synonymous_variant" |
           Consequence == "splice_region_variant,5_prime_UTR_variant") %>% 
  filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
dim(df)
#2124 for removal
  
caseukb$found <- caseukb$ID %in% df$ID
table(caseukb$found)
caseukb2 <- caseukb[which(caseukb$found == "FALSE"),]
dim(caseukb2)
caseukb2$found <- NULL

#the 2 P/LP TTN missense variants aren't in the case data
caseukb4 <- caseukb2[!(caseukb2$SYMBOL == "TTN" & (caseukb2$Consequence == "missense_variant" | caseukb2$Consequence == "inframe_deletion" | caseukb2$Consequence == "inframe_insertion")),]
dim(caseukb4)
#5779

#19 "missense_variant,splice_region_variant" in TTN
hmm <- caseukb4 %>%
  filter(SYMBOL == "TTN") %>%
  filter(Consequence == "missense_variant,splice_region_variant" |
         Consequence == "inframe_deletion,splice_region_variant") %>%
    filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
dim(hmm)
#316 - all to go
caseukb4$found <- caseukb4$ID %in% hmm$ID
table(caseukb4$found)
doneant2 <- caseukb4[which(caseukb4$found == "FALSE"),]
dim(doneant2)
#5463
doneant2$found <- NULL

#Test LOFTEE flags for each gene
table(doneant2$LoF_filter)
#5UTR_SPLICE   END_TRUNC #END_TRUNC The LoF variant falls in the last filter_position of the transcript (default = 0.05).
doneant2$NMD[doneant2$LoF_filter == "END_TRUNC"]
doneant2$Consequence[doneant2$LoF_filter == "END_TRUNC"]
#frameshifts + NMD escaping...
doneant2$SYMBOL[doneant2$LoF_filter == "END_TRUNC"]
#SCN5A
#.... i think ignore for SCN5A which is PAV

table(doneant2$LoF_flags)
#NAGNAG_SITE NON_CAN_SPLICE #NON_CAN_SPLICE The LoF is a splice variant that falls in a non-canonical splice site (not GT..AG).

doneant2$Consequence <- as.character(doneant2$Consequence)

#remove LC LoFs and ID PAV that are mislabelled.
doneant2$Consequence[doneant2$LoF_flags == "NAGNAG_SITE"] <- "inframe_insertion"
doneant2$Consequence[doneant2$LoF_filter == "5UTR_SPLICE"] <- "splice_donor_variant_LC"
doneant2$Consequence[doneant2$LoF_flags == "NON_CAN_SPLICE"] <- "splice_donor_variant_LC"
#NAGNAG site = splice acceptor site rescued by inframe acceptor site
#3UTR SPLICE = splice donor with pLof low confidence
#5UTR SPLICE = splice donor with pLof low confidence

df3 <- doneant2 %>% separate(HGVSc, c("HGVSc.a", "HGVSc.b"), sep=":")
df3$label <- paste0(df3$SYMBOL, ":", df3$HGVSc.b)
head(df3$label)
table(duplicated(df3$label))
table(is.na(df3$label))

DCM <- df3[which(df3$ukbAC_qced>0),]
dim(DCM)
#5111 

#check LC variants

table(DCM$Consequence)

DCM1 <- DCM[!(DCM$Consequence == "splice_donor_variant_LC"),]
dim(DCM1)
# 5108

table(DCM1$Consequence)
#go through all manual variants:
#start lost - below
#stop lost = inframe insertion
#inframe_deletion,splice_region_variant - below.

table(DCM1$NMD)
#NMDi:
#  - NMD_escaping_variant 
# 1936                   38 

PAV_List=c("frameshift_variant",
           "stop_gained",
           "splice_donor_variant",
           "splice_acceptor_variant",
           "splice_region_variant",
           "splice_polypyrimidine_tract_variant",
           "splice_donor_5th_base_variant",
           "splice_donor_region_variant",
           "splice_confirmed")

hmmx <- DCM1[grepl(paste(PAV_List, collapse="|"), DCM1$Consequence),]
dim(hmmx) 
#1200

hmmx$POS <- hmmx$ID
hmmx2 <- hmmx %>% separate(POS, c("chr", "POS", "ref", "alt"), sep=":")
hmmx2$chr <- NULL
hmmx2$ref <- NULL
hmmx2$alt <- NULL
hmmx2$POS <- as.integer(hmmx2$POS)
hmmx <- hmmx2

#frameshift use HGVSp and ID frameshifts upstream to this that could have a PTC at 55 bp into penultimate exon ***
hmmx$hgvk <- hmmx$HGVSp
hmmx$hgvk <- gsub("\\?", 0, hmmx$hgvk)
hmmx <- hmmx %>% separate(hgvk, c("hgvka", "hgvkb", "fs"), sep="\\*")
hmmx$fs <- as.integer(as.character(hmmx$fs)) 
hmmx$fs <- (hmmx$fs*3)

hmmx$POS <- as.integer(as.character(hmmx$POS))
hmmx$pos <- hmmx$POS

for(i in (1:nrow(hmmx))){
   if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") & (is.na(hmmx$fs[i]))) {
    hmmx$POS[i] <- hmmx$pos[i]
  }
   else if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") &
    (hmmx$SYMBOL[i] == "MYH7" | hmmx$SYMBOL[i] == "SCN5A" | hmmx$SYMBOL[i] == "TNNC1" | 
       hmmx$SYMBOL[i] == "TNNT2" | hmmx$SYMBOL[i] == "TTN")) {
    hmmx$POS[i] <- hmmx$POS[i] - hmmx$fs[i] 
        }
    else if((hmmx$Consequence[i] == "frameshift_variant" | hmmx$Consequence[i] == "frameshift_variant,splice_region_variant") & 
            (hmmx$SYMBOL[i] == "BAG3" | hmmx$SYMBOL[i] == "DES" | hmmx$SYMBOL[i] == "DSP"
            | hmmx$SYMBOL[i] == "FLNC" | hmmx$SYMBOL[i] == "LMNA" | hmmx$SYMBOL[i] == "PLN"
            | hmmx$SYMBOL[i] == "RBM20")){
            hmmx$POS[i] <- hmmx$POS[i] + hmmx$fs[i] 
    }
 
}

tail(data.frame(hmmx$Consequence, hmmx$POS),10)

hmm <- hmmx[which(hmmx$SYMBOL == "BAG3"),]
hmm2 <- hmm[which(hmm$POS > (119672656 - 55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
# 1 10:119672661:G:A splice_donor_5th_base_variant,intron_variant 119672661                         -                    -                 FAIL
# 2 10:119676479:C:T                                  stop_gained 119676479 ENSP00000358081.4:p.R309* NMD_escaping_variant                 FAIL
# 3 10:119676935:G:T                                  stop_gained 119676935 ENSP00000358081.4:p.E461* NMD_escaping_variant                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "DES"),]
hmm2 <- hmm[which(hmm$POS >(219425745-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 2:219425745:G:C missense_variant,splice_region_variant 219425745 ENSP00000363071.3:p.E457D        -                 FAIL
#2 2:219425746:G:A                   splice_donor_variant 219425746                         -        -                 PASS

hmm <- hmmx[which(hmmx$SYMBOL == "DSP"),]
hmm2 <- hmm[which(hmm$POS >(7581569-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
# 1   6:7581510:D:4 frameshift_variant  7581522     ENSP00000369129.3:p.ER1776*fs*4                    -                 FAIL
# 2   6:7582690:C:T        stop_gained  7582690          ENSP00000369129.3:p.Q1810* NMD_escaping_variant                 FAIL
# 3   6:7582693:G:T        stop_gained  7582693          ENSP00000369129.3:p.E1811* NMD_escaping_variant                 FAIL
# 4   6:7582927:G:T        stop_gained  7582927          ENSP00000369129.3:p.E1889* NMD_escaping_variant                 FAIL
# 5   6:7583139:G:A        stop_gained  7583139          ENSP00000369129.3:p.W1959* NMD_escaping_variant                 FAIL
# 6   6:7583393:D:2 frameshift_variant  7583438     ENSP00000369129.3:p.T2045*fs*15                    -                 FAIL
# 7   6:7583604:D:4 frameshift_variant  7583604    ENSP00000369129.3:p.*F2115*fs*19                    -                 FAIL
# 8   6:7583740:C:T        stop_gained  7583740          ENSP00000369129.3:p.R2160* NMD_escaping_variant                 FAIL
# 9   6:7583758:C:T        stop_gained  7583758          ENSP00000369129.3:p.R2166* NMD_escaping_variant                 FAIL
# 10  6:7583775:D:2 frameshift_variant  7583922     ENSP00000369129.3:p.F2172*fs*49                    -                 FAIL
# 11  6:7583872:C:T        stop_gained  7583872          ENSP00000369129.3:p.Q2204* NMD_escaping_variant                 FAIL
# 12  6:7583872:D:5 frameshift_variant  7583914    ENSP00000369129.3:p.RS2206*fs*14                    -                    -
# 13  6:7583982:D:2 frameshift_variant  7583991      ENSP00000369129.3:p.I2241*fs*3                    -                 FAIL
# 14  6:7584112:C:T        stop_gained  7584112          ENSP00000369129.3:p.R2284* NMD_escaping_variant                 FAIL
# 15  6:7584316:C:T        stop_gained  7584316          ENSP00000369129.3:p.Q2352* NMD_escaping_variant                 FAIL
# 16  6:7584338:D:1 frameshift_variant  7584368     ENSP00000369129.3:p.I2359*fs*10                    -                 FAIL
# 17 6:7584406:D:17 frameshift_variant  7584412 ENSP00000369129.3:p.ESHRLP2382*fs*2                    -                    -
# 18  6:7584825:D:4 frameshift_variant  7584939    ENSP00000369129.3:p.KK2522*fs*38                    -                 FAIL
# 19  6:7584845:I:1 frameshift_variant  7584848      ENSP00000369129.3:p.-2529*fs*1                    -                 FAIL
# 20  6:7584846:T:A        stop_gained  7584846          ENSP00000369129.3:p.Y2528* NMD_escaping_variant                 FAIL
# 21  6:7584903:C:G        stop_gained  7584903          ENSP00000369129.3:p.Y2547* NMD_escaping_variant                 FAIL
# 22  6:7584904:C:T        stop_gained  7584904          ENSP00000369129.3:p.R2548* NMD_escaping_variant                 FAIL
# 23  6:7585018:C:T        stop_gained  7585018          ENSP00000369129.3:p.R2586* NMD_escaping_variant                 FAIL
# 24  6:7585028:D:4 frameshift_variant  7585061    ENSP00000369129.3:p.SK2591*fs*11                    -                 FAIL
# 25  6:7585207:I:1 frameshift_variant  7585303     ENSP00000369129.3:p.-2650*fs*32                    -                 FAIL
# 26  6:7585228:D:5 frameshift_variant  7585300  ENSP00000369129.3:p.CTG2656W*fs*24                    -                    -
# 27  6:7585247:D:1 frameshift_variant  7585298     ENSP00000369129.3:p.P2663*fs*17                    -                 FAIL
# 28  6:7585328:D:1 frameshift_variant  7585352      ENSP00000369129.3:p.K2689*fs*8                    -                 FAIL
# 29  6:7585336:C:T        stop_gained  7585336          ENSP00000369129.3:p.Q2692* NMD_escaping_variant                 FAIL
# 30  6:7585336:D:4 frameshift_variant  7585348     ENSP00000369129.3:p.QR2692*fs*4                    -                 FAIL
# 31  6:7585448:D:1 frameshift_variant  7585496     ENSP00000369129.3:p.Q2730*fs*16                    -                 FAIL
# 32  6:7585713:C:A        stop_gained  7585713          ENSP00000369129.3:p.Y2817* NMD_escaping_variant                 FAIL
# 33  6:7585776:I:2 frameshift_variant  7585911     ENSP00000369129.3:p.-2839*fs*45                    -                    -
# 34  6:7585777:D:2 frameshift_variant  7585813     ENSP00000369129.3:p.S2839*fs*12                    -                 FAIL

#hmm <- hmmx[which(hmmx$SYMBOL == "FLNC"),]
#hmm2 <- hmm[which(hmm$POS > (128858217-55)),]
#data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$ClinVar)
#doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"

hmm <- hmmx[which(hmmx$SYMBOL == "LMNA"),]
hmm2 <- hmm[which(hmm$POS > (156138757-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 1:156138562:I:1 frameshift_variant 156138895 ENSP00000357283.4:p.-593*fs*111                    -                 FAIL
#2 1:156138749:C:T        stop_gained 156138749       ENSP00000357283.4:p.R654* NMD_escaping_variant                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "MYH7"),]
hmm2 <- hmm[which(hmm$POS < (23413759+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 14:23412872:C:A splice_acceptor_variant 23412872          -        -                 PASS
#2 14:23413757:A:G    splice_donor_variant 23413757          -        -                 PASS

hmm <- hmmx[which(hmmx$SYMBOL == "PLN"),]
hmm2 <- hmm[which(hmm$POS >(118548392-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 6:118558982:I:2 frameshift_variant 118559039 ENSP00000350132.5:p.Q22L*fs*19        -                    -
#2 6:118558997:D:1 frameshift_variant 118559036  ENSP00000350132.5:p.K27*fs*13        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "RBM20"),]
hmm2 <- hmm[which(hmm$POS >(110831182-55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1  10:110831180:C:A missense_variant,splice_region_variant 110831180             ENSP00000358532.3:p.Q1191K        -                 FAIL
#2  10:110831180:C:G missense_variant,splice_region_variant 110831180             ENSP00000358532.3:p.Q1191E        -                 FAIL
#3 10:110835876:I:19                     frameshift_variant 110835942 ENSP00000358532.3:p.S1195WNICGIF*fs*22        -                    -

hmm <- hmmx[which(hmmx$SYMBOL == "SCN5A"),]
hmm2 <- hmm[which(hmm$POS <(38554279+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
# 1  3:38550354:I:1 frameshift_variant 38550330 ENSP00000398266.2:p.P2005P*fs*8                    -                 FAIL
# 2  3:38550500:G:A        stop_gained 38550500      ENSP00000398266.2:p.R1957* NMD_escaping_variant                 FAIL
# 3  3:38550542:G:A        stop_gained 38550542      ENSP00000398266.2:p.R1943* NMD_escaping_variant                 FAIL
# 4  3:38550545:D:2 frameshift_variant 38550509 ENSP00000398266.2:p.P1941*fs*12                    -                 FAIL
# 5  3:38550546:D:1 frameshift_variant 38550516 ENSP00000398266.2:p.P1941*fs*10                    -                 FAIL
# 6  3:38550599:I:1 frameshift_variant 38550542 ENSP00000398266.2:p.-1924*fs*19                    -                 FAIL
# 7  3:38550606:D:2 frameshift_variant 38550543 ENSP00000398266.2:p.K1921*fs*21                    -                 FAIL
# 8  3:38550647:G:A        stop_gained 38550647      ENSP00000398266.2:p.Q1908* NMD_escaping_variant                 FAIL
# 9  3:38550786:D:1 frameshift_variant 38550756 ENSP00000398266.2:p.L1861*fs*10                    -                 FAIL
# 10 3:38551014:D:2 frameshift_variant 38551008  ENSP00000398266.2:p.L1785*fs*2                    -                 FAIL
# 11 3:38551487:G:A        stop_gained 38551487      ENSP00000398266.2:p.R1628* NMD_escaping_variant                 FAIL
# 12 3:38551505:G:A        stop_gained 38551505      ENSP00000398266.2:p.R1622* NMD_escaping_variant                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "TNNC1"),]
hmm2 <- hmm[which(hmm$POS < (52451391+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
# 1 3:52451441:I:1 frameshift_variant 52451390 ENSP00000232975.3:p.E135G*fs*17        -                 FAIL
# 2 3:52451480:I:2 frameshift_variant 52451429 ENSP00000232975.3:p.Q122L*fs*17        -                    -
# 3 3:52451481:D:2 frameshift_variant 52451391  ENSP00000232975.3:p.L121*fs*30        -                 FAIL

hmm <- hmmx[which(hmmx$SYMBOL == "TNNT2"),]
hmm2 <- hmm[which(hmm$POS < (201359623+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
# 1 1:201359217:C:T                            stop_gained 201359217 ENSP00000499593.1:p.W297* NMD_escaping_variant                 FAIL
# 2 1:201359651:G:A                            stop_gained 201359651 ENSP00000499593.1:p.R275* NMD_escaping_variant                 FAIL
# 3 1:201359663:T:A missense_variant,splice_region_variant 201359663 ENSP00000499593.1:p.I271F                    -                 FAIL
# 4 1:201359665:T:G                splice_acceptor_variant 201359665                         -                    -                 PASS

hmm <- hmmx[which(hmmx$SYMBOL == "TTN"),]
hmm2 <- hmm[which(hmm$POS < (178527446+55)),]
data.frame(hmm2$ID, hmm2$Consequence, hmm2$POS, hmm2$HGVSp, hmm2$NMD, hmm2$SpliceAI_cutoff)
doneant2$NMD[which(doneant2$ID %in% hmm2$ID)] <- "NMD_escaping_variant"
#1 2:178527062:C:A        stop_gained 178527062      ENSP00000467141.1:p.E35976* NMD_escaping_variant                 FAIL
#2 2:178527120:D:1 frameshift_variant 178527072 ENSP00000467141.1:p.L35956*fs*16                    -                 FAIL
#3 2:178527188:C:A        stop_gained 178527188      ENSP00000467141.1:p.G35934* NMD_escaping_variant                 FAIL
#4 2:178527481:D:1 frameshift_variant 178527448 ENSP00000467141.1:p.S35882*fs*11                    -                 FAIL
#5 2:178527491:G:A        stop_gained 178527491      ENSP00000467141.1:p.Q35879*                    -                 FAIL

doneant2$NMD[doneant2$Consequence == "missense_variant,splice_region_variant" & (doneant2$SpliceAI_cutoff == "FAIL" | doneant2$SpliceAI_cutoff == "-")] <- "-"

DCM1 <- doneant2

#TTN PSI 90 regions
psi <- read.table("TTNpsi90.txt", header=F)
DCM1$ttnfound <- ifelse(sapply(DCM1$POS, function(p) any (psi$V1 > p & psi$V2 < p)), "YES", "NO")
table(DCM1$ttnfound)
# NO  YES 
#4918  545
dcm <- DCM1[!(DCM1$ttnfound == "NO" & DCM1$SYMBOL == "TTN"),]
dim(dcm)
#5076
dcm$ttnfound <- NULL
dim(dcm)
#5076

DCM1 <- dcm

# dcma$SYMBOL == "BAG3" | #PAV
# dcma$SYMBOL == "DES" | #nTV
# dcma$SYMBOL == "TTN" | #TV
# dcma$SYMBOL == "MYH7"  | #nTV
# dcma$SYMBOL == "TNNC1"  | #nTV (missense only)
# dcma$SYMBOL == "TNNT2"  | #nTV (missense only)
# dcma$SYMBOL == "LMNA"  | #PAV
# dcma$SYMBOL == "FLNC"  | #TV
# dcma$SYMBOL == "PLN"  | #PAV
# dcma$SYMBOL == "SCN5A"  | #PAV
# dcma$SYMBOL == "RBM20"| #PAV
# dcma$SYMBOL == "DSP", #PAV

table(DCM1$Consequence)

#PAV
pav <- DCM1 %>%
  filter(SYMBOL == "BAG3"|
         SYMBOL == "LMNA"|
         SYMBOL == "PLN"|
         SYMBOL == "SCN5A"|
         SYMBOL == "RBM20"|
         SYMBOL == "DSP")
dim(pav)
#3382
  
miss_nmd <- DCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "DES" |
        SYMBOL == "TNNC1" |
        SYMBOL == "TNNT2") %>%
  filter(NMD == "NMD_escaping_variant")
dim(miss_nmd)
# 33

miss <- DCM1 %>%
  filter(SYMBOL == "MYH7"|
         SYMBOL == "DES" |
        SYMBOL == "TNNC1" |
        SYMBOL == "TNNT2") %>%
  filter(Consequence == "inframe_deletion" |
         Consequence == "inframe_insertion" |
         Consequence == "missense_variant" |
         Consequence == "missense_variant,splice_region_variant"|
           Consequence == "start_lost"|
           Consequence == "stop_lost")
dim(miss)
#1029

ttn <- DCM1 %>%
  filter(SYMBOL == "TTN") %>%
  filter(Consequence == "frameshift_variant" |
         Consequence == "frameshift_variant,splice_region_variant" |
         Consequence == "missense_variant,splice_region_variant" | #curated for splicing [there are none]
         Consequence == "splice_acceptor_variant" |
           Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
         Consequence == "splice_confirmed" |
           Consequence == "splice_donor_region_variant,intron_variant" |
         Consequence == "splice_donor_5th_base_variant,intron_variant" |
         Consequence == "splice_donor_variant" |
         Consequence == "splice_donor_variant,coding_sequence_variant" |
         Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
           Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
         Consequence == "splice_polypyrimidine_tract_variant,intron_variant" | 
           Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,intron_variant" |
           Consequence == "splice_region_variant,synonymous_variant" |
         Consequence == "stop_gained" |
         Consequence == "stop_gained,frameshift_variant" |
         Consequence == "stop_gained,splice_region_variant")
dim(ttn)
#544

DCM1$filter <- ifelse((DCM1$ID %in% pav$ID | DCM1$ID %in% miss_nmd$ID | DCM1$ID %in% miss$ID | DCM1$ID %in% ttn$ID), "TRUE", "FALSE")
table(DCM1$filter)
#FALSE  TRUE 
# 88  4988 

DCMfilt <- DCM1[which(DCM1$filter == "TRUE"),]
dim(DCMfilt)
#4988

DCMfilt <- DCMfilt[!DCMfilt$Consequence == "splice_donor_variant_LC",]
dim(DCMfilt)
# 4985 

#min = OMG
#0.000552987 gnomAD 0.0006031 UKB n=111

full5 <- DCM[which(DCM$gnomADg_AF <= 0.000552987 | is.na(DCM$gnomADg_AF)),]
dim(full5)
#4906

full5a <- full5[which(full5$ukbMAF_qced <= 0.0006031 | is.na(full5$ukbMAF_qced)),]
dim(full5a)
#4898

full5a$gnomADg_AC0 <- as.character(full5a$gnomADg_AC0)
full5a$gnomADg_AC0[full5a$gnomADg_AC == 0] <- "AC0" #call those flagged as bad by gnomAD
full5a$gnomADg_AN <- as.character(full5a$gnomADg_AN) 

which( colnames(full5a)=="gnomADg" )
which( colnames(full5a)=="gnomADg_AN_nfe_female" )

for (i in 1:nrow(full5a)) {
 if (full5a$gnomADg_AC0[i] == "AC0") {
   full5a[i,49:95] <- NA #all gnomAD data - call it NA
 }
}
```

# HCM estimate penetrance (aggregate)

```{r}
HCMx <- read.table("HCMx.txt", header=T, sep="\t") #before penetrance filter
HCMx$REVEL <- as.numeric(as.character(HCMx$REVEL))
ukb <- read.table("ukb_hcm_match.txt", header=T, sep="\t")

#### All HCM vars
my <- HCMx
dim(my)
#1340
sum(my$AC_ALL_HCM)
#4314
max(my$AN_ALL_HCM)
#20800

dim(ukb)
#1735
sum(ukb$ukbAC_qced)
#9196
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=97, n_D=52660, x_AgD=4314, n_AgD=20800, x_A=9196, n_A=334956)
#0.01391561 (0.01138069-0.01701516)

#### All HCM female

my <- HCMx[which(HCMx$AN_HCM_female >0),]
dim(my)
#1340
sum(my$AC_HCM_female)
#527
max(my$AN_HCM_female)
#1468

#female
 hmm <- ukb[which(ukb$ukbAN_fem >0),]
 sum(hmm$ukbAC_fem)
#5039
max(hmm$ukbAN_fem)
#183502
dim(hmm)
#1735

penetrance(x_D=15, n_D=19646, x_AgD=527, n_AgD=1468, x_A=5039, n_A=183502)
#0.009986866 (0.00603896-0.01651568)

#### All HCM male
my <- HCMx[which(HCMx$AN_HCM_male >0),]
dim(my)
#1340
sum(my$AC_HCM_male)
#850
max(my$AN_HCM_male)
#2708

#male
hmm <- ukb[which(ukb$ukbAN_man >0),]
sum(hmm$ukbAC_man)
#4157
max(hmm$ukbAN_man)
#151454
dim(hmm)
#1735

penetrance(x_D=68, n_D=24411, x_AgD=850, n_AgD=2708, x_A=4157, n_A=151454)
#0.03185708 (0.02494006-0.04069251)

#### ALL HCM penetrance in people with sex
my <- HCMx[which(HCMx$AN_HCM_male >0 | HCMx$AN_HCM_female >0),]
dim(my)
#1340
sum(my$AC_HCM_male) + sum(my$AC_HCM_female) 
#1377
max(my$AN_HCM_male) + max(my$AN_HCM_female)
#4176

hmm <- ukb[which(ukb$ukbAN_man >0 | ukb$ukbAN_fem >0),]
sum(hmm$ukbAC_man) + sum(hmm$ukbAC_fem)
#9196
max(hmm$ukbAN_man) + max(hmm$ukbAN_fem) 
#334956
dim(hmm)
#1735

penetrance(x_D=97, n_D=52660, x_AgD=1377, n_AgD=4176, x_A=9196, n_A=334956)
#0.02212376 (0.01804172-0.02712938)

#### ALL EUR
my <- HCMx[which(HCMx$AN_HCM_EUR >0),]
dim(my)
#1340
sum(my$AC_HCM_EUR)
#3932
max(my$AN_HCM_EUR)
#19384

hmm <- ukb[which(ukb$ukbAN_eu >0),]
sum(hmm$ukbAC_eu)
#8068
max(hmm$ukbAN_eu)
#310054
dim(hmm)
#1735

penetrance(x_D=97, n_D=52660, x_AgD=3932, n_AgD=19384, x_A=8068, n_A=310054)
#0.01435946 (0.01173988-0.01756357)

####ALL AFR
my <- HCMx[which(HCMx$AN_HCM_AFR >0),]
dim(my)
#1340
sum(my$AC_HCM_AFR)
#296
max(my$AN_HCM_AFR)
#984

hmm <- ukb[which(ukb$ukbAN_afr >0),]
sum(hmm$ukbAC_afr)
#254
max(hmm$ukbAN_afr)
#5806
dim(hmm)
#1735

penetrance(x_D=97, n_D=52660, x_AgD=296, n_AgD=984, x_A=254, n_A=5806)
#0.01266591 (0.009858047 - 0.01627353)

####ALL EAS
my <- HCMx[which(HCMx$AN_HCM_EAS >0),]
dim(my)
#1340
sum(my$AC_HCM_EAS)
#37
max(my$AN_HCM_EAS)
#172

hmm <- ukb[which(ukb$ukbAN_eas >0),]
sum(hmm$ukbAC_eas)
#63
max(hmm$ukbAN_eas)
#1210
dim(hmm)
#1735

penetrance(x_D=97, n_D=52660, x_AgD=37, n_AgD=172, x_A=63, n_A=1210)
#0.007610941 (0.004998257-0.01158932)

####ALL SAS
my <- HCMx[which(HCMx$AN_HCM_SAS >0),]
dim(my)
#1340
sum(my$AC_HCM_SAS)
#38
max(my$AN_HCM_SAS)
#204

hmm <- ukb[which(ukb$ukbAN_sas >0),]
sum(hmm$ukbAC_sas)
#316
max(hmm$ukbAN_sas)
#6272
dim(hmm)
#1735

penetrance(x_D=97, n_D=52660, x_AgD=38, n_AgD=204, x_A=316, n_A=6272)
#0.006810894 (0.004737069-0.00979261)

### HCM in those with ancestry data
my <- HCMx[which(HCMx$AN_HCM_SAS >0 | HCMx$AN_HCM_EUR >0 | HCMx$AN_HCM_EAS > 0 | HCMx$AN_HCM_AFR > 0),]
dim(my)
#1340
sum(my$AC_HCM_SAS) + sum(my$AC_HCM_EAS) + sum(my$AC_HCM_EUR) + sum(my$AC_HCM_AFR)
#4303
max(my$AN_HCM_SAS) + max(my$AN_HCM_EAS) + max(my$AN_HCM_EUR) + max(my$AN_HCM_AFR)
#20744

hmm <- ukb[which(ukb$ukbAN_sas >0 | ukb$ukbAN_eas > 0 | ukb$ukbAN_afr >0 | ukb$ukbAN_eu >0),]
sum(hmm$ukbAC_sas) + sum(hmm$ukbAC_eas) + sum(hmm$ukbAC_afr) + sum(hmm$ukbAC_eu)
#8701
max(hmm$ukbAN_sas) + max(hmm$ukbAN_eas) + max(hmm$ukbAN_afr) + max(hmm$ukbAN_eu)
#323342
dim(hmm)
#1735

penetrance(x_D=97, n_D=52660, x_AgD=4303, n_AgD=20744, x_A=8701, n_A=323342)
#0.01419935 (0.01161201-0.01736318)

#### Age

#0s
penetrance(x_D=(97*(max(HCMx$AN_HCM_0s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_0s), n_AgD=max(HCMx$AN_HCM_0s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.0002726665 (8.317719e-05-0.0008938394)

#10s
penetrance(x_D=(97*(max(HCMx$AN_HCM_10s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_10s), n_AgD=max(HCMx$AN_HCM_10s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.002236989 (0.00119264-0.004195836)

#20s
penetrance(x_D=(97*(max(HCMx$AN_HCM_20s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_20s), n_AgD=max(HCMx$AN_HCM_20s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.004436637 (0.002754769-0.007145336)

#30s
penetrance(x_D=(97*(max(HCMx$AN_HCM_30s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_30s), n_AgD=max(HCMx$AN_HCM_30s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.008081258 (0.005601847-0.01165807)

#40s
penetrance(x_D=(97*(max(HCMx$AN_HCM_40s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_40s), n_AgD=max(HCMx$AN_HCM_40s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.01310356 (0.00983553-0.01745746)

#50s
penetrance(x_D=(97*(max(HCMx$AN_HCM_50s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_50s), n_AgD=max(HCMx$AN_HCM_50s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.01705098 (0.01337991-0.02172929)

#60s
penetrance(x_D=(97*(max(HCMx$AN_HCM_60s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_60s), n_AgD=max(HCMx$AN_HCM_60s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.02064575 (0.01663709-0.0256203)

#70s
penetrance(x_D=(97*(max(HCMx$AN_HCM_70s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_70s), n_AgD=max(HCMx$AN_HCM_70s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.02222478 (0.01808904-0.02730608)

#80s
penetrance(x_D=(97*(max(HCMx$AN_HCM_80s)/max(HCMx$AN_HCM_80s))), n_D=(52660), x_AgD=sum(HCMx$AC_HCM_80s), n_AgD=max(HCMx$AN_HCM_80s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.02252714 (0.0183692-0.02762626)

one1 <- c("by 10", 0.0002726665, 8.317719e-05, 0.0008938394, max(HCMx$AN_HCM_0s)/max(HCMx$AN_HCM_80s))
one2 <- c("by 20", 0.002236989, 0.00119264, 0.004195836, max(HCMx$AN_HCM_10s)/max(HCMx$AN_HCM_80s))
one3 <- c("by 30", 0.004436637, 0.002754769, 0.007145336, max(HCMx$AN_HCM_20s)/max(HCMx$AN_HCM_80s))
one <- c("by 40", 0.008081258, 0.005601847, 0.01165807, max(HCMx$AN_HCM_30s)/max(HCMx$AN_HCM_80s))
two <- c("by 50", 0.01310356, 0.00983553, 0.01745746, max(HCMx$AN_HCM_40s)/max(HCMx$AN_HCM_80s))
thre <- c("by 60", 0.01705098, 0.01337991, 0.02172929, max(HCMx$AN_HCM_50s)/max(HCMx$AN_HCM_80s))
four <- c("by 70", 0.02064575, 0.01663709, 0.0256203, max(HCMx$AN_HCM_60s)/max(HCMx$AN_HCM_80s))
five <- c("by 80", 0.02222478, 0.01808904, 0.02730608, max(HCMx$AN_HCM_70s)/max(HCMx$AN_HCM_80s))
five1 <- c("by 100", 0.02252714, 0.0183692, 0.02762626, max(HCMx$AN_HCM_80s)/max(HCMx$AN_HCM_80s))

one1 <- c("by 10", 0.0002726665, 8.317719e-05, 0.0008938394)
one2 <- c("by 20", 0.002236989, 0.00119264, 0.004195836)
one3 <- c("by 30", 0.004436637, 0.002754769, 0.007145336)
one <- c("by 40", 0.008081258, 0.005601847, 0.01165807)
two <- c("by 50", 0.01310356, 0.00983553, 0.01745746)
thre <- c("by 60", 0.01705098, 0.01337991, 0.02172929)
four <- c("by 70", 0.02064575, 0.01663709, 0.0256203)
five <- c("by 80", 0.02222478, 0.01808904, 0.02730608)
five1 <- c("by 100", 0.02252714, 0.0183692, 0.02762626)

okay <- rbind(one1,one2,one3,one,two,thre,four,five,five1)
#colnames(okay) <- c("Curation", "Penetrance", "LCI", "UCI", "prop")
colnames(okay) <- c("Curation", "Penetrance", "LCI", "UCI")
okay <- data.frame(okay)
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
#okay$prop <- (as.numeric(okay$prop)*0.02252714)
okay$order <- seq(1,nrow(okay))

#library(forcats)
c <- okay %>%
  mutate(ordering = rev(order),
         Curation = fct_reorder(Curation, ordering, .desc = T)) %>%
ggplot( aes(x=Curation, y=Penetrance)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.4)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.4), size=5) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none") +
  ylab("Penetrance") +
  xlab("Variant") +
  ylim(0,0.03)+
  ggtitle("HCM")# +

### HCM in age cohort

my <- HCMx[which(HCMx$AN_HCM_0s >0 | HCMx$AN_HCM_10s >0 | HCMx$AN_HCM_20s > 0 | HCMx$AN_HCM_30s > 0 | HCMx$AN_HCM_40s >0 | HCMx$AN_HCM_50s >0 | HCMx$AN_HCM_60s >0 | HCMx$AN_HCM_70s >0 | HCMx$AN_HCM_80s>0),]
dim(my)
#1340
sum(my$AC_HCM_80s) #cumulative
#1341
max(my$AN_HCM_80s)
#3994

hmm <- ukb
sum(hmm$ukbAC_qced)
#9196
max(hmm$ukbAN_qced)
#334956
dim(hmm)
#1735

penetrance(x_D=97, n_D=52660, x_AgD=1341, n_AgD=3994, x_A=9196, n_A=334956)
#0.02252714 (0.0183692-0.02762626)

#######rarity
summary(HCMx$gnomADg_AC) # 1-92
none <- HCMx[is.na(HCMx$gnomADg_AC),]
one <- HCMx[which(HCMx$gnomADg_AC == 1),]
two <- HCMx[which(HCMx$gnomADg_AC == 2),]
three <- HCMx[which(HCMx$gnomADg_AC == 3),]
four <- HCMx[which(HCMx$gnomADg_AC == 4),]
five <- HCMx[which(HCMx$gnomADg_AC == 5),]
ftoe <- HCMx[which(HCMx$gnomADg_AC > 5 & HCMx$gnomADg_AC < 8),]
etot <- HCMx[which(HCMx$gnomADg_AC > 7 & HCMx$gnomADg_AC < 11),]
etotw <- HCMx[which(HCMx$gnomADg_AC > 10 & HCMx$gnomADg_AC < 21),]
twtoth <- HCMx[which(HCMx$gnomADg_AC > 20 & HCMx$gnomADg_AC < 141),]

ukb$gnomADg_AC <- as.integer(as.character(ukb$gnomADg_AC))
summary(as.integer(as.character(ukb$gnomADg_AC))) # 1-92
none2 <- ukb[is.na(ukb$gnomADg_AC),]
one2 <- ukb[which(ukb$gnomADg_AC == 1),]
two2 <- ukb[which(ukb$gnomADg_AC == 2),]
three2 <- ukb[which(ukb$gnomADg_AC == 3),]
four2 <- ukb[which(ukb$gnomADg_AC == 4),]
five2 <- ukb[which(ukb$gnomADg_AC == 5),]
ftoe2 <- ukb[which(ukb$gnomADg_AC > 5 & ukb$gnomADg_AC < 8),]
etot2 <- ukb[which(ukb$gnomADg_AC > 7 & ukb$gnomADg_AC < 11),]
etotw2 <- ukb[which(ukb$gnomADg_AC > 10 & ukb$gnomADg_AC < 21),]
twtoth2 <- ukb[which(ukb$gnomADg_AC > 20 & ukb$gnomADg_AC < 141),]

#AC=0 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(none$AC_ALL_HCM), n_AgD=max(none$AN_ALL_HCM), x_A=sum(none2$ukbAC_qced), n_A=max(none2$ukbAN_qced))
#0.03727329 (0.03025419-0.04592086)

#AC=1 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(one$AC_ALL_HCM), n_AgD=max(one$AN_ALL_HCM), x_A=sum(one2$ukbAC_qced), n_A=max(one2$ukbAN_qced))
#0.01831592 (0.01458007-0.02300902)

#AC=2 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(two$AC_ALL_HCM), n_AgD=max(two$AN_ALL_HCM), x_A=sum(two2$ukbAC_qced), n_A=max(two2$ukbAN_qced))
#0.01542251 (0.01200784-0.0198082)

#AC=3 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(three$AC_ALL_HCM), n_AgD=max(three$AN_ALL_HCM), x_A=sum(three2$ukbAC_qced), n_A=max(three2$ukbAN_qced))
#0.01960316 (0.01541343-0.02493174)

#AC=4 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(four$AC_ALL_HCM), n_AgD=max(four$AN_ALL_HCM), x_A=sum(four2$ukbAC_qced), n_A=max(four2$ukbAN_qced))
#0.02196391 (0.01706264-0.02827307)

#AC=5 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(five$AC_ALL_HCM), n_AgD=max(five$AN_ALL_HCM), x_A=sum(five2$ukbAC_qced), n_A=max(five2$ukbAN_qced))
#0.01005053 (0.007459804-0.01354101)

#AC=6-7 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(ftoe$AC_ALL_HCM), n_AgD=max(ftoe$AN_ALL_HCM), x_A=sum(ftoe2$ukbAC_qced), n_A=max(ftoe2$ukbAN_qced))
#0.007359102 (0.005662861-0.00956343)

#AC=8-10 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(etot$AC_ALL_HCM), n_AgD=max(etot$AN_ALL_HCM), x_A=sum(etot2$ukbAC_qced), n_A=max(etot2$ukbAN_qced))
#0.01082485 (0.008483144-0.01381297)

#AC=11-20 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(etotw$AC_ALL_HCM), n_AgD=max(etotw$AN_ALL_HCM), x_A=sum(etotw2$ukbAC_qced), n_A=max(etotw2$ukbAN_qced))
#0.004904673 (0.003862947-0.006227324)

#AC=21-140 in gnomAD
penetrance(x_D=97, n_D=52660, x_AgD=sum(twtoth$AC_ALL_HCM), n_AgD=max(twtoth$AN_ALL_HCM), x_A=sum(twtoth2$ukbAC_qced), n_A=max(twtoth2$ukbAN_qced))
#0.002959941 (0.002329424-0.003761122)

bag <- c("0", 0.03727329 ,0.03025419,0.04592086)
des <- c("1", 0.01831592 ,0.01458007,0.02300902)
dsp <- c("2", 0.01542251 ,0.01200784,0.0198082)
one <- c("3", 0.01960316 ,0.01541343,0.02493174)
two <- c("4", 0.02196391 ,0.01706264,0.02827307)
thre <- c("5", 0.01005053 ,0.007459804,0.01354101)
four <- c("6", 0.007359102 ,0.005662861,0.00956343)
five <- c("8", 0.01082485 ,0.008483144,0.01381297)
six <- c("11", 0.004904673 ,0.003862947,0.006227324)
sev <- c("21+", 0.002959941 ,0.002329424,0.003761122)

okay <- rbind(bag, des, dsp, one,two,thre,four,five,six, sev)
colnames(okay) <- c("gnomAD_AC", "Penetrance", "LCI", "UCI")
okay <- data.frame(okay)
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$order <- seq(nrow(okay),1)

#library(forcats)
b <- okay %>%
mutate(ordering = as.numeric(order),
        gnomAD_AC = fct_reorder(gnomAD_AC, ordering, .desc = T)) %>%
ggplot( aes(x=gnomAD_AC, y=Penetrance, group=gnomAD_AC)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.4)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.4), size=5) +
  theme_minimal() +
  theme(text = element_text(size=16), legend.position = "none") +
  ylab("Penetrance") +
  xlab("gnomAD AC") +
  ylim(0,0.05)+
  ggtitle("HCM")+
    scale_x_discrete(guide = guide_axis(n.dodge = 2)) 
```

# DCM estimate penetrance (aggregate)

```{r}
#DCM
DCMx <- read.table("DCMx.txt", header=T, sep="\t") #before penetrance filter
ukb <- read.table("ukb_dcm_match.txt", header=T, sep="\t")
DCMx$REVEL <- as.numeric(as.character(DCMx$REVEL))

#### All DCM vars
my <- DCMx
dim(my)
#665
sum(my$AC_ALL_DCM)
#833
max(my$AN_ALL_DCM)
#5128

dim(ukb)
#4898
sum(ukb$ukbAC_qced)
#22232
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=833, n_AgD=5128, x_A=22232, n_A=334956)
# 0.01110665 (0.009465513-0.01303234)

#### All DCM female
my <- DCMx[which(DCMx$AN_DCM_female >0),]
dim(my)
#665
sum(my$AC_DCM_female)
#149
max(my$AN_DCM_female)
#680

# #female
 hmm <- ukb[which(ukb$ukbAN_fem >0),]
  sum(hmm$ukbAC_fem)
# #12243
 max(hmm$ukbAN_fem)
# #183502
 dim(hmm)
# #4898

penetrance(x_D=60, n_D=20316, x_AgD=149, n_AgD=680, x_A=12243, n_A=183502)
#0.009699751 (0.007263621-0.01295293)

#### All DCM male
my <- DCMx[which(DCMx$AN_DCM_male >0),]
dim(my)
#665
sum(my$AC_DCM_male)
#299
max(my$AN_DCM_male)
#1452

#male
hmm <- ukb[which(ukb$ukbAN_man >0),]
sum(hmm$ukbAC_man)
#9989
max(hmm$ukbAN_man)
#151454
dim(hmm)
#4898

penetrance(x_D=117, n_D=18687, x_AgD=299, n_AgD=1452, x_A=9989, n_A=151454)
#0.01954853 (0.01588646-0.02405476)

#### ALL DCM penetrance in people with sex
my <- DCMx[which(DCMx$AN_DCM_male >0 | DCMx$AN_DCM_female >0),]
dim(my)
#665
sum(my$AC_DCM_male) + sum(my$AC_DCM_female) 
#448
max(my$AN_DCM_male) + max(my$AN_DCM_female)
#2132

hmm <- ukb[which(ukb$ukbAN_man >0 | ukb$ukbAN_fem >0),]
sum(hmm$ukbAC_man) + sum(hmm$ukbAC_fem)
#22232
max(hmm$ukbAN_man) + max(hmm$ukbAN_fem) 
#334956
dim(hmm)
#4898

penetrance(x_D=177, n_D=39003, x_AgD=448, n_AgD=2132, x_A=22232, n_A=334956)
#0.01436737 (0.01213669-0.01700804)

#### ALL EUR
my <- DCMx[which(DCMx$AN_DCM_EUR >0),]
dim(my)
#665
sum(my$AC_DCM_EUR)
#649
max(my$AN_DCM_EUR)
#4454

hmm <- ukb[which(ukb$ukbAN_eu >0),]
sum(hmm$ukbAC_eu)
#18375
max(hmm$ukbAN_eu)
#310054
dim(hmm)
#4898

penetrance(x_D=177, n_D=39003, x_AgD=649, n_AgD=4454, x_A=18375, n_A=310054)
#0.01115788 (0.009473157-0.01314221)

####ALL AFR
my <- DCMx[which(DCMx$AN_DCM_AFR >0),]
dim(my)
#665
sum(my$AC_DCM_AFR)
#78
max(my$AN_DCM_AFR)
#284

hmm <- ukb[which(ukb$ukbAN_afr >0),]
sum(hmm$ukbAC_afr)
#1193
max(hmm$ukbAN_afr)
#5806
dim(hmm)
#4898

penetrance(x_D=177, n_D=39003, x_AgD=78, n_AgD=284, x_A=1193, n_A=5806)
#0.006065938 (0.004752608 - 0.007742192)

### TTN AFR
my <- DCMx[which(DCMx$AN_DCM_AFR >0 & DCMx$SYMBOL == "TTN"),]
dim(my)
#191
sum(my$AC_DCM_AFR)
#14
max(my$AN_DCM_AFR)
#284

hmm <- ukb[which(ukb$ukbAN_afr >0 & ukb$SYMBOL == "TTN"),]
sum(hmm$ukbAC_afr)
#23
max(hmm$ukbAN_afr)
#5806
dim(hmm)
#542

penetrance(x_D=177, n_D=39003, x_AgD=14, n_AgD=284, x_A=23, n_A=5806)
#0.05649367 (0.02919498 - 0.1093179)

### TTN EUR

my <- DCMx[which(DCMx$AN_DCM_EUR >0 & DCMx$SYMBOL == "TTN"),]
dim(my)
#191
sum(my$AC_DCM_EUR)
#179
max(my$AN_DCM_EUR)
#4454

hmm <- ukb[which(ukb$ukbAN_eu >0 & ukb$SYMBOL == "TTN"),]
sum(hmm$ukbAC_eu)
#875
max(hmm$ukbAN_eu)
#310054
dim(hmm)
#542

penetrance(x_D=177, n_D=39003, x_AgD=179, n_AgD=4454, x_A=875, n_A=310054)
#0.06462651 (0.05209583 - 0.08017121)

####ALL EAS
my <- DCMx[which(DCMx$AN_DCM_EAS >0),]
dim(my)
#665
sum(my$AC_DCM_EAS)
#61
max(my$AN_DCM_EAS)
#210

hmm <- ukb[which(ukb$ukbAN_eas >0),]
sum(hmm$ukbAC_eas)
#180
max(hmm$ukbAN_eas)
#1210
dim(hmm)
#4898

penetrance(x_D=177, n_D=39003, x_AgD=61, n_AgD=210, x_A=180, n_A=1210)
#0.008861596 (0.006633488-0.0118381)

####ALL SAS
my <- DCMx[which(DCMx$AN_DCM_SAS >0),]
dim(my)
#665
sum(my$AC_DCM_SAS)
#30
max(my$AN_DCM_SAS)
#116

hmm <- ukb[which(ukb$ukbAN_sas >0),]
sum(hmm$ukbAC_sas)
#970
max(hmm$ukbAN_sas)
#6272
dim(hmm)
#4898

penetrance(x_D=177, n_D=39003, x_AgD=30, n_AgD=116, x_A=970, n_A=6272)
#0.007589785 (0.005384308-0.01069865)

### DCM in those with ancestry data
my <- DCMx[which(DCMx$AN_DCM_SAS >0 | DCMx$AN_DCM_EUR >0 | DCMx$AN_DCM_EAS > 0 | DCMx$AN_DCM_AFR > 0),]
dim(my)
#665
sum(my$AC_DCM_SAS) + sum(my$AC_DCM_EAS) + sum(my$AC_DCM_EUR) + sum(my$AC_DCM_AFR)
#818
max(my$AN_DCM_SAS) + max(my$AN_DCM_EAS) + max(my$AN_DCM_EUR) + max(my$AN_DCM_AFR)
#5064

hmm <- ukb[which(ukb$ukbAN_sas >0 | ukb$ukbAN_eas > 0 | ukb$ukbAN_afr >0 | ukb$ukbAN_eu >0),]
sum(hmm$ukbAC_sas) + sum(hmm$ukbAC_eas) + sum(hmm$ukbAC_afr) + sum(hmm$ukbAC_eu)
#20718
max(hmm$ukbAN_sas) + max(hmm$ukbAN_eas) + max(hmm$ukbAN_afr) + max(hmm$ukbAN_eu)
#323342
dim(hmm)
#4898

penetrance(x_D=177, n_D=39003, x_AgD=818, n_AgD=5064, x_A=20718, n_A=323342)
#0.01144066 (0.009747506-0.01342791)

#### Age

#0s
penetrance(x_D=(177*(max(DCMx$AN_DCM_0s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_0s), n_AgD=max(DCMx$AN_DCM_0s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.0001830636 (4.379194e-05-0.0007652612)

#10s
penetrance(x_D=(177*(max(DCMx$AN_DCM_10s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_10s), n_AgD=max(DCMx$AN_DCM_10s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.0006084541 (0.0002498235-0.001481912)

#20s
penetrance(x_D=(177*(max(DCMx$AN_DCM_20s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_20s), n_AgD=max(DCMx$AN_DCM_20s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.002068576 (0.001228916-0.003481935)

#30s
penetrance(x_D=(177*(max(DCMx$AN_DCM_30s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_30s), n_AgD=max(DCMx$AN_DCM_30s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.003779147 (0.002603608-0.005485446)

#40s
penetrance(x_D=(177*(max(DCMx$AN_DCM_40s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_40s), n_AgD=max(DCMx$AN_DCM_40s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.006416843 (0.004907914-0.008389689)

#50s
penetrance(x_D=(177*(max(DCMx$AN_DCM_50s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_50s), n_AgD=max(DCMx$AN_DCM_50s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.009945921 (0.00805285-0.01228402)

#60s
penetrance(x_D=(177*(max(DCMx$AN_DCM_60s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_60s), n_AgD=max(DCMx$AN_DCM_60s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.01261949 (0.01050649-0.01515745)

#70s
penetrance(x_D=(177*(max(DCMx$AN_DCM_70s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_70s), n_AgD=max(DCMx$AN_DCM_70s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.01365327 (0.01148092-0.01623667)

#80s
penetrance(x_D=(177*(max(DCMx$AN_DCM_80s)/max(DCMx$AN_DCM_80s))), n_D=(39003), x_AgD=sum(DCMx$AC_DCM_80s), n_AgD=max(DCMx$AN_DCM_80s), x_A=sum(ukb$ukbAC_qced), n_A=max(ukb$ukbAN_qced))
#0.01383151 (0.01164563-0.01642769)

one1 <- c("by 10", 0.0001830636, 4.379194e-05, 0.0007652612)
one2 <- c("by 20", 0.0006084541, 0.0002498235, 0.001481912)
one3 <- c("by 30", 0.002068576, 0.001228916, 0.003481935)
one <- c("by 40", 0.003779147, 0.002603608, 0.005485446)
two <- c("by 50", 0.006416843, 0.004907914, 0.008389689)
thre <- c("by 60", 0.009945921, 0.00805285, 0.01228402)
four <- c("by 70", 0.01261949, 0.01050649, 0.01515745)
five <- c("by 80", 0.01365327, 0.01148092, 0.01623667)
five1 <- c("by 100", 0.01383151, 0.01164563, 0.01642769)

okay <- rbind(one1,one2,one3,one,two,thre,four,five,five1)
colnames(okay) <- c("Curation", "Penetrance", "LCI", "UCI")
okay <- data.frame(okay)
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$order <- seq(1,nrow(okay))

#library(forcats)
f <- okay %>%
  mutate(ordering = rev(order),
         Curation = fct_reorder(Curation, ordering, .desc = T)) %>%
ggplot( aes(x=Curation, y=Penetrance)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.4, position = position_dodge(width = 0.4)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.4), size=4) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1), legend.position = "none") +
  ylab("Penetrance") +
  xlab("Age (years)") +
  ylim(0,0.02)+
  ggtitle("DCM") 
 
library(ggpubr)
ggarrange(a + rremove("x.text") +rremove("xlab"), b, ncol = 1, nrow = 2, heights=c(0.1,0.1), align='v')

### DCM in age cohort
my <- DCMx[which(DCMx$AN_DCM_0s >0 | DCMx$AN_DCM_10s >0 | DCMx$AN_DCM_20s > 0 | DCMx$AN_DCM_30s > 0 | DCMx$AN_DCM_40s >0 | DCMx$AN_DCM_50s >0 | DCMx$AN_DCM_60s >0 | DCMx$AN_DCM_70s >0 | DCMx$AN_DCM_80s>0),]
dim(my)
#665
sum(my$AC_DCM_80s) #cumulative
#388
max(my$AN_DCM_80s)
#1918

hmm <- ukb
sum(hmm$ukbAC_qced)
#22232
max(hmm$ukbAN_qced)
#334956
dim(hmm)
#4898

penetrance(x_D=177, n_D=39003, x_AgD=388, n_AgD=1918, x_A=22232, n_A=334956)
#0.01383151 (0.01164563-0.01642769)

#variant consequences - NMDi 
my <- DCMx[which(DCMx$NMD == "NMD_escaping_variant"),]
dim(my)
#88
sum(my$AC_ALL_DCM)
#102
max(my$AN_ALL_DCM)
#5128

 hmm <- ukb[which(ukb$NMD == "NMD_escaping_variant"),]
 sum(hmm$ukbAC_qced)
# #548
 max(hmm$ukbAN_qced)
# #334956
 dim(hmm)
# #316

penetrance(x_D=177, n_D=39003, x_AgD=102, n_AgD=5128, x_A=548, n_A=334956)
#0.05517487 (0.04273589-0.07123443)

#######rarity
summary(DCMx$gnomADg_AC) # 1-139.00
none <- DCMx[is.na(DCMx$gnomADg_AC),]
one <- DCMx[which(DCMx$gnomADg_AC == 1),]
two <- DCMx[which(DCMx$gnomADg_AC == 2),]
three <- DCMx[which(DCMx$gnomADg_AC == 3),]
four <- DCMx[which(DCMx$gnomADg_AC == 4),]
five <- DCMx[which(DCMx$gnomADg_AC == 5),]
ftoe <- DCMx[which(DCMx$gnomADg_AC > 5 & DCMx$gnomADg_AC < 8),]
etot <- DCMx[which(DCMx$gnomADg_AC > 7 & DCMx$gnomADg_AC < 11),]
etotw <- DCMx[which(DCMx$gnomADg_AC > 10 & DCMx$gnomADg_AC < 21),]
twtoth <- DCMx[which(DCMx$gnomADg_AC > 20 & DCMx$gnomADg_AC < 141),]

ukb$gnomADg_AC <- as.integer(as.character(ukb$gnomADg_AC))
summary(as.integer(as.character(ukb$gnomADg_AC))) # 1-139.000
none2 <- ukb[is.na(ukb$gnomADg_AC),]
one2 <- ukb[which(ukb$gnomADg_AC == 1),]
two2 <- ukb[which(ukb$gnomADg_AC == 2),]
three2 <- ukb[which(ukb$gnomADg_AC == 3),]
four2 <- ukb[which(ukb$gnomADg_AC == 4),]
five2 <- ukb[which(ukb$gnomADg_AC == 5),]
ftoe2 <- ukb[which(ukb$gnomADg_AC > 5 & ukb$gnomADg_AC < 8),]
etot2 <- ukb[which(ukb$gnomADg_AC > 7 & ukb$gnomADg_AC < 11),]
etotw2 <- ukb[which(ukb$gnomADg_AC > 10 & ukb$gnomADg_AC < 21),]
twtoth2 <- ukb[which(ukb$gnomADg_AC > 20 & ukb$gnomADg_AC < 141),]

#AC=0 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(none$AC_ALL_DCM), n_AgD=max(none$AN_ALL_DCM), x_A=sum(none2$ukbAC_qced), n_A=max(none2$ukbAN_qced))
#0.03023825 (0.02543394-0.03595006)

#AC=1 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(one$AC_ALL_DCM), n_AgD=max(one$AN_ALL_DCM), x_A=sum(one2$ukbAC_qced), n_A=max(one2$ukbAN_qced))
#0.01206737 (0.009305428-0.01564908)

#AC=2 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(two$AC_ALL_DCM), n_AgD=max(two$AN_ALL_DCM), x_A=sum(two2$ukbAC_qced), n_A=max(two2$ukbAN_qced))
#0.008365015 (0.005950548-0.01175916)

#AC=3 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(three$AC_ALL_DCM), n_AgD=max(three$AN_ALL_DCM), x_A=sum(three2$ukbAC_qced), n_A=max(three2$ukbAN_qced))
#0.006023212 (0.004020636-0.009023219)

#AC=4 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(four$AC_ALL_DCM), n_AgD=max(four$AN_ALL_DCM), x_A=sum(four2$ukbAC_qced), n_A=max(four2$ukbAN_qced))
#0.004845384 (0.002827329-0.008303858)

#AC=5 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(five$AC_ALL_DCM), n_AgD=max(five$AN_ALL_DCM), x_A=sum(five2$ukbAC_qced), n_A=max(five2$ukbAN_qced))
#0.007738987 (0.00482584-0.01241067)

#AC=6-7 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(ftoe$AC_ALL_DCM), n_AgD=max(ftoe$AN_ALL_DCM), x_A=sum(ftoe2$ukbAC_qced), n_A=max(ftoe2$ukbAN_qced))
#0.004738421 (0.002961176-0.007582337)

#AC=8-10 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(etot$AC_ALL_DCM), n_AgD=max(etot$AN_ALL_DCM), x_A=sum(etot2$ukbAC_qced), n_A=max(etot2$ukbAN_qced))
#0.003310089 (0.001965882-0.00557342)

#AC=11-20 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(etotw$AC_ALL_DCM), n_AgD=max(etotw$AN_ALL_DCM), x_A=sum(etotw2$ukbAC_qced), n_A=max(etotw2$ukbAN_qced))
#0.004320185 (0.003089112-0.006041865)

#AC=21-140 in gnomAD
penetrance(x_D=177, n_D=39003, x_AgD=sum(twtoth$AC_ALL_DCM), n_AgD=max(twtoth$AN_ALL_DCM), x_A=sum(twtoth2$ukbAC_qced), n_A=max(twtoth2$ukbAN_qced))
#0.005539311 (0.004367208-0.007025993)

bag <- c("0", 0.03023825, 0.02543394, 0.03595006)
des <- c("1", 0.01206737, 0.009305428, 0.01564908)
dsp <- c("2", 0.008365015, 0.005950548, 0.01175916)
one <- c("3", 0.006023212, 0.004020636, 0.009023219)
two <- c("4", 0.004845384, 0.002827329, 0.008303858)
thre <- c("5", 0.007738987, 0.00482584, 0.01241067)
four <- c("6", 0.004738421, 0.002961176, 0.007582337)
five <- c("8", 0.003310089, 0.001965882, 0.00557342)
six <- c("11", 0.004320185, 0.003089112, 0.006041865)
sev <- c("21+", 0.005539311, 0.004367208, 0.007025993)

okay <- rbind(bag, des, dsp, one,two,thre,four,five,six, sev)
colnames(okay) <- c("gnomAD_AC", "Penetrance", "LCI", "UCI")
okay <- data.frame(okay)
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$order <- seq(nrow(okay),1)

#library(forcats)
e <- okay %>%
mutate(ordering = as.numeric(order),
        gnomAD_AC = fct_reorder(gnomAD_AC, ordering, .desc = T)) %>%
ggplot( aes(x=gnomAD_AC, y=Penetrance, group=gnomAD_AC)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.4)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.4), size=5) +
  theme_minimal() +
  theme(text = element_text(size=16), legend.position = "none") +
  ylab("Penetrance") +
  xlab("gnomAD AC") +
  ylim(0,0.04)+
  ggtitle("DCM") +
        scale_x_discrete(guide = guide_axis(n.dodge = 2)) 

ggarrange(a + rremove("x.text") +rremove("xlab"), b, ncol = 1, nrow = 2, heights=c(0.1,0.12), align='v')
```

# aggregate - HCM and DCM by Gene + Conseq

```{r}
HCMx <- read.table("HCMx.txt", header=T, sep="\t") #before penetrance filter
HCMx$REVEL <- as.numeric(as.character(HCMx$REVEL))
ukb <- read.table("ukb_hcm_match.txt", header=T, sep="\t")

#### GENES
table(HCMx$Consequence)
table(ukb$Consequence)

genelist <- c("MYBPC3", "MYH7", "MYL2", "MYL3", "TNNT2", "TPM1", "TNNI3", "ACTC1")

#First script - cases
df <- data.frame("test", "test", "0", "0", "0")
colnames(df) <- c("Gene", "Variants", "AC_ALL_HCM", "AN_ALL_HCM", "count")

for (gene in genelist){
hmm <- HCMx[which(HCMx$SYMBOL == gene),]

#splice + splice ai pass
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df[(nrow(df)+1),] <- data.frame(gene, "splice region (SpliceAI>=0.80)", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#splice + splice ai fail
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
df[(nrow(df)+1),] <- data.frame(gene, "splice region (SpliceAI<0.80)", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                   Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-") 
df[(nrow(df)+1),] <- data.frame(gene, "PTCs", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

# NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df[(nrow(df)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#missense etc + revel pass
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant") %>%
  filter(REVEL >= 0.75)
df[(nrow(df)+1),] <- data.frame(gene, "missense (REVEL>=0.75)", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#missense etc + revel fail
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant" ) %>%
  filter(REVEL < 0.75 | is.na(REVEL))
df[(nrow(df)+1),] <- data.frame(gene, "missense (REVEL<0.75)", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#other PAV
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df[(nrow(df)+1),] <- data.frame(gene, "other PAV", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))
}

#Second script - UKB
df_ukb <- data.frame("test", "test", "0", "0", "0")
colnames(df_ukb) <- c("Gene", "Variants", "ukbAC_qced", "ukbAN_qced", "count")

for (gene in genelist){
hmm <- ukb[which(ukb$SYMBOL == gene),]

#splice ai pass
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "splice region (SpliceAI>=0.80)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#splice ai fail
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "splice region (SpliceAI<0.80)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

# NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#missense etc + revel pass
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant") %>%
  filter(REVEL >= 0.75)
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "missense (REVEL>=0.75)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#missense etc + revel fail
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant" ) %>%
  filter(REVEL < 0.75 | is.na(REVEL))
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "missense (REVEL<0.75)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#other PAV
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "other PAV", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))
}

### stick together and estimate penetrance

done <- data.frame(df, df_ukb[3:5])
done <- done[!done$Gene == "test",]
done <- done[!(as.integer(as.character(done$AC_ALL_HCM))) < 2,]
done <- done[!(as.integer(as.character(done$ukbAC_qced))) < 2,]

done[(ncol(done)+1):(ncol(done)+3)] <- penetrance(x_D=97, n_D=52660, 
                                                  x_AgD=(as.integer(as.character(done$AC_ALL_HCM))), n_AgD=(as.integer(as.character(done$AN_ALL_HCM))), 
                                                  x_A=(as.integer(as.character(done$ukbAC_qced))), n_A=(as.integer(as.character(done$ukbAN_qced))))
colnames(done)[(ncol(done)-2):(ncol(done))] <- c("Penetrance", "LCI", "UCI")
okay <- done
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$Type <- ""
okay$Type[okay$Variants == "splice region (SpliceAI>=0.80)"] <- "LoF"
okay$Type[okay$Variants == "splice region (SpliceAI<0.80)"] <- "LoF"
okay$Type[okay$Variants == "PTCs"] <- "LoF"
okay$Type[okay$Variants == "NMDi PTCs"] <- "non-LoF"
okay$Type[okay$Variants == "missense (REVEL>=0.75)"] <- "non-LoF"
okay$Type[okay$Variants == "missense (REVEL<0.75)"] <- "non-LoF"
okay$Type[okay$Variants == "other PAV"] <- "non-LoF"

#library(forcats)
a <- okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance") +   
  xlab("Gene") +
  coord_cartesian(ylim=c(0,1)) +
  ggtitle("HCM") +
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
  scale_shape_manual(values = c(0, 15, 16, 18, 1, 2, 17))+
 guides(shape=guide_legend(nrow=4))

b <- okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
#, axis.text.x = element_text(size=12, angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance") +   
  xlab("Gene") +
  coord_cartesian(ylim=c(0,0.2)) +
  ggtitle("HCM zoom")+
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
  scale_shape_manual(values = c(0, 15, 16, 18, 1, 2, 17))

library(ggpubr)
ggpubr::ggarrange(a + rremove("xlab") + rremove("x.text"), b + rremove("xlab"), ncol = 1, nrow = 2, align = "v",  heights=c(0.1,0.12), common.legend = TRUE, legend = "bottom")

############################

#DCM
DCMx <- read.table("DCMx.txt", header=T, sep="\t") #before penetrance filter
ukb <- read.table("ukb_dcm_match.txt", header=T, sep="\t")
DCMx$REVEL <- as.numeric(as.character(DCMx$REVEL))

## TTNtvs alone
my <- DCMx[which(DCMx$SYMBOL == "TTN"),]
my2 <- ukb[which(ukb$SYMBOL == "TTN"),]
penetrance(x_D=177, n_D=39003, sum(my$AC_ALL_DCM), max(my$AN_ALL_DCM), sum(my2$ukbAC_qced), max(my2$ukbAN_qced))
#0.09837075 (0.08010638-0.1207994)

#### GENES
table(DCMx$Consequence)
table(ukb$Consequence)

genelist <- c("TTN", "BAG3", "DES", "LMNA", "MYH7", "PLN", "RBM20", "SCN5A", "TNNC1", "TNNT2", "DSP")

#First script - cases
df <- data.frame("test", "test", "0", "0", "0")
colnames(df) <- c("Gene", "Variants", "AC_ALL_DCM", "AN_ALL_DCM", "count")

for (gene in genelist){
hmm <- DCMx[which(DCMx$SYMBOL == gene),]

# splice ai pass
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df[(nrow(df)+1),] <- data.frame(gene, "splice region (SpliceAI>=0.80)", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#splice ai fail
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
df[(nrow(df)+1),] <- data.frame(gene, "splice region (SpliceAI<0.80)", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant"|
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-")
df[(nrow(df)+1),] <- data.frame(gene, "PTCs", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df[(nrow(df)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#missense etc + revel pass
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant") %>%
  filter(REVEL >= 0.75)
df[(nrow(df)+1),] <- data.frame(gene, "missense (REVEL>=0.75)", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#missense etc + revel fail
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant" ) %>%
  filter(REVEL < 0.75 | is.na(REVEL))
df[(nrow(df)+1),] <- data.frame(gene, "missense (REVEL<0.75)", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#other PAVs
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df[(nrow(df)+1),] <- data.frame(gene, "other PAV", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))
}

#Second script - UKB
df_ukb <- data.frame("test", "test", "0", "0", "0")
colnames(df_ukb) <- c("Gene", "Variants", "ukbAC_qced", "ukbAN_qced", "count")

for (gene in genelist){
hmm <- ukb[which(ukb$SYMBOL == gene),]

#splice ai pass
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "splice region (SpliceAI>=0.80)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#splice ai fail
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "FAIL" | SpliceAI_cutoff == "-")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "splice region (SpliceAI<0.80)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                   Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#missense etc + revel pass
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant") %>%
  filter(REVEL >= 0.75)
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "missense (REVEL>=0.75)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#missense etc + revel fail
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant" ) %>%
  filter(REVEL < 0.75 | is.na(REVEL))
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "missense (REVEL<0.75)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#other PAVs
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "other PAV", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))
}

### stick together and estimate penetrance

done <- data.frame(df, df_ukb[3:5])
done <- done[!done$Gene == "test",]
done <- done[!(as.integer(as.character(done$AC_ALL_DCM))) < 2,]
done <- done[!(as.integer(as.character(done$ukbAC_qced))) < 2,]

done[(ncol(done)+1):(ncol(done)+3)] <- penetrance(x_D=177, n_D=39003, 
                                                  x_AgD=(as.integer(as.character(done$AC_ALL_DCM))), n_AgD=(as.integer(as.character(done$AN_ALL_DCM))), 
                                                  x_A=(as.integer(as.character(done$ukbAC_qced))), n_A=(as.integer(as.character(done$ukbAN_qced))))
colnames(done)[(ncol(done)-2):(ncol(done))] <- c("Penetrance", "LCI", "UCI")
okay <- done
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$Gene[okay$Gene == "TTN"] <- "TTN*"
okay$Type <- ""
okay$Type[okay$Variants == "splice region (SpliceAI>=0.80)"] <- "LoF"
okay$Type[okay$Variants == "splice region (SpliceAI<0.80)"] <- "LoF"
okay$Type[okay$Variants == "PTCs"] <- "LoF"
okay$Type[okay$Variants == "NMDi PTCs"] <- "non-LoF"
okay$Type[okay$Variants == "missense (REVEL>=0.75)"] <- "non-LoF"
okay$Type[okay$Variants == "missense (REVEL<0.75)"] <- "non-LoF"
okay$Type[okay$Variants == "other PAV"] <- "non-LoF"

library(forcats)
c <- okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=15), legend.title = element_text(size=10), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
#, axis.text.x = element_text(size=12, angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance and 95% CI") +   
  xlab("Gene") +
   coord_cartesian(ylim=c(0,1)) +
 ggtitle("DCM")+
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
  scale_shape_manual(values = c(0, 15, 16, 18, 1, 2, 17))

d <- okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=15), legend.title = element_text(size=10), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
#, axis.text.x = element_text(size=12, angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance and 95% CI") +   
  xlab("Gene") +
   coord_cartesian(ylim=c(0,0.2)) +
 ggtitle("DCM zoom")+
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
  scale_shape_manual(values = c(0, 15, 16, 18, 1, 2, 17))

ggpubr::ggarrange(a + rremove("xlab") + rremove("x.text"),  c + rremove("xlab") + rremove("x.text") + rremove("ylab"), b + rremove("xlab"), d + rremove("xlab") + rremove("ylab"), ncol = 2, nrow = 2, align = "v",  heights=c(0.1,0.125), common.legend = TRUE, legend = "bottom")

ggarrange(c + rremove("xlab") + rremove("x.text") + rremove("ylab"), d + rremove("xlab") + rremove("ylab"), ncol = 1, nrow = 2, align = "v",  heights=c(0.1,0.125), common.legend = TRUE, legend = "bottom")
```

# aggregate - HCM and DCM by Gene + Conseq - shorter - Figure 2

```{r}
HCMx <- read.table("HCMx.txt", header=T, sep="\t") #before penetrance filter
HCMx$REVEL <- as.numeric(as.character(HCMx$REVEL))
ukb <- read.table("ukb_hcm_match.txt", header=T, sep="\t")

#### GENES
table(HCMx$Consequence)
table(ukb$Consequence)

genelist <- c("MYBPC3", "MYH7", "MYL2", "MYL3", "TNNT2", "TPM1", "TNNI3", "ACTC1")

#First script - cases
df <- data.frame("test", "test", "0", "0", "0")
colnames(df) <- c("Gene", "Variants", "AC_ALL_HCM", "AN_ALL_HCM", "count")

for (gene in genelist){
hmm <- HCMx[which(HCMx$SYMBOL == gene),]

#splice pass >0.8
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df[(nrow(df)+1),] <- data.frame(gene, "splice region", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                   Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-") 
df[(nrow(df)+1),] <- data.frame(gene, "PTCs", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

# NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df[(nrow(df)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#missense
lof1 <- hmm %>% filter(Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant")
df[(nrow(df)+1),] <- data.frame(gene, "missense", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))

#other PAV
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df[(nrow(df)+1),] <- data.frame(gene, "other PAV", sum(lof1$AC_ALL_HCM), max(lof1$AN_ALL_HCM), nrow(lof1))
}

#Second script - UKB
df_ukb <- data.frame("test", "test", "0", "0", "0")
colnames(df_ukb) <- c("Gene", "Variants", "ukbAC_qced", "ukbAN_qced", "count")

for (gene in genelist){
hmm <- ukb[which(ukb$SYMBOL == gene),]

#splice
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "splice region (SpliceAI>=0.80)", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

# NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#missense
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "missense", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#other PAV
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "other PAV", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))
}

### stick together and estimate penetrance

done <- data.frame(df, df_ukb[3:5])
done <- done[!done$Gene == "test",]
done <- done[!(as.integer(as.character(done$AC_ALL_HCM))) < 2,]
done <- done[!(as.integer(as.character(done$ukbAC_qced))) < 2,]

done[(ncol(done)+1):(ncol(done)+3)] <- penetrance(x_D=97, n_D=52660, 
                                                  x_AgD=(as.integer(as.character(done$AC_ALL_HCM))), n_AgD=(as.integer(as.character(done$AN_ALL_HCM))), 
                                                  x_A=(as.integer(as.character(done$ukbAC_qced))), n_A=(as.integer(as.character(done$ukbAN_qced))))
colnames(done)[(ncol(done)-2):(ncol(done))] <- c("Penetrance", "LCI", "UCI")
okay <- done
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$Type <- ""
okay$Type[okay$Variants == "splice region"] <- "LoF"
okay$Type[okay$Variants == "PTCs"] <- "LoF"
okay$Type[okay$Variants == "NMDi PTCs"] <- "non-LoF"
okay$Type[okay$Variants == "missense"] <- "non-LoF"
okay$Type[okay$Variants == "other PAV"] <- "non-LoF"
okay$Variants[okay$Variants == "splice region"] <- "splice region, LoF"
okay$Variants[okay$Variants == "PTCs"] <- "PTCs, LoF"
okay$Variants[okay$Variants == "NMDi PTCs"] <- "NMDi PTCs, non-LoF"
okay$Variants[okay$Variants == "missense"] <- "missense, non-LoF"
okay$Variants[okay$Variants == "other PAV"] <- "other PAV, non-LoF"

culr <- c("cornflowerblue", "cornflowerblue", "cornflowerblue", "darkturquoise","darkturquoise")

#library(forcats)    
a <- okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance") +   
  xlab("Gene") +
  coord_cartesian(ylim=c(0,1)) +
  ggtitle("HCM") +
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
  scale_shape_manual(values = c(17, 15, 16, 15, 18))+
 guides(shape=guide_legend(nrow=2))+
labs(color=NULL, shape=NULL)

b <-  okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance") +   
  xlab("Gene") +
  coord_cartesian(ylim=c(0,0.2)) +
  ggtitle("HCM zoom") +
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
  scale_shape_manual(values = c(17, 15, 16, 15, 18))+
 guides(shape=guide_legend(nrow=2))+
  labs(color = NULL, shape = NULL)

############################

#DCM
DCMx <- read.table("DCMx.txt", header=T, sep="\t") #before penetrance filter
ukb <- read.table("ukb_dcm_match.txt", header=T, sep="\t")
DCMx$REVEL <- as.numeric(as.character(DCMx$REVEL))

#### GENES
table(DCMx$Consequence)
table(ukb$Consequence)

genelist <- c("TTN", "BAG3", "DES", "LMNA", "MYH7", "PLN", "RBM20", "SCN5A", "TNNC1", "TNNT2", "DSP")

#First script - cases
df <- data.frame("test", "test", "0", "0", "0")
colnames(df) <- c("Gene", "Variants", "AC_ALL_DCM", "AN_ALL_DCM", "count")

for (gene in genelist){
hmm <- DCMx[which(DCMx$SYMBOL == gene),]

# splice
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df[(nrow(df)+1),] <- data.frame(gene, "splice region", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant"|
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-")
df[(nrow(df)+1),] <- data.frame(gene, "PTCs", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df[(nrow(df)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#missense
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant")
df[(nrow(df)+1),] <- data.frame(gene, "missense", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))

#other PAVs
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df[(nrow(df)+1),] <- data.frame(gene, "other PAV", sum(lof1$AC_ALL_DCM), max(lof1$AN_ALL_DCM), nrow(lof1))
}

#Second script - UKB
df_ukb <- data.frame("test", "test", "0", "0", "0")
colnames(df_ukb) <- c("Gene", "Variants", "ukbAC_qced", "ukbAN_qced", "count")

for (gene in genelist){
hmm <- ukb[which(ukb$SYMBOL == gene),]

#splice
lof1 <- hmm %>% filter(Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant") %>%
          filter(NMD == "-") %>%
  filter(SpliceAI_cutoff == "PASS")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "splice region", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#NMDc PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                   Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "-")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#NMDi PTCs
lof1 <- hmm %>% filter(Consequence == "frameshift_variant" |
                   Consequence == "frameshift_variant,splice_region_variant" |
                  Consequence == "splice_acceptor_variant" |
                    Consequence == "splice_acceptor_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                    Consequence == "splice_confirmed" |
                  Consequence == "splice_donor_5th_base_variant,intron_variant" |
                   Consequence == "splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_donor_variant" |
                   Consequence == "splice_donor_variant,coding_sequence_variant" |
                   Consequence == "splice_donor_variant,splice_donor_5th_base_variant,coding_sequence_variant,intron_variant" |
                    Consequence == "splice_donor_variant,splice_donor_region_variant,coding_sequence_variant,intron_variant" |
                   Consequence == "splice_donor_variant,splice_donor_region_variant,intron_variant" |
                  Consequence == "splice_polypyrimidine_tract_variant,intron_variant" |
                  Consequence == "splice_region_variant,splice_polypyrimidine_tract_variant,intron_variant" |
                   Consequence == "splice_polypyrimidine_tract_variant,splice_region_variant,intron_variant" |
                   Consequence == "splice_region_variant,intron_variant" |
                    Consequence == "splice_region_variant,synonymous_variant" |
                  Consequence == "stop_gained" |
                   Consequence == "stop_gained,frameshift_variant" |
                   Consequence == "stop_gained,splice_region_variant") %>%
          filter(NMD == "NMD_escaping_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "NMDi PTCs", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#missense
lof1 <- hmm %>% filter(
                  Consequence == "missense_variant" |
                  Consequence == "missense_variant,splice_region_variant") 
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "missense", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))

#other PAVs
lof1 <- hmm %>% filter(Consequence == "inframe_deletion" |
                    Consequence == "inframe_deletion,splice_region_variant" |
                  Consequence == "inframe_insertion" |
                  Consequence == "start_lost" |
                    Consequence == "stop_lost" |
                    Consequence == "start_lost,splice_region_variant")
df_ukb[(nrow(df_ukb)+1),] <- data.frame(gene, "other PAV", sum(lof1$ukbAC_qced), max(lof1$ukbAN_qced), nrow(lof1))
}

### stick together and estimate penetrance

done <- data.frame(df, df_ukb[3:5])
done <- done[!done$Gene == "test",]
done <- done[!(as.integer(as.character(done$AC_ALL_DCM))) < 2,]
done <- done[!(as.integer(as.character(done$ukbAC_qced))) < 2,]

done[(ncol(done)+1):(ncol(done)+3)] <- penetrance(x_D=177, n_D=39003, 
                                                  x_AgD=(as.integer(as.character(done$AC_ALL_DCM))), n_AgD=(as.integer(as.character(done$AN_ALL_DCM))), 
                                                  x_A=(as.integer(as.character(done$ukbAC_qced))), n_A=(as.integer(as.character(done$ukbAN_qced))))
colnames(done)[(ncol(done)-2):(ncol(done))] <- c("Penetrance", "LCI", "UCI")
okay <- done
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$Gene[okay$Gene == "TTN"] <- "TTN*"
okay$Type <- ""
okay$Type[okay$Variants == "splice region"] <- "LoF"
okay$Type[okay$Variants == "PTCs"] <- "LoF"
okay$Type[okay$Variants == "NMDi PTCs"] <- "non-LoF"
okay$Type[okay$Variants == "missense"] <- "non-LoF"
okay$Type[okay$Variants == "other PAV"] <- "non-LoF"
okay$Variants[okay$Variants == "splice region"] <- "splice region, LoF"
okay$Variants[okay$Variants == "PTCs"] <- "PTCs, LoF"
okay$Variants[okay$Variants == "NMDi PTCs"] <- "NMDi PTCs, non-LoF"
okay$Variants[okay$Variants == "missense"] <- "missense, non-LoF"
okay$Variants[okay$Variants == "other PAV"] <- "other PAV, non-LoF"

#library(forcats)
c <- okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance") +   
  xlab("Gene") +
   coord_cartesian(ylim=c(0,1)) +
 ggtitle("DCM")+
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
   scale_shape_manual(values = c(17, 15, 16, 15, 18))

d <- okay %>%
  ggplot( aes(x=Gene, y=Penetrance, color=Type, shape=Variants)) +
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.5)) +
    geom_point(position = position_dodge(width = 0.5), size=3) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) +
  ylab("Penetrance") +   
  xlab("Gene") +
   coord_cartesian(ylim=c(0,0.2)) +
 ggtitle("DCM zoom")+
    theme(legend.position="top", legend.box="vertical", legend.margin=margin()) +
  scale_color_manual(values=c("darkturquoise","cornflowerblue") ) +
   scale_shape_manual(values = c(17, 15, 16, 15, 18))

ggarrange(a + rremove("xlab") + rremove("x.text"),  c + rremove("xlab") + rremove("x.text") + rremove("ylab"), b + rremove("xlab"), d + rremove("xlab") + rremove("ylab"), ncol = 2, nrow = 2, align = "v",  heights=c(0.1,0.125), common.legend = TRUE, legend = "bottom")
```

# aggregate - Pathogenicity curation

```{r}
HCM <- read.table("HCMx.txt", header=T, sep="\t")
DCM <- read.table("DCMx.txt", header=T, sep="\t")
hcm <- read.table("ukb_hcm_match.txt", header=T, sep="\t")
dcm <- read.table("ukb_dcm_match.txt", header=T, sep="\t")

### HCM
#e.g. pathogenic
my <- HCM[which(HCM$Curation == "P"),]
dim(my)
#173
sum(my$AC_ALL_HCM)
#1424
max(my$AN_ALL_HCM)
#20800 

ukb <- hcm[which(hcm$Curation == "P"),]
dim(ukb)
#30
sum(ukb$ukbAC_qced)
#188
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=97, n_D=52660, x_AgD=1424, n_AgD=20800, x_A=188, n_A=334956)
#0.2246838 (0.1750868-0.28833)

#e.g. LP
my <- HCM[which(HCM$Curation == "LP"),]
dim(my)
#316
sum(my$AC_ALL_HCM)
#1322
max(my$AN_ALL_HCM)
#20800 

ukb <- hcm[which(hcm$Curation == "LP"),]
dim(ukb)
#98
sum(ukb$ukbAC_qced)
#573
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=97, n_D=52660, x_AgD=1322, n_AgD=20800, x_A=573, n_A=334956)
#0.06843807 (0.0548813-0.08534362)

###################
### P/LP for comparison with JACC
#e.g. LP
my <- HCM[which(HCM$Curation == "LP" | HCM$Curation == "P"),]
dim(my)
#489
sum(my$AC_ALL_HCM)
#2746
max(my$AN_ALL_HCM)
#20800 

ukb <- hcm[which(hcm$Curation == "LP" | hcm$Curation == "P"),]
dim(ukb)
#128
sum(ukb$ukbAC_qced)
#761
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=97, n_D=52660, x_AgD=2746, n_AgD=20800, x_A=761, n_A=334956)
#0.1070377 (0.08646167-0.1325104)

#####
#e.g. VUS
my <- HCM[which(HCM$Curation == "VUS"),]
dim(my)
#832
sum(my$AC_ALL_HCM)
#1498
max(my$AN_ALL_HCM)
#20800 

ukb <- hcm[which(hcm$Curation == "VUS"),]
dim(ukb)
#1552
sum(ukb$ukbAC_qced)
#7622
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=97, n_D=52660, x_AgD=1498, n_AgD=20800, x_A=7622, n_A=334956)
#0.005829938 (0.004747342-0.007159412)

#e.g. LB
my <- HCM[which(HCM$Curation == "LB"),]
dim(my)
#19
sum(my$AC_ALL_HCM)
#70
max(my$AN_ALL_HCM)
#20800 

ukb <- hcm[which(hcm$Curation == "LB"),]
dim(ukb)
#53
sum(ukb$ukbAC_qced)
#772
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=97, n_D=52660, x_AgD=70, n_AgD=20800, x_A=772, n_A=334956)
#0.002689755 (0.001964929-0.003681954)

###try match SARC-IND JACC paper
my <- HCM[which(HCM$Curation == "LB" | HCM$Curation == "VUS"),]
dim(my)
#851
sum(my$AC_ALL_HCM)
#1568
max(my$AN_ALL_HCM)
#20800 

ukb <- hcm[which(hcm$Curation == "LB" | hcm$Curation == "VUS"),]
dim(ukb)
#1605
sum(ukb$ukbAC_qced)
#8394
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=97, n_D=52660, x_AgD=1568, n_AgD=20800, x_A=8394, n_A=334956)
#0.005541128 (0.00451393-0.006802076)

# NO Benign

#plot
one <- c("Pathogenic", 0.2246838 , 0.1750868, 0.28833)
thre <- c("Likely Pathogenic",        0.06843807 , 0.0548813, 0.08534362)
five <- c("Unknown Significance",       0.005829938 , 0.004747342, 0.007159412)
sev <- c("Benign / Likely Benign",         0.002689755 , 0.001964929, 0.003681954)
okay <- rbind(one,thre,five,sev)
colnames(okay) <- c("Curation", "Penetrance", "LCI", "UCI")
okay <- data.frame(okay)
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$Curation <- factor(okay$Curation, levels = c("Pathogenic", "Likely Pathogenic", "Unknown Significance", "Benign / Likely Benign"))

culr <- c("red", "orange", "lightgoldenrod4","darkgrey")
scaleFUN <- function(x) sprintf("%.2f", x)

library(forcats)
a <- okay %>% 
  mutate(ordering = (as.numeric(Penetrance)),
         Curation = fct_reorder(Curation, ordering, .desc = T)) %>%
ggplot( aes(x=Curation, y=Penetrance, color=Curation)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.4)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.4), size=5) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, color=culr)) +
  ylab("Penetrance") +
  ggtitle("HCM")+
 scale_color_manual(values=c("Pathogenic"="red", "Likely Pathogenic"="orange", "Unknown Significance"="lightgoldenrod4", "Benign / Likely Benign" = "darkgrey"), drop = FALSE)+
 theme(legend.position="none", legend.box="vertical")+
  scale_y_continuous(labels=scaleFUN)

################## DCM

#e.g. pathogenic
my <- DCM[which(DCM$Curation == "P"),]
dim(my)
#21
sum(my$AC_ALL_DCM)
#39
max(my$AN_ALL_DCM)
#5128 

ukb <- dcm[which(dcm$Curation == "P"),]
dim(ukb)
#22
sum(ukb$ukbAC_qced)
#59
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=39, n_AgD=5128, x_A=59, n_A=334956)
#0.1959517 (0.1278112-0.3004203)

#e.g. LP
my <- DCM[which(DCM$Curation == "LP"),]
dim(my)
#247
sum(my$AC_ALL_DCM)
#315
max(my$AN_ALL_DCM)
#5128 

ukb <- dcm[which(dcm$Curation == "LP"),]
dim(ukb)
#577
sum(ukb$ukbAC_qced)
#942
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=315, n_AgD=5128, x_A=942, n_A=334956)
#0.09912355 (0.08177105-0.1201584)

##### for analysis to other papers
#e.g. pathogenic
my <- DCM[which(DCM$Curation == "P" | DCM$Curation == "LP"),]
dim(my)
#268
sum(my$AC_ALL_DCM)
#354
max(my$AN_ALL_DCM)
#5128 

ukb <- dcm[which(dcm$Curation == "P" | dcm$Curation == "LP"),]
dim(ukb)
#599
sum(ukb$ukbAC_qced)
#1001
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=354, n_AgD=5128, x_A=1001, n_A=334956)
#0.1048302 (0.0868382-0.1265499)

####

#e.g. VUS
my <- DCM[which(DCM$Curation == "VUS"),]
dim(my)
#356
sum(my$AC_ALL_DCM)
#415
max(my$AN_ALL_DCM)
#5128 

ukb <- dcm[which(dcm$Curation == "VUS"),]
dim(ukb)
#4183
sum(ukb$ukbAC_qced)
#18429
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=415, n_AgD=5128, x_A=18429, n_A=334956)
#0.006675186 (0.005609769-0.007942948)

#e.g. LB
my <- DCM[which(DCM$Curation == "LB"),]
dim(my)
#37
sum(my$AC_ALL_DCM)
#56
max(my$AN_ALL_DCM)
#5128 

ukb <- dcm[which(dcm$Curation == "LB"),]
dim(ukb)
#109
sum(ukb$ukbAC_qced)
#2552
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=56, n_AgD=5128, x_A=2552, n_A=334956)
#0.006504908 (0.004816721-0.00878478)

# Benign
my <- DCM[which(DCM$Curation == "B"),]
dim(my)
#4
sum(my$AC_ALL_DCM)
#8
max(my$AN_ALL_DCM)
#5128 

ukb <- dcm[which(dcm$Curation == "B"),]
dim(ukb)
#7
sum(ukb$ukbAC_qced)
#250
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=8, n_AgD=5128, x_A=250, n_A=334956)
#0.009502722 (0.004725481-0.01910953)

### Benign or LB
my <- DCM[which(DCM$Curation == "B" | DCM$Curation == "LB"),]
dim(my)
#41
sum(my$AC_ALL_DCM)
#64
max(my$AN_ALL_DCM)
#5128 

ukb <- dcm[which(dcm$Curation == "B" | dcm$Curation == "LB"),]
dim(ukb)
#116
sum(ukb$ukbAC_qced)
#2802
max(ukb$ukbAN_qced)
#334956

penetrance(x_D=177, n_D=39003, x_AgD=64, n_AgD=5128, x_A=2802, n_A=334956)
#0.006770826 (0.005087423-0.00901126)

#plot
one <- c("Pathogenic",            0.1959517 , 0.1278112, 0.3004203)
thre <- c("Likely Pathogenic",    0.09912355 , 0.08177105, 0.1201584)
five <- c("Unknown Significance", 0.006675186 , 0.005609769, 0.007942948)
sev <- c("Benign / Likely Benign",0.006770826, 0.005087423, 0.00901126)
#eig <- c("Benign",                0.009502722 , 0.004725481, 0.01910953)

okay <- rbind(one,thre,five,sev)
colnames(okay) <- c("Curation", "Penetrance", "LCI", "UCI")
okay <- data.frame(okay)
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))
okay$order <- c(4,3,2,1)
scaleFUN <- function(x) sprintf("%.2f", x)

#library(forcats)
d <- okay %>%
  mutate(ordering = as.numeric(order),
         Curation = fct_reorder(Curation, ordering, .desc = T)) %>%
ggplot( aes(x=Curation, y=Penetrance, color=Curation)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.5, position = position_dodge(width = 0.4)) +
    geom_point(position = position_dodge(width = 0.4), size=5) +
  theme_minimal() +
  theme(text = element_text(size=16), axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1, color=culr, size=13), legend.position = "none") +
  ylab("Penetrance") +
  xlab("Variant curation") +
  ggtitle("DCM")+
   scale_color_manual(values=c("Benign / Likely Benign" = "grey", "Unknown Significance"="lightgoldenrod4", "Likely Pathogenic"="orange", "Pathogenic"="red"), drop = FALSE)+
  scale_y_continuous(labels=scaleFUN)

library(ggpubr)
ggarrange(a + rremove("xlab") + rremove("x.text"), b + rremove("xlab") + rremove("x.text"), ncol = 1, nrow = 2, align = "v",  heights=c(0.1,0.1), common.legend = TRUE, legend = "bottom")

####

thre <- c("LP", 0.3071534, 0.1904164, 0.4954573, "ALL")
four <- c("LP", 0.3305651, 0.194836, 0.5608474, "EUR")
five <- c("VUS", 0.02370721, 0.01803703, 0.0311599, "ALL")
six <- c("VUS", 0.0164344, 0.01151188, 0.0234618, "EUR")
sev <- c("LB", 0.0140806, 0.009502998, 0.02086323, "ALL")
eig <- c("LB", 0.01935581, 0.01101795, 0.03400336, "EUR")
one <- c("Benign", 0.01062347, 0.004819192, 0.02341846, "ALL")
okay <- rbind(thre,four,five,six,sev,eig,one)
colnames(okay) <- c("Curation", "Penetrance", "LCI", "UCI", "type")
okay <- data.frame(okay)
okay$Penetrance <- as.numeric(as.character(okay$Penetrance))
okay$UCI <- as.numeric(as.character(okay$UCI))
okay$LCI <- as.numeric(as.character(okay$LCI))

library(forcats)
okay %>%
  mutate(ordering = as.numeric(Penetrance),
         Curation = fct_reorder(Curation, ordering, .desc = T)) %>%
ggplot( aes(x=Curation, y=Penetrance, colour=type, group=type)) + 
  geom_errorbar(aes(ymin=LCI, ymax=UCI), width=.1, position = position_dodge(width = 0.4)) +
  # geom_line(position = position_dodge(width = 0.2)) +
    geom_point(position = position_dodge(width = 0.4), size=2) +
  theme_minimal() +
  theme(text = element_text(size=20), axis.text.x = element_text(size=12)) +
  ylab("Estimate of penetrance") +
  xlab("Variant") +
  ylim(0,1)+
  ggtitle("DCM")
```

# Figure 1

```{r}
#library(ggpubr)

ggarrange(a + rremove("x.text")+rremove("xlab"), b + rremove("x.text")+rremove("xlab")+rremove("ylab"), c + rremove("x.text")+rremove("xlab")+rremove("ylab"), d, e+rremove("ylab"), f+rremove("ylab"), ncol = 3, nrow = 2, align='v', heights=c(0.1,0.15))
```

# Supplementary information

# HCM prevalence plot - baseline risk

```{r}
hcm <- read.table("HCMprev.txt", header=T)
colnames(hcm) <- c("Reference", "Prevalence", "HCM", "Cohort", "colour")
pvl <- function(a, b, p=0.5) {binom.test(a, b, 0.5, alternative=
                            c("two.sided"), conf.level = 0.95)$p.value}
lci <- function(a, b, p=0.5) {binom.test(a, b, 0.5, alternative=
                            c("two.sided"), conf.level = 0.95)$conf.int[1]}
uci <- function(a, b, p=0.5) {binom.test(a, b, 0.5, alternative=
                            c("two.sided"), conf.level = 0.95)$conf.int[2]}
#slow
hcm$p <- mapply(pvl, a=as.integer(hcm$HCM), b=hcm$Cohort)
hcm$lci <- mapply(lci, a=as.integer(hcm$HCM), b=hcm$Cohort)
hcm$uci <- mapply(uci, a=as.integer(hcm$HCM), b=hcm$Cohort)
library(ggplot2)
library(forcats)
#binom.test(1, 500, 0.5, alternative="two.sided", conf.level=0.95)
#hcm$Reference <- as.character(hcm$Reference)
#hcm[23,] <- c("1 in 500", "0.002", "0", "0", "0", "0", "5.063433e-05", "1.109248e-02")
hcm$Prevalence <- as.numeric(hcm$Prevalence)
hcm$Reference <- as.factor(hcm$Reference)
hcm$uci <- as.numeric(hcm$uci)
hcm$lci <- as.numeric(hcm$lci)
library(scales)
library(dplyr)

a <-hcm %>%
   mutate(Reference = fct_relevel(Reference, 
            "Aronow et al. 1988", "Maro et al. 2006", "Massera et al. 2019", 
            "Shapiro et al. 1983", "Maron et al. 1999", "Hada et al. 1987", 
            "Basavarajaiah et al. 2008", "Corrado et al. 1998",
            "Nistri et al. 2003", "Ng et al. 2011",  
            
            "Maron et al. 2004", "de Marvao et al. 2021", "Maron et al. 1995", 
            "Zou et al. 2004", 
            
            "Husser et al. 2018", "Magnusson et al. 2017", "Maron et al. 2016", 
            "Moon et al. 2020", "Codd et al. 1989", "Miura et al. 2002", 
            "Lannou et al. 2020", "Bagger et al. 1984"             )) %>%
ggplot( aes(x=Reference,y=Prevalence, colour=as.factor(colour))) +
        geom_point(position = position_dodge(width = 1)) +
        geom_errorbar(aes(ymax = uci, ymin = lci), size=1) +
    coord_flip() +
        ylab("Prevalence and 95% CI") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=16), axis.title = element_text(size=16)) +
        scale_color_manual(values = c("black", "red", "steelblue2"))+
  annotate("text", x=22, y=0.035, label= "Coding system", size=6) + 
  annotate("text", x=14, y=0.035, label= "Imaging", size=6, colour="red") +
  annotate("text", x=10, y=0.035, label= "Selection bias", size=6, colour="steelblue2")

b <- hcm %>%
   mutate(Reference = fct_relevel(Reference, 
            "Aronow et al. 1988", "Maro et al. 2006", "Massera et al. 2019", 
            "Shapiro et al. 1983", "Maron et al. 1999", "Hada et al. 1987", 
            "Basavarajaiah et al. 2008", "Corrado et al. 1998",
            "Nistri et al. 2003", "Ng et al. 2011",  
            
            "Maron et al. 2004", "de Marvao et al. 2021", "Maron et al. 1995", 
            "Zou et al. 2004", 
            
            "Husser et al. 2018", "Magnusson et al. 2017", "Maron et al. 2016", 
            "Moon et al. 2020", "Codd et al. 1989", "Miura et al. 2002", 
            "Lannou et al. 2020", "Bagger et al. 1984"             )) %>%
ggplot( aes(x=Reference,y=Prevalence, colour=as.factor(colour))) +
        geom_point(position = position_dodge(width = 1)) +
        geom_errorbar(aes(ymax = uci, ymin = lci), size=1) +
    coord_flip() +
        ylab("Prevalence and 95% CI") +
        theme_minimal() +
        theme(legend.position="none", axis.text.x = element_text(size=16), axis.title.y = element_text(size=16), axis.title.x = element_blank(), axis.text.y=element_blank()) +
    scale_y_continuous(breaks=c(0, 0.003), limits=c(0,0.005)) +
  xlab("zoom") +
        scale_color_manual(values = c("black", "red", "steelblue2")) #+
 # annotate("text", x=22, y=0.005, label= "Coding system", size=6) + 
#  annotate("text", x=14, y=0.005, label= "Imaging", size=6, colour="red") +
#  annotate("text", x=10, y=0.005, label= "Selection Bias", size=6, colour="steelblue2")     

library(ggpubr)
ggarrange(a, b, ncol = 2, nrow = 1, widths=c(0.4,0.1), align='h')
```

### HCM meta analysis - baseline risk

```{r}
library(tidyverse)
library(meta)
library(metafor)

#https://rpubs.com/pekong/532068
#https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/metareg.html#metareg-R
#https://www.rdocumentation.org/packages/meta/versions/4.18-2/topics/forest.meta

hcm <- read.table("HCMprev.txt", header=T)
colnames(hcm) <- c("Reference", "Prevalence", "HCM", "Cohort", "Confidence_Groups")

meta <- metaprop(as.integer(HCM), Cohort, studlab=Reference, sm="PFT", data=hcm, method="Inverse", method.tau="DL")
summary(meta)
# Number of studies combined: k = 22
# 
#                      proportion           95%-CI
# Fixed effect model       0.0002 [0.0002; 0.0003]
# Random effects model     0.0007 [0.0006; 0.0009]
# 
# Quantifying heterogeneity:
#  tau^2 < 0.0001 [0.0001; 0.0011]; tau = 0.0042 [0.0078; 0.0329];
#  I^2 = 99.6% [99.6%; 99.7%]; H = 16.66 [15.78; 17.59]
# 
# Test of heterogeneity:
#        Q d.f. p-value
#  5828.02   21       0
# 
# Details on meta-analytical method:
# - Inverse variance method
# - DerSimonian-Laird estimator for tau^2
# - Jackson method for confidence interval of tau^2 and tau
# - Freeman-Tukey double arcsine transformation

forest.meta(meta, layout="RevMan5", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.08), fontsize=10, digits=6, col.square=hcm$Confidence_Groups, col.square.lines=hcm$Confidence_Groups, col.inside="black")

forest.meta(meta, layout="RevMan5", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.005), fontsize=10, digits=6, col.square=hcm$Confidence_Groups, col.square.lines=hcm$Confidence_Groups, col.inside="black")

meta2 <- metareg(meta, ~ Confidence_Groups)
meta2
# Mixed-Effects Model (k = 22; tau^2 estimator: DL)
# 
# tau^2 (estimated amount of residual heterogeneity):     0.0000 (SE = 0.0000)
# tau (square root of estimated tau^2 value):             0.0042
# I^2 (residual heterogeneity / unaccounted variability): 99.65%
# H^2 (unaccounted variability / sampling variability):   287.85
# R^2 (amount of heterogeneity accounted for):            1.13%
# 
# Test for Residual Heterogeneity:
# QE(df = 20) = 5756.9047, p-val < .0001
# 
# Test of Moderators (coefficient 2):
# QM(df = 1) = 47.7316, p-val < .0001
# 
# Model Results:
# 
#                    estimate      se    zval    pval   ci.lb   ci.ub 
# intrcpt              0.0093  0.0031  3.0414  0.0024  0.0033  0.0153   ** 
# Confidence_Groups    0.0082  0.0012  6.9088  <.0001  0.0059  0.0105  *** 
# 
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

bubble(meta2, studlab = TRUE)

###just group 2
hcm2 <- hcm[which(hcm$Confidence_Groups == "2"),]

meta <- metaprop(as.integer(HCM), Cohort, studlab=Reference, sm="PFT", data=hcm2, method="Inverse", method.tau="DL")
summary(meta)
# Number of studies combined: k = 4
# 
#                      proportion           95%-CI
# Fixed effect model       0.0018 [0.0015; 0.0022]
# Random effects model     0.0018 [0.0015; 0.0022]
# 
# Quantifying heterogeneity:
#  tau^2 = 0 [0.0000; 0.0001]; tau = 0 [0.0000; 0.0090];
#  I^2 = 0.0% [0.0%; 37.2%]; H = 1.00 [1.00; 1.26]
# 
# Test of heterogeneity:
#     Q d.f. p-value
#  0.73    3  0.8657
# 
# Details on meta-analytical method:
# - Inverse variance method
# - DerSimonian-Laird estimator for tau^2
# - Jackson method for confidence interval of tau^2 and tau
# - Freeman-Tukey double arcsine transformation

forest.meta(meta, layout="meta", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.005), fontsize=10, digits=6, col.square="black", col.square.lines="black", col.inside="black")
```

### DCM prevalence plot - baseline risk

```{r}
dcm <- read.table("DCMprev.txt", header=T)
colnames(dcm) <- c("Reference", "Prevalence", "DCM", "Cohort", "colour")
pvl <- function(a, b, p=0.5) {binom.test(a, b, 0.5, alternative=
                            c("two.sided"), conf.level = 0.95)$p.value}
lci <- function(a, b, p=0.5) {binom.test(a, b, 0.5, alternative=
                            c("two.sided"), conf.level = 0.95)$conf.int[1]}
uci <- function(a, b, p=0.5) {binom.test(a, b, 0.5, alternative=
                            c("two.sided"), conf.level = 0.95)$conf.int[2]}
dcm$p <- mapply(pvl, a=as.integer(dcm$DCM), b=dcm$Cohort)
dcm$lci <- mapply(lci, a=as.integer(dcm$DCM), b=dcm$Cohort)
dcm$uci <- mapply(uci, a=as.integer(dcm$DCM), b=dcm$Cohort)
library(ggplot2)
library(forcats)
#binom.test(1, 250, 0.5, alternative="two.sided", conf.level=0.95)
#dcm$Reference <- as.character(dcm$Reference)
#dcm[13,] <- c("1 in 250", 0.004, 0, 0, 0, 0, 0.0001012661, 0.0220838650)
dcm$Prevalence <- as.numeric(dcm$Prevalence)
dcm$Reference <- as.factor(dcm$Reference)
dcm$uci <- as.numeric(dcm$uci)
dcm$lci <- as.numeric(dcm$lci)
library(scales)
library(dplyr)

a <-dcm %>%
   mutate(Reference = fct_relevel(Reference, 
            "Haggerty et al. 2019 PMBB", "Haggerty et al. 2019 MyCode", "Rakar et al. 1997", "Torp, 1978", 
            "UK Biobank Imaging data", "Razali et al. 2018", "Pirruccello et al. 2020", 
            "Lannou et al. 2020", "Codd et al. 1989",
            "Miura et al. 2002",  
            
            "Williams and Olsen, 1985", "Bagger et al. 1984"          )) %>%
ggplot( aes(x=Reference,y=Prevalence, colour=as.factor(colour))) +
        geom_point(position = position_dodge(width = 1)) +
        geom_errorbar(aes(ymax = uci, ymin = lci), size=1) +
    coord_flip() +
        ylab("Prevalence and 95% CI") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=16), axis.title = element_text(size=16)) +
        scale_color_manual(values = c("black", "red", "steelblue2"))+
  annotate("text", x=12, y=0.035, label= "Coding system", size=6) + 
  annotate("text", x=6, y=0.035, label= "Imaging", size=6, colour="red") +
  annotate("text", x=4, y=0.035, label= "Selection bias", size=6, colour="steelblue2")

b <- dcm %>%
   mutate(Reference = fct_relevel(Reference, 
            "Haggerty et al. 2019 PMBB", "Haggerty et al. 2019 MyCode", "Rakar et al. 1997", "Torp, 1978", 
            "UK Biobank Imaging data", "Razali et al. 2018", "Pirruccello et al. 2020", 
            "Lannou et al. 2020", "Codd et al. 1989",
            "Miura et al. 2002",  
            
            "Williams and Olsen, 1985", "Bagger et al. 1984"          )) %>%
ggplot( aes(x=Reference,y=Prevalence, colour=as.factor(colour))) +
        geom_point(position = position_dodge(width = 1)) +
        geom_errorbar(aes(ymax = uci, ymin = lci), size=1) +
    coord_flip() +
        ylab("Prevalence and 95% CI") +
        theme_minimal() +
        theme(legend.position="none", axis.text.x = element_text(size=16), axis.title.y = element_text(size=16), axis.title.x = element_blank(), axis.text.y=element_blank()) +
    scale_y_continuous(breaks=c(0, 0.005), limits=c(0,0.008)) +
  xlab("zoom") +
        scale_color_manual(values = c("black", "red", "steelblue2")) #+
 # annotate("text", x=22, y=0.005, label= "Coding system", size=6) + 
#  annotate("text", x=14, y=0.005, label= "Imaging", size=6, colour="red") +
#  annotate("text", x=10, y=0.005, label= "Selection Bias", size=6, colour="steelblue2")     

library(ggpubr)
ggarrange(a, b, ncol = 2, nrow = 1, widths=c(0.4,0.1), align='h')
```

### DCM meta analysis - baseline risk

```{r}
library(tidyverse)
library(meta)
library(metafor)

#https://rpubs.com/pekong/532068
#https://bookdown.org/MathiasHarrer/Doing_Meta_Analysis_in_R/metareg.html#metareg-R
#https://www.rdocumentation.org/packages/meta/versions/4.18-2/topics/forest.meta

dcm <- read.table("DCMprev.txt", header=T)
colnames(dcm) <- c("Reference", "Prevalence", "dcm", "Cohort", "Confidence_Groups")

meta <- metaprop(as.integer(dcm), Cohort, studlab=Reference, sm="PFT", data=dcm, method="Inverse", method.tau="DL")
summary(meta)
# Number of studies combined: k = 12
# 
#                      proportion           95%-CI
# Fixed effect model       0.0001 [0.0001; 0.0001]
# Random effects model     0.0026 [0.0015; 0.0040]
# 
# Quantifying heterogeneity:
#  tau^2 = 0.0004 [0.0006; 0.0060]; tau = 0.0204 [0.0243; 0.0776];
#  I^2 = 99.8% [99.8%; 99.8%]; H = 24.28 [22.93; 25.71]
# 
# Test of heterogeneity:
#        Q d.f. p-value
#  6486.98   11       0
# 
# Details on meta-analytical method:
# - Inverse variance method
# - DerSimonian-Laird estimator for tau^2
# - Jackson method for confidence interval of tau^2 and tau
# - Freeman-Tukey double arcsine transformation

forest.meta(meta, layout="RevMan5", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.08), fontsize=10, digits=6, col.square=dcm$Confidence_Groups, col.square.lines=dcm$Confidence_Groups, col.inside="black")

forest.meta(meta, layout="RevMan5", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.005), fontsize=10, digits=6, col.square=dcm$Confidence_Groups, col.square.lines=dcm$Confidence_Groups, col.inside="black")

meta2 <- metareg(meta, ~ Confidence_Groups)
meta2
# Mixed-Effects Model (k = 12; tau^2 estimator: DL)
# 
# tau^2 (estimated amount of residual heterogeneity):     0.0003 (SE = 0.0003)
# tau (square root of estimated tau^2 value):             0.0185
# I^2 (residual heterogeneity / unaccounted variability): 99.81%
# H^2 (unaccounted variability / sampling variability):   528.40
# R^2 (amount of heterogeneity accounted for):            17.52%
# 
# Test for Residual Heterogeneity:
# QE(df = 10) = 5284.0077, p-val < .0001
# 
# Test of Moderators (coefficient 2):
# QM(df = 1) = 40.7404, p-val < .0001
# 
# Model Results:
# 
#                    estimate      se     zval    pval    ci.lb    ci.ub 
# intrcpt             -0.0667  0.0193  -3.4611  0.0005  -0.1045  -0.0289  *** 
# Confidence_Groups    0.0563  0.0088   6.3828  <.0001   0.0390   0.0736  *** 
# 
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1   

bubble(meta2, studlab = TRUE)

###just group 2
dcm2 <- dcm[which(dcm$Confidence_Groups == "1"),]

meta <- metaprop(as.integer(dcm), Cohort, studlab=Reference, sm="PFT", data=dcm2, method="Inverse", method.tau="DL")
summary(meta)
# Number of studies combined: k = 2
# 
#                      proportion           95%-CI
# Fixed effect model       0.0042 [0.0036; 0.0048]
# Random effects model     0.0022 [0.0000; 0.0076]
# 
# Quantifying heterogeneity:
#  tau^2 = 0.0007; tau = 0.0274; I^2 = 91.9% [71.9%; 97.7%]; H = 3.51 [1.89; 6.53]
# 
# Test of heterogeneity:
#      Q d.f. p-value
#  12.32    1  0.0004
# 
# Details on meta-analytical method:
# - Inverse variance method
# - DerSimonian-Laird estimator for tau^2
# - Freeman-Tukey double arcsine transformation

forest.meta(meta, layout="RevMan5", xlab="Proportion", comb.r=T, comb.f=F, xlim = c(0,0.005), fontsize=10, digits=6, col.square="black", col.square.lines="black", col.inside="black")
```

### 1 - Script_R_1_Solution_n_x_Binomial_prop --> using meta analysis p and CI, get n + x via Agresti-Couli method (error is best)

```{r}
rm(list = ls())
set.seed(28061971)
digits <- 6   # ***

# Setup #

alpha <- 0.05  # ***
z <- qnorm(1 - alpha /2)

# Input from meta-analysis #
#0.001841 (95%CI=0.001492-0.002225) 
phat <- 0.001841   # From meta-analysis results
qhat <- 1 - phat
CI_phat_LL <- 0.001492   # From meta-analysis results
CI_phat_UL <- 0.002225   # From meta-analysis results

# Storing #

Table_Methods <- matrix(NA, 4, 2)
rownames(Table_Methods) <- c("Wald method", "Arcsine method", "Agresti–Coull method", "Clopper–Pearson method")
colnames(Table_Methods) <- c("Est. x", "Est. n")

Table_Methods_Error <- matrix(NA, 4, 10) 
rownames(Table_Methods_Error) <- c("Wald method", "Arcsine method", "Agresti–Coull method", "Clopper–Pearson method")
colnames(Table_Methods_Error) <- c("MA phat", "Est. phat", "Rel. Error (%) phat",
                                   "MA CI(phat) LL", "Est. CI(phat) LL", "Rel. Error (%) CI(phat) LL",
                                   "MA CI(phat) UL", "Est. CI(phat) UL", "Rel. Error (%) CI(phat) UL", 
                                   "Distance")

# Wald method #

sigma2hat <- ((CI_phat_UL - CI_phat_LL) / (2 * z)) ^2

n <- ceiling(phat * qhat / sigma2hat)
x <- ceiling(n * phat)
pest <- x / n
qest <- 1 - pest

print("*** Wald method ***")
c("Wald method estimated n:", n, "Wald method estimated x:", x)

Table_Methods[1, 1] <- x
Table_Methods[1, 2] <- n

# Wald method: Error (%) #

print("*** Wald method Error (%) ***")
c("Meta-analysis phat:", phat, "Wald method phat:", round(pest, digits = digits))
c("Real. Error (%) Wald method phat:", round((phat - pest) / phat * 100, digits = digits /2))

c("Meta-analysis CI(phat) lower:", CI_phat_LL, "Wald method CI(phat) lower:", round(pest - z * sqrt(pest * qest / n), digits = digits))
c("Real. Error (%) Wald method CI(phat) lower:", round((CI_phat_LL - (pest - z * sqrt(pest * qest / n))) / CI_phat_LL * 100, digits = digits /2))

c("Meta-analysis CI(phat) upper:", CI_phat_UL, "Wald method CI(phat) upper:", round(pest + z * sqrt(pest * qest / n), digits = digits))
c("Real. Error (%) Wald method CI(phat) upper:", round((CI_phat_UL - (pest + z * sqrt(pest * qest / n))) / CI_phat_UL * 100, digits = digits /2))

Table_Methods_Error[1, 1] <- phat
Table_Methods_Error[1, 2] <- round(pest, digits = digits)
Table_Methods_Error[1, 3] <- round((phat - pest) / phat * 100, digits = digits /2)
Table_Methods_Error[1, 4] <- CI_phat_LL
Table_Methods_Error[1, 5] <- round(pest - z * sqrt(pest * qest / n), digits = digits)
Table_Methods_Error[1, 6] <- round((CI_phat_LL - (pest - z * sqrt(pest * qest / n))) / CI_phat_LL * 100, digits = digits /2)
Table_Methods_Error[1, 7] <- CI_phat_UL
Table_Methods_Error[1, 8] <- round(pest + z * sqrt(pest * qest / n), digits = digits)
Table_Methods_Error[1, 9] <- round((CI_phat_UL - (pest + z * sqrt(pest * qest / n))) / CI_phat_UL * 100, digits = digits /2)
Table_Methods_Error[1, 10] <- round(sqrt((phat - pest) ^2 + 
                                         (CI_phat_LL - (pest - z * sqrt(pest * qest / n))) ^2 + 
                                         (CI_phat_UL - (pest + z * sqrt(pest * qest / n))) ^2), digits = digits)

# Arcsine method #

n <- ceiling((z / (2 * (asin(sqrt(CI_phat_LL)) - asin(sqrt(phat))))) ^2)
x <- ceiling(n * phat)
pest <- x / n
qest <- 1 - pest

print("*** Arcsine method ***")
c("Arcsine method estimated n:", n, "Arcsine method estimated x:", x)

Table_Methods[2, 1] <- x
Table_Methods[2, 2] <- n

# Arcsine method: Error (%) #

print("*** Arcsine method Error (%) ***")
c("Meta-analysis phat:", phat, "Arcsine method phat:", round(pest, digits = digits))
c("Real. Error (%) Arcsine method phat:", round((phat - pest) / phat * 100, digits = digits /2))

c("Meta-analysis CI(phat) lower:", CI_phat_LL, "Arcsine method CI(phat) lower:", round(sin(asin(sqrt(pest) - z / (2 * sqrt(n)))) ^2, digits = digits))
c("Real. Error (%) Arcsine method CI(phat) lower:", round((CI_phat_LL - (sin(asin(sqrt(pest) - z / (2 * sqrt(n)))) ^2)) / CI_phat_LL * 100, digits = digits /2))

c("Meta-analysis CI(phat) upper:", CI_phat_UL, "Arcsine method CI(phat) upper:", round(sin(asin(sqrt(pest) + z / (2 * sqrt(n)))) ^2, digits = digits))
c("Real. Error (%) Arcsine method CI(phat) upper:", round((CI_phat_UL - (sin(asin(sqrt(pest) + z / (2 * sqrt(n)))) ^2)) / CI_phat_UL * 100, digits = digits /2))

Table_Methods_Error[2, 1] <- phat
Table_Methods_Error[2, 2] <- round(pest, digits = digits)
Table_Methods_Error[2, 3] <- round((phat - pest) / phat * 100, digits = digits /2)
Table_Methods_Error[2, 4] <- CI_phat_LL
Table_Methods_Error[2, 5] <- round(sin(asin(sqrt(pest) - z / (2 * sqrt(n)))) ^2, digits = digits)
Table_Methods_Error[2, 6] <- round((CI_phat_LL - (sin(asin(sqrt(pest) - z / (2 * sqrt(n)))) ^2)) / CI_phat_LL * 100, digits = digits /2)
Table_Methods_Error[2, 7] <- CI_phat_UL
Table_Methods_Error[2, 8] <- round(sin(asin(sqrt(pest) + z / (2 * sqrt(n)))) ^2, digits = digits)
Table_Methods_Error[2, 9] <- round((CI_phat_UL - (sin(asin(sqrt(pest) + z / (2 * sqrt(n)))) ^2)) / CI_phat_UL * 100, digits = digits /2)
Table_Methods_Error[2, 10] <- round(sqrt((phat - pest) ^2 + 
                                         (CI_phat_LL - (sin(asin(sqrt(pest) - z / (2 * sqrt(n)))) ^2)) ^2 + 
                                         (CI_phat_UL - (sin(asin(sqrt(pest) + z / (2 * sqrt(n)))) ^2)) ^2), digits = digits)

# Agresti–Coull method #

nStore <- NULL
xStore <- NULL

dif <- 1
C0 <- 1e-6   # ***
iter <- 1

while (dif > 0)
{
    ptilda <- phat + C0 * (iter - 1)
    qtilda <- 1 - ptilda
    
    n <- ceiling(ptilda * qtilda / ((ptilda - CI_phat_LL) / z) ^2 - z ^2)
    ntilda <- n + z ^2
    xtilda <- ntilda * ptilda
    x <- ceiling(xtilda - z ^2 /2)
    pest <- x / n
    
    nStore <- c(nStore, n)
    xStore <- c(xStore, x)
    
    dif <- phat - pest
    
    print(c("Iter:", iter))
    print(c("phat:", round(phat, digits = digits), "pest:", round(pest, digits = digits)))
    print(c("ptilda:", round((x + z ^2 /2) / (n + z ^2), digits = digits), "ptildaest:", round((phat + z ^2 / (2 * n)) / (1 + z ^2 / n), digits = digits)))
    
    print(c("Agresti–Coull method estimated n:", n, "Agresti–Coull method estimated x:", x))
    print(c("Agresti–Coull method estimated ntilda:", round(ntilda, digits = digits), "Agresti–Coull method estimated xtilda:", round(xtilda, digits = digits)))
    
    iter <- iter + 1
}

print("*** Agresti–Coull method ***")
c("Agresti–Coull method n:", n, "Agresti–Coull method x:", x)
c("Agresti–Coull method ntilda:", round(ntilda, digits = digits), "Agresti–Coull method xtilda:", round(xtilda, digits = digits))

Table_Methods[3, 1] <- x
Table_Methods[3, 2] <- n

# Agresti–Coull method: Error (%) #

print("*** Agresti–Coull method Error (%) ***")
c("Meta-analysis phat:", phat, "Agresti–Coull method phat:", round(pest, digits = digits))
c("Real. Error (%) Agresti–Coull method phat:", round((phat - pest) / phat * 100, digits = digits /2))

c("Meta-analysis CI(phat) lower:", CI_phat_LL, "Agresti–Coull method CI(phat) lower:", round(xtilda / ntilda - z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda), digits = digits))
c("Real. Error (%) Agresti–Coull method CI(phat) lower:", round((CI_phat_LL - (xtilda / ntilda - z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda))) / CI_phat_LL * 100, digits = digits /2))

c("Meta-analysis CI(phat) upper:", CI_phat_UL, "Agresti–Coull method CI(phat) upper:", round(xtilda / ntilda + z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda), digits = digits))
c("Real. Error (%) Agresti–Coull method CI(phat) upper:", round((CI_phat_UL - (xtilda / ntilda + z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda))) / CI_phat_UL * 100, digits = digits /2))

Table_Methods_Error[3, 1] <- phat
Table_Methods_Error[3, 2] <- round(pest, digits = digits)
Table_Methods_Error[3, 3] <- round((phat - pest) / phat * 100, digits = digits /2)
Table_Methods_Error[3, 4] <- CI_phat_LL
Table_Methods_Error[3, 5] <- round(xtilda / ntilda - z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda), digits = digits)
Table_Methods_Error[3, 6] <- round((CI_phat_LL - (xtilda / ntilda - z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda))) / CI_phat_LL * 100, digits = digits /2)
Table_Methods_Error[3, 7] <- CI_phat_UL
Table_Methods_Error[3, 8] <- round(xtilda / ntilda + z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda), digits = digits)
Table_Methods_Error[3, 9] <- round((CI_phat_UL - (xtilda / ntilda + z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda))) / CI_phat_UL * 100, digits = digits /2)
Table_Methods_Error[3, 10] <- round(sqrt((phat - pest) ^2 + 
                                         (CI_phat_LL - (xtilda / ntilda - z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda))) ^2 + 
                                         (CI_phat_UL - (xtilda / ntilda + z * sqrt((xtilda / ntilda) * (1 - xtilda / ntilda) / ntilda))) ^2), digits = digits)

# Clopper–Pearson method # takes a long time to run

magnitude <- 10 ^(floor(log10(n)))
n_min <- floor(n / magnitude) * magnitude
n_max <- ceiling(n / magnitude) * magnitude
q_LLStore <- NULL
q_ULStore <- NULL
nStore <- NULL
xStore <- NULL

n_min <- 55000 #added this to speed it up, knowing the answer
n_max <- 55200 #added this to speed it up, knowing the answer

for (n in n_min : n_max)
   {
    
    for (x in 1 : ceiling(2 * n * phat))
    {
        q_LL <- qbeta(alpha /2, x, n - x + 1)
        q_UL <- qbeta(1 - alpha /2, x + 1, n - x)
        q_LLStore <- c(q_LLStore, qbeta(alpha /2, x, n - x + 1))
        q_ULStore <- c(q_ULStore, qbeta(1 - alpha /2, x + 1, n - x))
        nStore <- c(nStore, n)
        xStore <- c(xStore, x)
    }
    
    if (n %% 100 == 0)
    {
        print(c("n:", n))
    }

}

dist <- sqrt((CI_phat_LL - q_LLStore) ^2 + (CI_phat_UL - q_ULStore) ^2)
idx <- which.min(dist)
n <- nStore[idx]
x <- xStore[idx]
pest <- x / n

print("*** Clopper–Pearson method ***")
c("Clopper–Pearson method n:", n, "Clopper–Pearson method x:", x)

Table_Methods[4, 1] <- x
Table_Methods[4, 2] <- n

# Clopper–Pearson method: Error (%) #

print("*** Clopper–Pearson method Error (%) ***")
c("Meta-analysis phat:", phat, "Clopper–Pearson method phat:", round(pest, digits = digits))
c("Real. Error (%) Clopper–Pearson method phat:", round((phat - pest) / phat * 100, digits = digits /2))

c("Meta-analysis CI(phat) lower:", CI_phat_LL, "Clopper–Pearson method CI(phat) lower:", round(qbeta(alpha /2, x, n - x + 1), digits = digits))
c("Real. Error (%) Clopper–Pearson method CI(phat) lower:", round((CI_phat_LL - qbeta(alpha /2, x, n - x + 1)) / CI_phat_LL * 100, digits = digits /2))

c("Meta-analysis CI(phat) upper:", CI_phat_UL, "Clopper–Pearson method CI(phat) upper:", round(qbeta(1 - alpha /2, x + 1, n - x), digits = digits))
c("Real. Error (%) Clopper–Pearson method CI(phat) upper:", round((CI_phat_UL - qbeta(1 - alpha /2, x + 1, n - x)) / CI_phat_UL * 100, digits = digits /2))

Table_Methods_Error[4, 1] <- phat
Table_Methods_Error[4, 2] <- round(pest, digits = digits)
Table_Methods_Error[4, 3] <- round((phat - pest) / phat * 100, digits = digits /2)
Table_Methods_Error[4, 4] <- CI_phat_LL
Table_Methods_Error[4, 5] <- round(qbeta(alpha /2, x, n - x + 1), digits = digits)
Table_Methods_Error[4, 6] <- round((CI_phat_LL - qbeta(alpha /2, x, n - x + 1)) / CI_phat_LL * 100, digits = digits /2)
Table_Methods_Error[4, 7] <- CI_phat_UL
Table_Methods_Error[4, 8] <- round(qbeta(1 - alpha /2, x + 1, n - x), digits = digits)
Table_Methods_Error[4, 9] <- round((CI_phat_UL - qbeta(1 - alpha /2, x + 1, n - x)) / CI_phat_UL * 100, digits = digits /2)
Table_Methods_Error[4, 10] <- round(sqrt((phat - pest) ^2 + 
                                         (CI_phat_LL - (qbeta(alpha /2, x, n - x + 1))) ^2 + 
                                         (CI_phat_UL - (qbeta(1 - alpha /2, x + 1, n - x))) ^2), digits = digits)

# Summary Tables #

print("*** All methods estimated x and n ***")
print(Table_Methods)
#                       Est. x Est. n
#Wald method                97  52554
#Arcsine method             97  52328
#Agresti–Coull method       97  52660
#Clopper–Pearson method     101  55146

print("*** All methods relative error (%) ***")
print(Table_Methods_Error)
#                        MA phat Est. phat Rel. Error (%) phat MA CI(phat) LL Est. CI(phat) LL Rel. Error (%) CI(phat) LL MA CI(phat) UL
# Wald method            0.001841  0.001846              -0.256       0.001492         0.001479                      0.888       0.002225
# Arcsine method         0.001841  0.001854              -0.689       0.001492         0.001503                     -0.747       0.002225
# Agresti–Coull method   0.001841  0.001842              -0.055       0.001492         0.001492                      0.000       0.002225
# Clopper–Pearson method 0.001841  0.001832               0.516       0.001492         0.001492                     -0.002       0.002225
#                        Est. CI(phat) UL Rel. Error (%) CI(phat) UL Distance
# Wald method                    0.002213                      0.553  1.9e-05
# Arcsine method                 0.002241                     -0.716  2.3e-05
# Agresti–Coull method           0.002228                     -0.135  3.0e-06
# Clopper–Pearson method         0.002225                      0.000  9.0e-06
```

### 2 - Script_R_2_Approx_RR_AR_Binomial_props --> Use meta analysis + agresti couli data. Approx Penetrance p(DgA) + CI (estimate!!) using relative risk as model

```{r}
rm(list = ls())
set.seed(28061971)
digits <- 6   # ***

# Loading libraries #

library("ratesci")
library("plotrix")

# Setup #

alpha <- 0.05   # ***

# p_D <-     # Baseline risk (the lifetime risk of the disease in the general population)
             # This is prevalence of HCM in UK Biobank cohort. We can't get lifetime risk so we use prevalence as a proxy

# p_AgD <-   # Allele frequency in cases
             # This is the number of times the allele is seen / the number of times the allele is counted, i.e.,  where 
             # an allele is measured twice in each individual (everyone has two copies - so 10k people were measured for 
             # 20k alleles)

# p_A <-     # Allele frequency in population controls
             # This is the number of times the allele is seen / the number of times the allele is counted, i.e.,  where 
             # an allele is measured twice in each individual (everyone has two copies - so 125k people were measured for 
             # 250k alleles)

# Input #

#0.001841 (95%CI=0.001492-0.002225)
#97	52660
x_D <- 97          # From Agresti–Coull method estimated x
n_D <- 52660       # From Agresti–Coull method estimated n

x_A <- 3           # Input example provided by Kathryn
n_A <- 600000      # Input example provided by Kathryn

x_AgD <- 10        # Input example provided by Kathryn
n_AgD <- 20000     # Input example provided by Kathryn

# Input from meta-analysis #

phat <- 0.001841   # From meta-analysis results
CI_phat_LL <- 0.001492   # From meta-analysis results
CI_phat_UL <- 0.002225   # From meta-analysis results

p_D <- x_D / n_D
p_AgD <- x_AgD / n_AgD
p_A <- x_A / n_A

#We need also to think about the degeneracy correction (if we correct) (now in the code, x_d = x +dx and n_d = n + dn). If we think we can have only degeneracy because any probabilities could be = 0, then dx=dn=0.5 is fine. However, if we have also p=1, the choice dx=0.5 and dn=1 is better.

d <- 0.5   # ***
p_D_wod <- (x_D + d) / (n_D + d)
p_AgD_wod <- (x_AgD + d) / (n_AgD + d)
p_A_wod <- (x_A + d) / (n_A + d)

# Printing #

print("*** P(D), P(A|D) and P(A) ***")
print(c("p D:", round(p_D, digits = digits)))
print(c("p AgD:", round(p_AgD, digits = digits)))
print(c("p A:", round(p_A, digits = digits)))

# Storing #

PE_CI <- matrix(0, 10, 3)
rownames(PE_CI) <- c("DM log(RR)", "DM log(RR) w/o deg.", 
                     "moverci RR", "moverci RR cc", 
                     "moverbci RR", "moverbci RR cc", 
                     "scasci RR", "scasci RR cc", 
                     "DM log(Penetrance)", "DM log(Penetrance) w/o deg.")
colnames(PE_CI) <- c("Penetrance hat", "CI(Penetrance hat) LL", "CI(Penetrance hat) UL")

# Solution 1.1: Delta Method on log(Relative Risk)                          #
#                                                                           #
# - Noether, G. E. (1957). Two confidence intervals for the ratio of two    #
#   probabilities and some measures of effectiveness. Journal of the        #
#   American Statistical Association 52, 36-45.                             #

RR <- (x_AgD / n_AgD) / (x_A / n_A)
Var_log_RR <- 1 / x_AgD - 1 / n_AgD + 1 / x_A - 1 / n_A
Var_log_RR_check <- (1 / p_AgD) * (1 - p_AgD) / (n_AgD) + 
                    (1 / p_A) * (1 - p_A) / (n_A)
log_CI_RR <- c(log(RR) - qnorm(1 - alpha / 2) * sqrt(Var_log_RR), 
               log(RR) + qnorm(1 - alpha / 2) * sqrt(Var_log_RR))

PE_RR <- RR
PE_Penetrance <- p_D * PE_RR
CI_RR <- exp(log_CI_RR)
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[1, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 1.2: Delta Method on log(Relative Risk) without degeneracy       #
#                                                                           #
# - Katz, D., Baptista, J., Azen, S. P., and Pike, M. C. (1978). Obtaining  #
#   confidence intervals for the risk ratio in cohort studies. Biometrics   #
#   34, 469-474.                                                            #
# - Pettigrew, H. M., Gart, J. J., and Thomas, D. G. (1986). The bias and   #
#   higher cumulants of the logarithm of a binomial variate. Biometrika 73, #
#   425-435.                                                                #

RR <- ((x_AgD + d) / (n_AgD + d)) / ((x_A + d) / (n_A + d))
Var_log_RR <- 1 / (x_AgD + d) - 1 / (n_AgD + d) + 1 / (x_A + d) - 1 / (n_A + d)
Var_log_RR_check <- 1 / (x_AgD + d) - 1 / (n_AgD + d) + 1 / (x_A + d) - 1 / (n_A + d)
log_CI_RR <- c(log(RR) - qnorm(1 - alpha / 2) * sqrt(Var_log_RR), 
               log(RR) + qnorm(1 - alpha / 2) * sqrt(Var_log_RR))

PE_RR <- RR
PE_Penetrance <- p_D * PE_RR
CI_RR <- exp(log_CI_RR)
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[2, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 2.1: Relative Risk as Ratio of Binomial proportions              #
#                                                                           #
# "MOVER" confidence intervals for comparisons of independent binomial or   #
# Poisson rates                                                             #
#                                                                           #
# - Laud, P. J. (2017). Equal-tailed confidence intervals for comparison of #
#   rates. Pharmaceutical Statistics. 16, 334-348.                          #
# - Donner, A., and Zou, G. Y. (2012). Closed-form confidence intervals for #
#   functions of the normal mean and standard deviation. Statistical Methods#
#   in Medical Research. 21, 347-359.                                       #
# - Newcombe, R. G. (1998). Interval estimation for the difference between  #
#   independent proportions: comparison of eleven methods. Statistics in    #
#   Medicine. 17, 857-872.                                                  #

res <- moverci(x1 = x_AgD, n1 = n_AgD, x2 = x_A, n2 = n_A, 
               distrib = "bin", contrast = "RR", type = "exact", level = 1 - alpha, cc = FALSE)

PE_RR <- res[2]
PE_Penetrance <- p_D * PE_RR
CI_RR <- res[c(1, 3)]
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[3, ] = c(PE_Penetrance, CI_Penetrance)

res <- moverci(x1 = x_AgD, n1 = n_AgD, x2 = x_A, n2 = n_A, 
               distrib = "bin", contrast = "RR", type = "exact", level = 1 - alpha, cc = TRUE)

PE_RR <- res[2]
PE_Penetrance <- p_D * PE_RR
CI_RR <- res[c(1, 3)]
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[4, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 2.2: Relative Risk as Ratio of Binomial proportions              #
#                                                                           #
# Approximate Bayesian ("MOVER-B") confidence intervals for comparisons of  #
# independent binomial or Poisson rates                                     #
#                                                                           #
# - Laud, P. J. (2017). Equal-tailed confidence intervals for comparison of #
#   rates. Pharmaceutical Statistics. 16, 334-348.                          #
# - Donner, A., and Zou, G. Y. (2012). Closed-form confidence intervals for #
#   functions of the normal mean and standard deviation. Statistical Methods#
#   in Medical Research. 21, 347-359.                                       #
# - Li, Y., Koval, J. J., Donner, A., and Zou, G.Y. (2010). Interval        #
#   estimation for the area under the receiver operating characteristic     #
#   curve when data are subject to error. Statistics in Medicine. 29, 2521– #
#   2531.                                                                   #
# - Newcombe, R. G. (1998). Interval estimation for the difference between  #
#   independent proportions: Comparison of eleven methods. Statistics in    #
#   Medicine. 17, 857-872.                                                  #

res <- moverbci(x1 = x_AgD, n1 = n_AgD, x2 = x_A, n2 = n_A, 
                a1 = 0.5, b1 = 0.5, a2 = 0.5, b2 = 0.5, 
                distrib = "bin", contrast = "RR", level = 1 - alpha, cc = FALSE)

PE_RR <- res[2]
PE_Penetrance <- p_D * PE_RR
CI_RR <- res[c(1, 3)]
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[5, ] = c(PE_Penetrance, CI_Penetrance)

res <- moverbci(x1 = x_AgD, n1 = n_AgD, x2 = x_A, n2 = n_A, 
                a1 = 0.5, b1 = 0.5, a2 = 0.5, b2 = 0.5, 
                distrib = "bin", contrast = "RR", level = 1 - alpha, cc = TRUE)

PE_RR <- res[2]
PE_Penetrance <- p_D * PE_RR
CI_RR <- res[c(1, 3)]
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[6, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 3.1: Relative Risk as Ratio of Binomial proportions              #
#                                                                           #
# Skewness-corrected asymptotic score ("SCAS") confidence intervals for     #
# comparisons of independent binomial or Poisson rates                      #
#                                                                           #
# - Laud, P. J., and Dane, A. (2014). Confidence intervals for the          #
#   difference between independent binomial proportions: Comparison using a #
#   graphical approach and moving averages. Pharmaceutical Statistics. 13,  #
#   294–308.                                                                #
# - Gart, J.J., and Nam, J. M. (1990). Approximate interval estimation of   #
#   the difference in binomial parameters: Correction for skewness and      #
#   extension to multiple tables. Biometrics. 46, 637-643.                  #
# - Gart, J.J., and Nam, J. M. (1988). Approximate interval estimation of   #
#   the ratio of binomial parameters: A review and corrections for skewness.#
#   Biometrics, 44, 323-338.                                                #

res <- scasci(x1 = x_AgD, n1 = n_AgD, x2 = x_A, n2 = n_A, 
              distrib = "bin", contrast = "RR", level = 1 - alpha, cc = FALSE)

PE_RR <- res$estimates[2]
PE_Penetrance <- p_D * PE_RR
CI_RR <- res$estimates[c(1, 3)]
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[7, ] = c(PE_Penetrance, CI_Penetrance)

res <- scasci(x1 = x_AgD, n1 = n_AgD, x2 = x_A, n2 = n_A, 
              distrib = "bin", contrast = "RR", level = 1 - alpha, cc = TRUE)

PE_RR <- res$estimates[2]
PE_Penetrance <- p_D * PE_RR
CI_RR <- res$estimates[c(1, 3)]
CI_Penetrance <- p_D * CI_RR
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[8, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 4.1: Delta Method on log(Absolute Risk)                          #

log_Penetrance <- log(p_D * p_AgD / p_A)
Var_log_Penetrance <- (1 / p_D) * (1 - p_D) / n_D + 
              (1 / p_AgD) * (1 - p_AgD) / n_AgD + 
              (1 / p_A) * (1 - p_A) / n_A
log_CI_Penetrance <- c(log_Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance), 
                       log_Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance))

PE_Penetrance <- exp(log_Penetrance)
CI_Penetrance <- exp(log_CI_Penetrance)
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[9, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 4.2: Delta Method on log(Absolute Risk) without degeneracy       #
#                                                                           #
# - Katz, D., Baptista, J., Azen, S. P., and Pike, M. C. (1978). Obtaining  #
#   confidence intervals for the risk ratio in cohort studies. Biometrics   #
#   34, 469-474.                                                            #
# - Pettigrew, H. M., Gart, J. J., and Thomas, D. G. (1986). The bias and   #
#   higher cumulants of the logarithm of a binomial variate. Biometrika 73, #
#   425-435.                                                                #

log_Penetrance_ <- log(p_D_wod * p_AgD_wod / p_A_wod)
Var_log_Penetrance <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
                      (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
                      (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_CI_Penetrance <- c(log_Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance), 
                       log_Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance))

PE_Penetrance <- exp(log_Penetrance)
CI_Penetrance <- exp(log_CI_Penetrance)
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[10, ] = c(PE_Penetrance, CI_Penetrance)

# Printing #

print("*** All methods estimated Absolute Risk and CI ***")
print(round(PE_CI, digits = digits))

#                             Penetrance hat CI(Penetrance hat) LL CI(Penetrance hat) UL
# DM log(RR)                        0.184201              0.050698              0.669248
# DM log(RR) w/o deg.               0.165776              0.049452              0.555725
# moverci RR                        0.177096              0.049337              0.974654
# moverci RR cc                     0.177096              0.049337              0.974654
# moverbci RR                       0.177096              0.055448              0.725399
# moverbci RR cc                    0.177096              0.049337              0.974654
# scasci RR                         0.177412              0.054657              0.794391
# scasci RR cc                      0.177412              0.046482              1.000000
# DM log(Penetrance)                0.184201              0.049932              0.679519
# DM log(Penetrance) w/o deg.       0.184201              0.054068              0.627540

# Setup plot #

cex <- 0.8
cex.axis <- 0.8
cex.lab <- 0.9

# Plotting #

df <- data.frame(PE_CI)
colnames(df)<- c("PE", "LL", "UL")
n_df <- nrow(df)

plot(df$PE, xaxt = "n", xlim = c(1, n_df), xlab = "Method", ylim = c(0, 1), ylab = paste("Est. Penetrance and", 1 - alpha, "% CI(Penetrance hat)", sep = " "),
     main = paste("P(D) =", round(p_D, digits = digits), "and P(A|D) =", round(p_AgD, digits = digits)))
plotCI(df$PE, y = NULL, uiw = df$UL - df$PE, liw = df$PE- df$LL, err="y", pch = 20, slty = 3, scol = "black", add = TRUE)
axis(1, at = 1 : n_df, labels = FALSE)
text(x = 1 : n_df, y = par()$usr[3] - 0.075 * (par()$usr[4] - par()$usr[3]), 
     labels = rownames(PE_CI), srt = 45, adj = 1, xpd = TRUE)
abline(h = CI_phat_LL, lty = 3, lwd = 1, col = "grey")
abline(h = CI_phat_UL, lty = 3, lwd = 1, col = "grey")
abline(v = 8.5, lty = 2, lwd = 2, col = "grey")

idx <- c(1, 3, 5, 7, 9)
df <- data.frame(PE_CI[idx, ])
colnames(df)<- c("PE", "LL", "UL")
n_df <- nrow(df)

plot(df$PE, xaxt = "n", xlim = c(1, n_df), xlab = "Method", ylim = c(0, 1), ylab = paste("Est. Penetrance and", 1 - alpha, "% CI(Penetrance hat)", sep = " "),
     main = paste("P(D) =", round(p_D, digits = digits), "and P(A|D) =", round(p_AgD, digits = digits)))
plotCI(df$PE, y = NULL, uiw = df$UL - df$PE, liw = df$PE- df$LL, err="y", pch = 20, slty = 3, scol = "black", add = TRUE)
axis(1, at = 1 : n_df, labels = FALSE)
text(x = 1 : n_df, y = par()$usr[3] - 0.075 * (par()$usr[4] - par()$usr[3]), 
     labels = rownames(PE_CI)[idx], srt = 45, adj = 1, xpd = TRUE)
abline(h = CI_phat_LL, lty = 3, lwd = 1, col = "grey")
abline(h = CI_phat_UL, lty = 3, lwd = 1, col = "grey")
abline(v = 8.5, lty = 2, lwd = 2, col = "grey")

idx <- c(2, 4, 6, 8, 10)
df <- data.frame(PE_CI[idx, ])
colnames(df)<- c("PE", "LL", "UL")
n_df <- nrow(df)

plot(df$PE, xaxt = "n", xlim = c(1, n_df), xlab = "Method", ylim = c(0, 1), ylab = paste("Est. Penetrance and", 1 - alpha, "% CI(Penetrance hat)", sep = " "),
     main = paste("P(D) =", round(p_D, digits = digits), "and P(A|D) =", round(p_AgD, digits = digits)))
plotCI(df$PE, y = NULL, uiw = df$UL - df$PE, liw = df$PE- df$LL, err="y", pch = 20, slty = 3, scol = "black", add = TRUE)
axis(1, at = 1 : n_df, labels = FALSE)
text(x = 1 : n_df, y = par()$usr[3] - 0.075 * (par()$usr[4] - par()$usr[3]), 
     labels = rownames(PE_CI)[idx], srt = 45, adj = 1, xpd = TRUE)
abline(h = CI_phat_LL, lty = 3, lwd = 1, col = "grey")
abline(h = CI_phat_UL, lty = 3, lwd = 1, col = "grey")
abline(v = 8.5, lty = 2, lwd = 2, col = "grey")
```

### 3 - Script_R_3_Approx_g_Binomial_props_DeltaMeth --> As with 2 but Approx Penetrance p(DgA) + CI using asymptotic. LB recommends Delta log improved mean cc method

```{r}
rm(list = ls())
set.seed(28061971)
digits <- 6   # ***

# Loading libraries #

library("plotrix")

# Setup #

alpha <- 0.05   # ***

# p_D <-     # Baseline risk (the lifetime risk of the disease in the general population)
             # This is prevalence of HCM in UK Biobank cohort. We can't get lifetime risk so we use prevalence as a proxy

# p_AgD <-   # Allele frequency in cases
             # This is the number of times the allele is seen / the number of times the allele is counted, i.e.,  where 
             # an allele is measured twice in each individual (everyone has two copies - so 10k people were measured for 
             # 20k alleles)

# p_A <-     # Allele frequency in population controls
             # This is the number of times the allele is seen / the number of times the allele is counted, i.e.,  where 
             # an allele is measured twice in each individual (everyone has two copies - so 125k people were measured for 
             # 250k alleles)

# Input #

#0.001841 (95%CI=0.001492-0.002225)
#97	52660

x_D <- 97          # From Agresti–Coull method estimated X
n_D <- 52660       # From Agresti–Coull method estimated n

x_A <- 3           # Input example provided by Kathryn
n_A <- 600000      # Input example provided by Kathryn
  
x_AgD <- 10        # Input example provided by Kathryn
n_AgD <- 20000     # Input example provided by Kathryn

# Input from meta-analysis #

phat <- 0.001841   # From meta-analysis results
CI_phat_LL <- 0.001492   # From meta-analysis results
CI_phat_UL <- 0.002225   # From meta-analysis results

p_D <- x_D / n_D
p_AgD <- x_AgD / n_AgD
p_A <- x_A / n_A

d <- 0.5   # ***
p_D_wod <- (x_D + d) / (n_D + d)
p_AgD_wod <- (x_AgD + d) / (n_AgD + d)
p_A_wod <- (x_A + d) / (n_A + d)

# Printing #

print("*** P(D), P(A|D) and P(A) ***")
print(c("p D:", round(p_D, digits = digits)))
print(c("p AgD:", round(p_AgD, digits = digits)))
print(c("p A:", round(p_A, digits = digits)))

# Storing #

PE_CI <- matrix(0, 8, 3)
rownames(PE_CI) <- c("DM log(Penetrance)", "Delta log(Penetrance) w/o deg.", 
                     "DM log(Penetrance) impr. mean", "DM log(Penetrance) impr. mean w/o deg.", 
                     "DM Penetrance", "DM Penetrance w/o deg.", 
                     "DM Penetrance impr. mean", "DM Penetrance impr. mean w/o deg.")
colnames(PE_CI) <- c("Penetrance hat", "CI(Penetrance hat) LL", "CI(Penetrance hat) UL")

# Solution 1.1: Delta Method on log(Absolute Risk)                          #

log_Penetrance <- log(p_D * p_AgD / p_A)
Var_log_Penetrance <- (1 / p_D) * (1 - p_D) / n_D + 
                      (1 / p_AgD) * (1 - p_AgD) / n_AgD + 
                      (1 / p_A) * (1 - p_A) / n_A
log_CI_Penetrance <- c(log_Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance), 
                       log_Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance))

PE_Penetrance <- exp(log_Penetrance)
CI_Penetrance <- exp(log_CI_Penetrance)
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[1, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 1.2: Delta Method on log(Absolute Risk) without degeneracy       #

log_Penetrance_ <- log(p_D_wod * p_AgD_wod / p_A_wod)
Var_log_Penetrance <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
                      (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
                      (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_CI_Penetrance <- c(log_Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance), 
                       log_Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance))

PE_Penetrance <- exp(log_Penetrance)
CI_Penetrance <- exp(log_CI_Penetrance)
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[2, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 1.3: Delta Method on log(Absolute Risk) with improved mean      # 
#               approximation                                              #

log_Penetrance <- log(p_D * p_AgD / p_A) + 1 /2 * ((1 / p_A) * (1 - p_A) / n_A - 
                                                   (1 / p_D) * (1 - p_D) / n_D - 
                                                   (1 / p_AgD) * (1 - p_AgD) / n_AgD)
Var_log_Penetrance <- (1 / p_D) * (1 - p_D) / n_D + 
                      (1 / p_AgD) * (1 - p_AgD) / n_AgD + 
                      (1 / p_A) * (1 - p_A) / n_A
log_CI_Penetrance <- c(log_Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance), 
                       log_Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance))

PE_Penetrance <- exp(log_Penetrance)
CI_Penetrance <- exp(log_CI_Penetrance)
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[3, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 1.4: Delta Method on log(Absolute Risk) with improved mean      # 
#               approximation without degeneracy                           #

log_Penetrance <- log(p_D_wod * p_AgD_wod / p_A_wod) + 1 /2 * ((1 / p_A_wod) * (1 - p_A_wod) / (n_A + d) - 
                                                               (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) - 
                                                               (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d))
Var_log_Penetrance <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
                      (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
                      (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_CI <- c(log_Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance), 
            log_Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance))

PE_Penetrance <- exp(log_Penetrance)
CI_Penetrance <- exp(log_CI)
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[4, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 2.1: Delta Method on Absolute Risk                               #

Penetrance <- p_D * p_AgD / p_A
Var_Penetrance <- Penetrance ^2 * ((1 / p_D) * (1 - p_D) / n_D + 
                                  (1 / p_AgD) * (1 - p_AgD) / n_AgD + 
                                  (1 / p_A) * (1 - p_A) / n_A)
CI <- c(Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_Penetrance), 
        Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_Penetrance))

PE_Penetrance <- Penetrance
CI_Penetrance <- CI
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[5, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 2.2: Delta Method on Absolute Risk without degeneracy            #

Penetrance <- p_D_wod * p_AgD_wod / p_A_wod
Var_Penetrance <- Penetrance ^2 * ((1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
                                   (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
                                   (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d))
CI <- c(Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_Penetrance), 
        Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_Penetrance))

PE_Penetrance <- Penetrance
CI_Penetrance <- CI
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[6, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 2.3: Delta Method on Absolute Risk with improved mean           #
#               approximation                                              #

Penetrance <- (p_D * p_AgD / p_A) * (1 + (1 / p_A) * (1 - p_A) / n_A)
Var_Penetrance <- (p_D * p_AgD / p_A) ^2 * ((1 / p_D) * (1 - p_D) / n_D + 
                                            (1 / p_AgD) * (1 - p_AgD) / n_AgD + 
                                            (1 / p_A) * (1 - p_A) / n_A)
CI <- c(Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_Penetrance), 
        Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_Penetrance))

PE_Penetrance <- Penetrance
CI_Penetrance <- CI
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[7, ] = c(PE_Penetrance, CI_Penetrance)

# Solution 2.4: Delta Method on Absolute Risk with improved mean           #
#               approximation without degeneracy                           #

Penetrance <- (p_D_wod * p_AgD_wod / p_A_wod)  * (1 + (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d))
Var_Penetrance <- (p_D_wod * p_AgD_wod / p_A_wod) ^2 * ((1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
                                                        (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
                                                        (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d))
CI <- c(Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_Penetrance), 
        Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_Penetrance))

PE_Penetrance <- Penetrance
CI_Penetrance <- CI
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
PE_CI[8, ] = c(PE_Penetrance, CI_Penetrance)

# Printing #

print("*** Delta Methods on log(AR) and AR with Confidence Interval ***")
print(round(PE_CI, digits = digits))

#                                        Penetrance hat CI(Penetrance hat) LL CI(Penetrance hat) UL
# DM log(Penetrance)                           0.184201              0.049932              0.679519
# Delta log(Penetrance) w/o deg.               0.184201              0.054068              0.627540
# DM log(Penetrance) impr. mean                0.205937              0.055824              0.759705
# DM log(Penetrance) impr. mean w/o deg.       0.182348              0.053524              0.621228
# DM Penetrance                                0.184201              0.000000              0.424648
# DM Penetrance w/o deg.                       0.166629              0.000000              0.370881
# DM Penetrance impr. mean                     0.245600              0.005152              0.486048
# DM Penetrance impr. mean w/o deg.            0.214238              0.009986              0.418489

# Setup plot #

cex <- 0.8
cex.axis <- 0.8
cex.lab <- 0.9

# Plotting #

df <- data.frame(PE_CI)
colnames(df)<- c("PE", "LL", "UL")
n_df <- nrow(df)

plot(df$PE, xaxt = "n", xlim = c(1, n_df), xlab = "Method", ylim = c(0, 1), ylab = paste("Est. Penetrance and", 1 - alpha, "% CI(Penetrance hat)", sep = " "),
     main = paste("P(D) =", round(p_D, digits = digits), "and P(A|D) =", round(p_AgD, digits = digits)))
plotCI(df$PE, y = NULL, uiw = df$UL - df$PE, liw = df$PE- df$LL, err="y", pch = 20, slty = 3, scol = "black", add = TRUE)
axis(1, at = 1 : n_df, labels = FALSE)
text(x = 1 : n_df, y = par()$usr[3] - 0.075 * (par()$usr[4] - par()$usr[3]), 
     labels = rownames(PE_CI), srt = 45, adj = 1, xpd = TRUE, cex = cex.axis)
abline(h = CI_phat_LL, lty = 3, lwd = 1, col = "grey")
abline(h = CI_phat_UL, lty = 3, lwd = 1, col = "grey")
abline(v = 4.5, lty = 2, lwd = 2, col = "grey")

idx <- c(1, 3, 5, 7)
df <- data.frame(PE_CI[idx, ])
colnames(df)<- c("PE", "LL", "UL")
n_df <- nrow(df)

plot(df$PE, xaxt = "n", xlim = c(1, n_df), xlab = "Method", ylim = c(0, 1), ylab = paste("Est. Penetrance and", 1 - alpha, "% CI(Penetrance hat)", sep = " "),
     main = paste("P(D) =", round(p_D, digits = digits), "and P(A|D) =", round(p_AgD, digits = digits)))
plotCI(df$PE, y = NULL, uiw = df$UL - df$PE, liw = df$PE- df$LL, err="y", pch = 20, slty = 3, scol = "black", add = TRUE)
axis(1, at = 1 : n_df, labels = FALSE)
text(x = 1 : n_df, y = par()$usr[3] - 0.075 * (par()$usr[4] - par()$usr[3]), 
     labels = rownames(PE_CI)[idx], srt = 45, adj = 1, xpd = TRUE, cex = cex.axis)
abline(h = CI_phat_LL, lty = 3, lwd = 1, col = "grey")
abline(h = CI_phat_UL, lty = 3, lwd = 1, col = "grey")
abline(v = 8.5, lty = 2, lwd = 2, col = "grey")

idx <- c(2, 4, 6, 8)
df <- data.frame(PE_CI[idx, ])
colnames(df)<- c("PE", "LL", "UL")
n_df <- nrow(df)

plot(df$PE, xaxt = "n", xlim = c(1, n_df), xlab = "Method", ylim = c(0, 1), ylab = paste("Est. Penetrance and", 1 - alpha, "% CI(Penetrance hat)", sep = " "),
     main = paste("P(D) =", round(p_D, digits = digits), "and P(A|D) =", round(p_AgD, digits = digits)))
plotCI(df$PE, y = NULL, uiw = df$UL - df$PE, liw = df$PE- df$LL, err="y", pch = 20, slty = 3, scol = "black", add = TRUE)
axis(1, at = 1 : n_df, labels = FALSE)
text(x = 1 : n_df, y = par()$usr[3] - 0.075 * (par()$usr[4] - par()$usr[3]), 
     labels = rownames(PE_CI)[idx], srt = 45, adj = 1, xpd = TRUE, cex = cex.axis)
abline(h = CI_phat_LL, lty = 3, lwd = 1, col = "grey")
abline(h = CI_phat_UL, lty = 3, lwd = 1, col = "grey")
abline(v = 8.5, lty = 2, lwd = 2, col = "grey")
```

### plot - different LB methods for penetrance equations

```{r}
meth <- read.table("methods.txt", header=T)

ggplot(meth, aes(x=Method,y=AR.hat)) +
        geom_point() +
          theme_minimal() +
        geom_errorbar(aes(ymax = AR.hat.CI.UL, ymin = AR.hat.CI.LL)) +
    coord_flip() +
        ylab("Penetrance and 95% CI") +
        xlab("Method") +
        theme(axis.text = element_text(size=16), axis.title = element_text(size=16), legend.position="none") + 
  scale_x_discrete(labels = c(
   #       expression("DM "[p]*'(D) x '[p]*'(A|D) / '[p]*'(A)'),
         expression("DM "[p]*'(D) x '[p]*'(A|D) / '[p]*'(A) d'),
   #   expression("DM "[p]*'(D) x '[p]*'(A|D) / '[p]*'(A) mean approx.'),
       expression("DM "[p]*'(D) x '[p]*'(A|D) / '[p]*'(A) mean approx. d'),
        
    #    expression("DM log("[p]*'(A|D) / '[p]*'(A))'),
        expression("DM log("[p]*'(A|D) / '[p]*'(A)) d'),
    #    expression("DM log("[p]*'(D) x '[p]*'(A|D) / '[p]*'(A))'),
        expression("DM log("[p]*'(D) x '[p]*'(A|D) / '[p]*'(A)) d'),
    #    expression("DM log("[p]*'(D) x '[p]*'(A|D) / '[p]*'(A)) mean approx.'),
        expression("DM log("[p]*'(D) x '[p]*'(A|D) / '[p]*'(A)) mean approx. d'),
     
 #   expression("MOVER "[p]*'(A|D) / '[p]*'(A)'),
    expression("MOVER "[p]*'(A|D) / '[p]*'(A) cc'),
 #   expression("MOVER-B "[p]*'(A|D) / '[p]*'(A)'),
    expression("MOVER-B "[p]*'(A|D) / '[p]*'(A) cc'),
  #  expression("SCAS "[p]*'(A|D) / '[p]*'(A)'),
      expression("SCAS "[p]*'(A|D) / '[p]*'(A) cc')
    ))
```

### 4 - Script_R_4_Bayesian_way - Bayesian penetrance

```{r}
rm(list = ls())
set.seed(28061971)
digits <- 6   # ***

# Setup #

alpha <- 0.05  # ***

# Input from meta-analysis #

#0.001841 (95%CI=0.001492-0.002225)
#97	52660

phat <- 0.001841   # From meta-analysis results
pCIL <- 0.001492   # From meta-analysis results
pCIU <- 0.002225   # From meta-analysis results

# Loading libraries #

library("TailRank")
#download correct biovase version needed for HPC R version 3.6.0, then install TailRank.

# Baseline risk (Prevalence)  #

X_D <- 97   # From Agresti–Coull method estimated X
n_D <- 52660   # From Agresti–Coull method estimated n

alpha_D <- X_D + 1
beta_D <- n_D - X_D + 1

# Allele frequency in cases #

X_AgD <- 10      # Input example provided by Kathryn
n_AgD <- 20000   # Input example provided by Kathryn

# Allele frequency in general population #

X_A <- 3         # Input example provided by Kathryn
n_A <- 600000    # Input example provided by Kathryn

# Printing #

print("*** Baseline risk (Prevalence) as a Beta density ***")
c("Est. phat:", round(X_D / n_D, digits = digits),  
  "Mean baseline risk:", round(alpha_D / (alpha_D + beta_D), digits = digits), 
  "Std baseline risk:", round(sqrt((alpha_D * beta_D) / ((alpha_D + beta_D) ^2 * (alpha_D + beta_D + 1))), digits = digits))

# Penetrance (posterior distribution) #

mu_D <- alpha_D / (alpha_D + beta_D)
M_D <- alpha_D + beta_D

alpha_DgA <- X_AgD + M_D * mu_D
beta_DgA <- n_AgD - X_AgD + M_D * (1 - mu_D) - 1

# Printing #

print("*** Credible interval Penetrance ***")
c("CI(p_DgA) lower:", round(qbeta(alpha /2, alpha_DgA, beta_DgA), digits = digits),  
  "CI(p_DgA) upper:", round(qbeta(1 - alpha /2, alpha_DgA, beta_DgA), digits = digits))

# Plotting #

ys <- seq(0, 1, by = 1e-6)
f_D <- dbeta(ys, alpha_D, beta_D)
f_DgA <- dbeta(ys, alpha_DgA, beta_DgA)
legend_texts = c(  expression(""[p]*"(D)"), expression(""[p]*'(D) 95% CI'),  expression(""[p]*'(D|A)'),   expression(""[p]*'(D|A) 95% CI'))

plot(ys, f_D, xlim = c(0, 0.01), xlab = "Prior and posterior probability", 
              ylim = c(0, max(max(f_D), max(f_DgA)) * 1.1), ylab = "Density", 
              type = "l", lty = 1, lwd = 2, col = "blue",
     cex.main=1.25, cex.lab=1.5, cex.axis=1.5)
abline(v = qbeta(alpha /2, alpha_D, beta_D), lty = 2, lwd = 1, col = "blue")
abline(v = qbeta(1 - alpha /2, alpha_D, beta_D), lty = 2, lwd = 1, col = "blue")
lines(ys, f_DgA, type = "l", lty = 1, lwd = 2, col = "red")
abline(v = qbeta(alpha /2, alpha_DgA, beta_DgA), lty = 2, lwd = 1, col = "red")
abline(v = qbeta(1 - alpha /2, alpha_DgA, beta_DgA), lty = 2, lwd = 1, col = "red")
legend("center", legend = , lty = c(1, 2, 1, 2), lwd = c(2, 1, 2, 1), col = c("blue", "blue", "red", "red"), box.lwd = 0, box.col = "white", cex = 1.5)

# Plotting #

x <- 0 : (X_A * 20)
y <- dbinom(x, n_A, X_A / n_A)
yy <- dbb(x, n_AgD, alpha_D, beta_D)
legend_texts = c(
  expression("Binomial "[p]*'(A)'), 
  expression("Beta-Binomial "[p]*'(A)'))
  
barplot(t(matrix(c(y, yy), ncol = 2)), beside = TRUE, col = c("blue", "red") ,
     cex.main=1.25, cex.lab=1.5, cex.axis=1.5)
legend("center", legend = legend_texts, col = c("blue", "red"), pch = 15, box.lwd = 0, box.col = "white" , cex = 1.5)
abline(v = X_A, lty = 2, lwd = 2, col = "grey")
```

### Concl 

```{r}
rm(list = ls())
set.seed(28061971)
digits <- 6   # ***
alpha <- 0.05  # **
library("ratesci")
library("plotrix")

#HCM
#HCM meta analysis
#0.001841 [0.001492-0.002225]

phat <- 0.001841   # From meta-analysis results
CI_p_LL <- 0.001492   # From meta-analysis results
CI_p_UL <- 0.002225   # From meta-analysis results

x_D <- 97          # From Agresti–Coull method estimated X
n_D <- 52660       # From Agresti–Coull method estimated n

x_A <- 3           # Input example provided by Kathryn
n_A <- 600000      # Input example provided by Kathryn
  
x_AgD <- 10        # Input example provided by Kathryn
n_AgD <- 20000     # Input example provided by Kathryn

p_D <- x_D / n_D
p_AgD <- x_AgD / n_AgD
p_A <- x_A / n_A

d <- 0.5   # ***
p_D_wod <- (x_D + d) / (n_D + d)
p_AgD_wod <- (x_AgD + d) / (n_AgD + d)
p_A_wod <- (x_A + d) / (n_A + d)

# Solution 1.4: Delta Method on log(Absolute Risk) with improved mean      # 
#               approximation without degeneracy                           #

log_Penetrance <- log(p_D_wod * p_AgD_wod / p_A_wod) + 1 /2 * ((1 / p_A_wod) * (1 - p_A_wod) / (n_A + d) - 
                                                               (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) - 
                                                               (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d))
Var_log_Penetrance <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
                      (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
                      (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_CI <- c(log_Penetrance - qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance), 
            log_Penetrance + qnorm(1 - alpha / 2) * sqrt(Var_log_Penetrance))

PE_Penetrance <- exp(log_Penetrance)
CI_Penetrance <- exp(log_CI)
PE_Penetrance[PE_Penetrance < 0] <- 0
PE_Penetrance[PE_Penetrance > 1] <- 1
CI_Penetrance[CI_Penetrance < 0] <- 0
CI_Penetrance[CI_Penetrance > 1] <- 1
```

# testing 
### testing info

```{r}
#Need to test for samples:
#HCM_AF
#min = 0.0001
#median = 0.0003
#max = 0.008
#gnukb_AF
#min = 0.000002
#median = 0.00001
#max = 0.0009

#HCM_AF <- c(0.00005, 0.0001, 0.0002, 0.0004, 0.0005, 0.001, 0.002, 0.004, 0.008)
#HCM_AC <- c(1, 2, 4, 8, 10, 20, 40, 80, 160) in 20,000 AN = 10,000 samples
#hcman <- c(150,250,500,1000,2000,4000,6000,8000,10000,20000,40000,60000,80000,100000)
#75 - 50,000 samples

#gnukb_AF <- c(0.000002, 0.000003, 0.000007, 0.00001, 0.00002, 0.00008, 0.0002, 0.0004, 0.0009)
#gnukb_AC <- c(1, 2, 4, 8, 10, 50, 100, 250, 550) in 600,000 AN = 300,000 samples
#gnukb_AN <- c(100000, 200000, 400000, 600000, 800000, 1000000, 1200000, 1400000, 1600000, 1800000, 2000000)
#50,000 to 1,000,000 samples
```

### testing function

```{r}
#no truncation
penetrance <- function(x_D, n_D, x_AgD, n_AgD, x_A, n_A){
set.seed(28061971)
digits <- 6   
alpha <- 0.05  
p_D <- x_D / n_D
p_AgD <- x_AgD / n_AgD
p_A <- x_A / n_A
d <- 0.5   
p_D_wod <- (x_D + d) / (n_D + d)
p_AgD_wod <- (x_AgD + d) / (n_AgD + d)
p_A_wod <- (x_A + d) / (n_A + d)
log_AR <- log(p_D_wod * p_AgD_wod / p_A_wod) + 1 /2 * ((1 / p_A_wod) * (1 - p_A_wod) / (n_A + d) - 
                                                       (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) - 
                                                       (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d))
Var_log_AR <- (1 / p_D_wod) * (1 - p_D_wod) / (n_D + d) + 
              (1 / p_AgD_wod) * (1 - p_AgD_wod) / (n_AgD + d) + 
              (1 / p_A_wod) * (1 - p_A_wod) / (n_A + d)
log_LCI <- log_AR - qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
log_UCI <- log_AR + qnorm(1 - alpha / 2) * sqrt(Var_log_AR)
penetrance <- exp(log_AR) #diff
log_lCI <- exp(log_LCI) #diff
log_uCI <- exp(log_UCI) #diff
my_list <- list("penetrance" = penetrance, "lci" = log_lCI, "uci" = log_uCI)
return(my_list)
}
```

### gnukb - gnomac = 2

```{r}
#HCM_AF = 0.008 + min gnubk = 0.000002
h_af <- c(0.00005, 0.0001, 0.0002, 0.0004, 0.0005, 0.001, 0.002, 0.004, 0.008)
hcman <- c(100,500,1000,5000,10000,50000,100000,150000,200000)
#75 - 50,000 samples
done <- as.integer(unlist(lapply(1:length(h_af), function(x) lapply(1:length(hcman), function(y) as.matrix(h_af[[x]]) %*%  hcman[[y]]))))
done <- data.frame(done)
done$done <- as.integer(done$done)
done$an <- hcman
dim(done)
# 81
colnames(done) <- c("h_ac", "h_an")

done$group <- rep(c(1,2,3,4,5,6,7,8,9), each = 9)
done$samples <- done$h_an/2
done$num <- as.numeric(row.names(done))
done$group <- as.factor(done$group)
done$freq <- done$h_ac/done$h_an
done2 <- done[which(
  (done$freq == 0.00005 & done$group == "1") |
  (done$freq == 0.0001 & done$group == "2") |
  (done$freq == 0.0002 & done$group == "3") |
  (done$freq == 0.0004 & done$group == "4") |
  (done$freq == 0.0005 & done$group == "5") |
  (done$freq == 0.001 & done$group == "6") |
  (done$freq == 0.002 & done$group == "7") |
  (done$freq == 0.004 & done$group == "8") |
  (done$freq == 0.008 & done$group == "9")
  ),]
dim(done2)
#55  

gnoman <- 600000
gnomac <- 2

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

library(ggrepel)
library(ggpubr)

done2$samples <- as.factor(done2$samples)
axis <- unique(done2$freq)

a <- ggplot(data=done2, aes(x=num, y=uci, colour=(samples))) +
         geom_line(aes(x = num, y = lci, color=(samples), group=freq), size = 0.5) + 
         geom_line(aes(x = num, y = uci, color=(samples), group=freq), size = 0.5) + 
         geom_point(aes(x = num, y = lci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = uci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = penetrance, group=freq), size = 2, color="black") +
         ggtitle("Population frequency of 2/600,000") +
        ylab("Penetrance") +
        theme_minimal() +
          geom_hline(yintercept=1, color ="black", size=0.5, linetype="dashed") +
   theme(legend.position="top", axis.text = element_text(size=14), axis.title = element_text(size=16), legend.title = element_text(size=16), legend.text = element_text(size=14),
         title = element_text(size=16),
          panel.grid.minor.y = element_line( size = 0.4),
                   panel.grid.minor.x = element_blank(),
          panel.grid.major.y = element_line( size = 0.4))  +
          scale_x_continuous(breaks = seq(1, 82, 9), labels=SoilSciGuylabs)+labs(colour="Samples")+
          xlab("Variant frequency in cases")

b <- ggplot(data=done2, aes(x=num, y=uci, colour=(samples))) +
         geom_line(aes(x = num, y = lci, color=(samples), group=freq), size = 0.5) + 
         geom_line(aes(x = num, y = uci, color=(samples), group=freq), size = 0.5) + 
         geom_point(aes(x = num, y = lci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = uci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = penetrance, group=freq), size = 2, color="black") +
        ylab("Penetrance") +
        xlab("Variant frequency in cases")+
        theme_minimal() +
          geom_hline(yintercept=1, color ="black", size=0.5, linetype="dashed") +
         theme(legend.position="top", axis.text = element_text(size=14), axis.title = element_text(size=16), legend.title = element_text(size=16), legend.text = element_text(size=14),
         title = element_text(size=16),
          panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.minor.x = element_blank(),
          panel.grid.major.y = element_line( size = 0.4))  +
          scale_x_continuous(breaks = seq(1, 82, 9), labels=SoilSciGuylabs)+labs(colour="Samples")+
  ylim(0,1)

#library(ggpubr)
ggarrange(a+rremove("x.text")+rremove("x.title"), b, ncol = 1, nrow = 2, widths=c(0.1,0.1), align='v', common.legend=TRUE)

#### 4

gnomac <- 4

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

#use same ggplot code but change title in a

#### 8

gnomac <- 8

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

#### 10

gnomac <- 10

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

#### 50

gnomac <- 50

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

#ylim is 0-1
a

#### 100

gnomac <- 100

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

a

#### 250

gnomac <- 250

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

a

#### 550

gnomac <- 550

#format( 9.19143e-06, scientific = FALSE)
#AF = 0.00000919143

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac, n_AgD=done2$h_an, x_A=gnomac, n_A=gnoman)
SoilSciGuylabs <- c("", unique(done2$freq))

a
```

### gnukb testing - HCMAC = 2

```{r}
rm(done)
rm(done2)
gnukb_AF <- c(0.000002, 0.000003, 0.000007, 0.00001, 0.00002, 0.00008, 0.0002, 0.0004, 0.0009)
gnukb_AN <- c(50000, 100000, 500000, 1000000, 5000000, 10000000)
#50,000 to 1,000,000 samples
done <- as.integer(unlist(lapply(1:length(gnukb_AF), function(x) lapply(1:length(gnukb_AN), function(y) as.matrix(gnukb_AF[[x]]) %*%  gnukb_AN[[y]]))))
done <- data.frame(done)
done$done <- as.integer(done$done)
done$an <- gnukb_AN
dim(done)
# 54
colnames(done) <- c("gnukb_AC", "gnukb_AN")

done$group <- rep(c(1,2,3,4,5,6,7,8,9), each = 6)
done$samples <- done$gnukb_AN/2
done$num <- as.numeric(row.names(done))
done$group <- as.factor(done$group)
done$freq <- done$gnukb_AC/done$gnukb_AN
done2 <- done[which(
  (done$freq == 0.000002 & done$group == "1") |
  (done$freq == 0.000003 & done$group == "2") |
  (done$freq == 0.000007 & done$group == "3") |
  (done$freq == 0.00001 & done$group == "4") |
  (done$freq == 0.00002 & done$group == "5") |
  (done$freq == 0.00008 & done$group == "6") |
  (done$freq == 0.0002 & done$group == "7") |
  (done$freq == 0.0004 & done$group == "8") |
  (done$freq == 0.0009 & done$group == "9")
  ),]
dim(done2)
#45 

hcman <- 20000
hcmac <- 2

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))

done2$samples <- as.factor(done2$samples)
axis <- unique(done2$freq)

ggplot(data=done2, aes(x=num, y=uci, colour=(samples))) +
         geom_line(aes(x = num, y = lci, color=(samples), group=freq), size = 0.5) + 
         geom_line(aes(x = num, y = uci, color=(samples), group=freq), size = 0.5) + 
         geom_point(aes(x = num, y = lci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = uci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = penetrance, group=freq), size = 2, color="black") +
         ggtitle("Case frequency of 2/20,000") +
        ylab("Penetrance") +
        theme_minimal() +
          geom_hline(yintercept=1, color ="black", size=0.5, linetype="dashed") +
   theme(legend.position="top", axis.text = element_text(size=14), axis.title = element_text(size=16), legend.title = element_text(size=16), legend.text = element_text(size=14),
         title = element_text(size=16),
          panel.grid.minor.y = element_line( size = 0.4),
                   panel.grid.minor.x = element_blank(),
          panel.grid.major.y = element_line( size = 0.4))  +
          scale_x_continuous(breaks = seq(1, 55, 6), labels=SoilSciGuylabs)+labs(colour="Population")+
          xlab("Variant frequency in the population")

#### 4

hcmac <- 4

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))

#### 8

hcmac <- 8

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))

a <- ggplot(data=done2, aes(x=num, y=uci, colour=(samples))) +
         geom_line(aes(x = num, y = lci, color=(samples), group=freq), size = 0.5) + 
         geom_line(aes(x = num, y = uci, color=(samples), group=freq), size = 0.5) + 
         geom_point(aes(x = num, y = lci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = uci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = penetrance, group=freq), size = 2, color="black") +
         ggtitle("Case frequency of 8/20,000") +
        ylab("Penetrance") +
        theme_minimal() +
          geom_hline(yintercept=1, color ="black", size=0.5, linetype="dashed") +
   theme(legend.position="top", axis.text = element_text(size=14), axis.title = element_text(size=16), legend.title = element_text(size=16), legend.text = element_text(size=14),
         title = element_text(size=16),
          panel.grid.minor.y = element_line( size = 0.4),
                   panel.grid.minor.x = element_blank(),
          panel.grid.major.y = element_line( size = 0.4))  +
          scale_x_continuous(breaks = seq(1, 55, 6), labels=SoilSciGuylabs)+labs(colour="Population")+
          xlab("Variant frequency in the population")

b <- ggplot(data=done2, aes(x=num, y=uci, colour=(samples))) +
         geom_line(aes(x = num, y = lci, color=(samples), group=freq), size = 0.5) + 
         geom_line(aes(x = num, y = uci, color=(samples), group=freq), size = 0.5) + 
         geom_point(aes(x = num, y = lci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = uci, color=(samples), group=freq), size = 3) + 
         geom_point(aes(x = num, y = penetrance, group=freq), size = 2, color="black") +
#         ggtitle("Case frequency of 8/20,000") +
        ylab("Penetrance") +
        theme_minimal() +
          geom_hline(yintercept=1, color ="black", size=0.5, linetype="dashed") +
   theme(legend.position="top", axis.text = element_text(size=14), axis.title = element_text(size=16), legend.title = element_text(size=16), legend.text = element_text(size=14),
         title = element_text(size=16),
          panel.grid.minor.y = element_line( size = 0.4),
                   panel.grid.minor.x = element_blank(),
          panel.grid.major.y = element_line( size = 0.4))  +
          scale_x_continuous(breaks = seq(1, 55, 6), labels=SoilSciGuylabs)+labs(colour="Population")+
          xlab("Variant frequency in the population")+
  ylim(0,1)

ggarrange(a+rremove("x.text")+rremove("x.title"), b, ncol = 1, nrow = 2, widths=c(0.1,0.1), align='v', common.legend=TRUE)

#### 10

hcmac <- 10

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))

#### 20

hcmac <- 20

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))

#### 40

hcmac <- 40

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))

#### 80

hcmac <- 80

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))

#### 160

hcmac <- 160

done2[7:9] <- penetrance(x_D=97, n_D=52660, x_AgD=hcmac, n_AgD=hcman, x_A=done2$gnukb_AC, n_A=done2$gnukb_AN)
SoilSciGuylabs <- c("", unique(done2$freq))
```

### Penetrance Systematic Plots - 10% penetrant variant

```{r}
#the fixed variant is found twice* in 2,500 cases (0.0008) and 4 times in 300k population (0.000013).
#*changes with penetrance increase

#for ~10% penetrance
rm(done)
rm(done2)
rm(done3)
options(scipen=999)

h_ac <- c(2, 4, 8, 16, 64, 160)
h_ac2 <- rep(h_ac, each= 6)
h_an <- c(2500, 5000, 10000, 20000, 80000, 200000)
h_an2 <- rep(h_an, each= 6)
g_ac <- c(2, 4, 8, 16, 40, 1200)
g_an <- c(150000, 300000, 600000, 1200000, 3e+06, 90000000)
done <- data.frame(h_ac2)
done$h_an2 <- h_an2
done$g_an <- g_an
done$g_ac <- g_ac
done$samples <- done$h_an2/2
done$num <- as.numeric(row.names(done))
done$freq <- done$h_ac2/done$h_an2
done$gfreq <- done$g_ac/done$g_an
done$gsamples <- done$g_an/2
dim(done)
#36  

done2 <- done
dim(done2)
#36
done2[10:12] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac2, n_AgD=done2$h_an2, x_A=done2$g_ac, n_A=done2$g_an)
SoilSciGuylabs <- c("", unique(done2$freq))

scaleFUN <- function(x) sprintf("%.f", x)

a <- ggplot(data=done2, aes(x=as.factor(samples), y=penetrance, colour=as.factor(gsamples))) +
  geom_line(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 0.5) + 
  geom_point(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 0.5) + 
    geom_point(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = penetrance, group=as.factor(gsamples)), size = 0.5, color="black") +
  labs(col="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
  scale_x_continuous(trans='log', labels=scaleFUN, breaks = c(1250, 10000,100000)) +
  geom_vline(xintercept=0, color="black") +
       xlab("Number of cases (log scale)") +
        ylab("Penetrance") +
        ggtitle("10% penetrance") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=18), 
              axis.title = element_text(size=18),
              plot.title = element_text(size = 18, face = "bold"),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=18),
          legend.title=element_text(size=18)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3)

#rate of convergence - UCI

attach(done2)
done3 <- done2[order(gsamples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$samples == 1250),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

a <- ggplot(data=done3, aes(x=samples, y=error_diff)) +
  geom_line(aes(color=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Percentage decrease in CI uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=12), 
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
  scale_x_continuous(trans='log', breaks = c(0, 1250,10000,100000))+
  labs(color="Population") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")+
  ggtitle("10% penetrance")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$samples[i+1]-done3$samples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$gsamples[i+1] != done3$gsamples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

b <- ggplot(data=done3, aes(x=samples, y=diff_slope)) +
  geom_line(aes(color=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=10000, color="darkgrey") +
  scale_x_continuous(trans='log', breaks = c(1250, 10000, 100000))+ labs(col="Population")+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

########## gnomAD samples

c <- ggplot(data=done2, aes(x=gsamples, y=penetrance, col=as.factor(samples))) +
   #     geom_point(position = position_dodge(width = 1), size = 2) +
        xlab("Number of population participants (log scale)")+
        ylab("Penetrance and 95% CI") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=16), axis.title = element_text(size=16),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=16),
          axis.text.x = element_text(size=16),
          legend.text=element_text(size=16),
          legend.title=element_text(size=16)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = gsamples, y = lci), size = 0.5, color="black", group=done2$gsamples) + 
  geom_point(aes(x = gsamples, y = lci), size = 2) +
  geom_line(aes(x = gsamples, y = uci), size = 0.5, color="black", group=done2$gsamples) + 
    geom_point(aes(x = gsamples, y = uci), size = 2) +
  geom_line(aes(x = gsamples, y = penetrance), size = 0.5, color="black", group=done2$gsamples) + labs(col="Number of cases")   +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 300000,1500000,45000000))+
  geom_vline(xintercept=0, color="black")

#rate of convergence

attach(done2)
done3 <- done2[order(samples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$gsamples == 75000),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

d <- ggplot(data=done3, aes(x=gsamples, y=error_diff)) +
  geom_line(aes(colour=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Percentage decrease in uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Cases") +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
  labs(color="Cases") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  ylim(0,100) +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$gsamples[i+1]-done3$gsamples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$samples[i+1] != done3$samples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

e <- ggplot(data=done3, aes(x=gsamples, y=diff_slope)) +
  geom_line(aes(color=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=300000, color="darkgrey") +
      scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
 labs(col="Cases") +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

#####
ggarrange(a , b , d, e, ncol = 2, nrow = 2, align = "v")
```

### Penetrance Systematic Plots - 20% penetrant variant - Figure 8

```{r}
#the fixed variant is found twice* in 2,500 cases (0.0008) and 4 times in 300k population (0.000013).
#*changes with penetrance increase

#for ~20% penetrance
rm(done)
rm(done2)
rm(done3)
options(scipen=999)

h_ac <- c(2*2, 4*2, 8*2, 16*2, 64*2, 160*2)
h_ac2 <- rep(h_ac, each= 6)
h_an <- c(2500, 5000, 10000, 20000, 80000, 200000)
h_an2 <- rep(h_an, each= 6)
g_ac <- c(2, 4, 8, 16, 40, 1200)
g_an <- c(150000, 300000, 600000, 1200000, 3e+06, 90000000)
done <- data.frame(h_ac2)
done$h_an2 <- h_an2
done$g_an <- g_an
done$g_ac <- g_ac
done$samples <- done$h_an2/2
done$num <- as.numeric(row.names(done))
done$freq <- done$h_ac2/done$h_an2
done$gfreq <- done$g_ac/done$g_an
done$gsamples <- done$g_an/2
dim(done)
#36  

done2 <- done
dim(done2)
#36
done2[10:12] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac2, n_AgD=done2$h_an2, x_A=done2$g_ac, n_A=done2$g_an)
SoilSciGuylabs <- c("", unique(done2$freq))

g1 <- done2[which(done2$samples == "10000" & done2$gsamples == "300000"),]

scaleFUN <- function(x) sprintf("%.f", x)

b <-  ggplot(data=done2, aes(x=as.factor(samples), y=penetrance, colour=as.factor(gsamples))) +
  geom_line(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 0.5) + 
  geom_point(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 0.5) + 
    geom_point(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = penetrance, group=as.factor(gsamples)), size = 0.5, color="black") +
  labs(col="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
  scale_x_continuous(trans='log', labels=scaleFUN, breaks = c(1250, 10000,100000)) +
  geom_vline(xintercept=0, color="black") +
       xlab("Number of cases (log scale)") +
        ylab("Penetrance") +
  ggtitle("20% penetrance") +
        theme_minimal() +
        theme(legend.position="top", 
              axis.text = element_text(size=18), 
              axis.title = element_text(size=18),
              plot.title = element_text(size = 18, face = "bold"),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=18),
          legend.title=element_text(size=18)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) #+
  # annotate(geom="text", x=10000, y=0.4816566, label="O", color="black")+
  #   annotate(geom="text", x=10000, y=0.1011015, label="O", color="black")+
  #     annotate(geom="text", x=10000, y=0.2206722, label="O", color="black")+
# annotate('segment', x = 10000, y = 0.4816566, xend = 10000, yend = 0.1011015,  size=10, alpha=1/3, color="grey")+
#  scale_alpha(guide="none")+scale_size(guide="none")

ggplot(data=done2, aes(x=as.factor(samples), y=penetrance, colour=as.factor(gsamples))) +
  geom_line(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 0.5) + 
  geom_point(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 0.5) + 
    geom_point(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = penetrance, group=as.factor(gsamples)), size = 0.5, color="black") +
  labs(col="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
  scale_x_continuous(trans='log', labels=scaleFUN, breaks = c(1250, 10000,100000)) +
  geom_vline(xintercept=0, color="black") +
       xlab("Number of cases (log scale)") +
        ylab("Penetrance") +
        theme_minimal() +
        theme(legend.position="top", 
              axis.text = element_text(size=18), 
              axis.title = element_text(size=18),
              plot.title = element_text(size = 18, face = "bold"),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=18),
          legend.title=element_text(size=18)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
   annotate(geom="text", x=10000, y=0.4816566, label="O", color="black")+
     annotate(geom="text", x=10000, y=0.1011015, label="O", color="black")+
       annotate(geom="text", x=10000, y=0.2206722, label="O", color="black")+
 annotate('segment', x = 10000, y = 0.4816566, xend = 10000, yend = 0.1011015,  size=10, alpha=1/3, color="grey")+
  scale_alpha(guide="none")+scale_size(guide="none")

#rate of convergence - UCI

attach(done2)
done3 <- done2[order(gsamples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$samples == 1250),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

a <- ggplot(data=done3, aes(x=samples, y=error_diff)) +
  geom_line(aes(colour=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Percentage decrease in CI uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=14), 
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
  scale_x_continuous(trans='log', breaks = c(0, 1250,10000, 100000))+
  labs(color="Population") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")+
  ggtitle("20% penetrance")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$samples[i+1]-done3$samples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$gsamples[i+1] != done3$gsamples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

b <- ggplot(data=done3, aes(x=samples, y=diff_slope)) +
  geom_line(aes(color=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=10000, color="darkgrey") +
  scale_x_continuous(trans='log', breaks = c(1250, 10000, 100000))+ labs(col="Population")+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

########## gnomAD samples

c <- ggplot(data=done2, aes(x=gsamples, y=penetrance, col=as.factor(samples))) +
   #     geom_point(position = position_dodge(width = 1), size = 2) +
        xlab("Number of population participants (log scale)")+
        ylab("Penetrance and 95% CI") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=12),
              axis.title = element_text(size=12),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=12),
          axis.text.x = element_text(size=12),
          legend.text=element_text(size=12),
          legend.title=element_text(size=12)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = gsamples, y = lci), size = 0.5, color="black", group=done2$gsamples) + 
  geom_point(aes(x = gsamples, y = lci), size = 2) +
  geom_line(aes(x = gsamples, y = uci), size = 0.5, color="black", group=done2$gsamples) + 
    geom_point(aes(x = gsamples, y = uci), size = 2) +
  geom_line(aes(x = gsamples, y = penetrance), size = 0.5, color="black", group=done2$gsamples) + labs(col="Number of cases")   +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 300000,1500000,45000000))+
  geom_vline(xintercept=0, color="black")

#rate of convergence

attach(done2)
done3 <- done2[order(samples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$gsamples == 75000),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

d <- ggplot(data=done3, aes(x=gsamples, y=error_diff)) +
  geom_line(aes(color=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Percentage decrease in uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Cases") +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
  labs(color="Cases") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  ylim(0,100) +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$gsamples[i+1]-done3$gsamples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$samples[i+1] != done3$samples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

e <- ggplot(data=done3, aes(x=gsamples, y=diff_slope)) +
  geom_line(aes(colour=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=300000, color="darkgrey") +
      scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
 labs(col="Cases") +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

#### plot

ggarrange(a , b , d, e, ncol = 2, nrow = 2, align = "v")
```

### Penetrance Systematic Plots - 55%

```{r}
#the fixed variant is found twice* in 2,500 cases (0.0008) and 4 times in 300k population (0.000013).
#*changes with penetrance increase

#for ~10% penetrance
rm(done)
rm(done2)
rm(done3)
options(scipen=999)

h_ac <- c(2*5, 4*5, 8*5, 16*5, 64*5, 160*5)
h_ac2 <- rep(h_ac, each= 6)
h_an <- c(2500, 5000, 10000, 20000, 80000, 200000)
h_an2 <- rep(h_an, each= 6)
g_ac <- c(2, 4, 8, 16, 40, 1200)
g_an <- c(150000, 300000, 600000, 1200000, 3e+06, 90000000)
done <- data.frame(h_ac2)
done$h_an2 <- h_an2
done$g_an <- g_an
done$g_ac <- g_ac
done$samples <- done$h_an2/2
done$num <- as.numeric(row.names(done))
done$freq <- done$h_ac2/done$h_an2
done$gfreq <- done$g_ac/done$g_an
done$gsamples <- done$g_an/2
dim(done)
#36  

done2 <- done
dim(done2)
#36
done2[10:12] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac2, n_AgD=done2$h_an2, x_A=done2$g_ac, n_A=done2$g_an)
SoilSciGuylabs <- c("", unique(done2$freq))

scaleFUN <- function(x) sprintf("%.f", x)
c <-  ggplot(data=done2, aes(x=as.factor(samples), y=penetrance, colour=as.factor(gsamples))) +
  geom_line(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 0.5) + 
  geom_point(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 0.5) + 
    geom_point(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = penetrance, group=as.factor(gsamples)), size = 0.5, color="black") +
  labs(col="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
    scale_x_continuous(trans='log', labels=scaleFUN, breaks = c(1250, 10000,100000)) +
  geom_vline(xintercept=0, color="black") +
       xlab("Number of cases (log scale)") +
        ylab("Penetrance") +
  ggtitle("55% penetrance") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=18), 
              axis.title = element_text(size=18),
              plot.title = element_text(size = 18, face = "bold"),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=18),
          legend.title=element_text(size=18)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3)

#rate of convergence - UCI

attach(done2)
done3 <- done2[order(gsamples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$samples == 1250),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

a <- ggplot(data=done3, aes(x=samples, y=error_diff)) +
  geom_line(aes(color=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Percentage decrease in CI uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
  scale_x_continuous(trans='log', breaks = c(0, 1250,10000,100000))+
  labs(color="Population") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")+
  ggtitle("55% penetrance")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$samples[i+1]-done3$samples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$gsamples[i+1] != done3$gsamples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

b <- ggplot(data=done3, aes(x=samples, y=diff_slope)) +
  geom_line(aes(color=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=10000, color="darkgrey") +
  scale_x_continuous(trans='log', breaks = c(1250, 10000, 100000))+ labs(col="Population")+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

########## gnomAD samples

c <- ggplot(data=done2, aes(x=gsamples, y=penetrance, col=as.factor(samples))) +
   #     geom_point(position = position_dodge(width = 1), size = 2) +
        xlab("Number of population participants (log scale)")+
        ylab("Penetrance and 95% CI") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=16), axis.title = element_text(size=16),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=16),
          axis.text.x = element_text(size=16),
          legend.text=element_text(size=16),
          legend.title=element_text(size=16)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = gsamples, y = lci), size = 0.5, color="black", group=done2$gsamples) + 
  geom_point(aes(x = gsamples, y = lci), size = 2) +
  geom_line(aes(x = gsamples, y = uci), size = 0.5, color="black", group=done2$gsamples) + 
    geom_point(aes(x = gsamples, y = uci), size = 2) +
  geom_line(aes(x = gsamples, y = penetrance), size = 0.5, color="red", group=done2$gsamples) + labs(col="Number of cases")   +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 300000,1500000,45000000))+
  geom_vline(xintercept=0, color="black")

#rate of convergence

attach(done2)
done3 <- done2[order(samples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$gsamples == 75000),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

d <- ggplot(data=done3, aes(x=gsamples, y=error_diff)) +
  geom_line(aes(color=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Percentage decrease in uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Cases") +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
  labs(color="Cases") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  ylim(0,100) +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$gsamples[i+1]-done3$gsamples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$samples[i+1] != done3$samples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

e <- ggplot(data=done3, aes(x=gsamples, y=diff_slope)) +
  geom_line(aes(color=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=12), 
            axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=300000, color="darkgrey") +
      scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
 labs(col="Cases") +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

#####

ggarrange(a , b , d, e, ncol = 2, nrow = 2, align = "v")
```

### Penetrance Systematic Plots - 75%

```{r}
#the fixed variant is found twice* in 2,500 cases (0.0008) and 4 times in 300k population (0.000013).
#*changes with penetrance increase

#for ~10% penetrance
rm(done)
rm(done2)
rm(done3)
options(scipen=999)

h_ac <- c(2*7, 4*7, 8*7, 16*7, 64*7, 160*7)
h_ac2 <- rep(h_ac, each= 6)
h_an <- c(2500, 5000, 10000, 20000, 80000, 200000)
h_an2 <- rep(h_an, each= 6)
g_ac <- c(2, 4, 8, 16, 40, 1200)
g_an <- c(150000, 300000, 600000, 1200000, 3e+06, 90000000)
done <- data.frame(h_ac2)
done$h_an2 <- h_an2
done$g_an <- g_an
done$g_ac <- g_ac
done$samples <- done$h_an2/2
done$num <- as.numeric(row.names(done))
done$freq <- done$h_ac2/done$h_an2
done$gfreq <- done$g_ac/done$g_an
done$gsamples <- done$g_an/2
dim(done)
#36  

done2 <- done
dim(done2)
#36
done2[10:12] <- penetrance(x_D=97, n_D=52660, x_AgD=done2$h_ac2, n_AgD=done2$h_an2, x_A=done2$g_ac, n_A=done2$g_an)
SoilSciGuylabs <- c("", unique(done2$freq))

scaleFUN <- function(x) sprintf("%.f", x)

d <-  ggplot(data=done2, aes(x=as.factor(samples), y=penetrance, colour=as.factor(gsamples))) +
  geom_line(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 0.5) + 
  geom_point(aes(x = samples, y = lci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 0.5) + 
    geom_point(aes(x = samples, y = uci, color=as.factor(gsamples)), size = 3) +
  geom_line(aes(x = samples, y = penetrance, group=as.factor(gsamples)), size = 0.5, color="black") +
  labs(col="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
scale_x_continuous(trans='log', labels=scaleFUN, breaks = c(1250, 10000,100000)) +
  geom_vline(xintercept=0, color="black") +
       xlab("Number of cases (log scale)") +
        ylab("Penetrance and 95% CI") +
  ggtitle("75% penetrance") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=20), 
              axis.title = element_text(size=18),
              plot.title = element_text(size = 18, face = "bold"),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=18),
          legend.title=element_text(size=18)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3)

#rate of convergence - UCI

attach(done2)
done3 <- done2[order(gsamples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$samples == 1250),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

a <- ggplot(data=done3, aes(x=samples, y=error_diff)) +
  geom_line(aes(color=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Percentage decrease in CI uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Population") +
  geom_vline(xintercept=10000, color="darkgrey")+
  scale_x_continuous(trans='log', breaks = c(0, 1250,10000, 100000))+
  labs(color="Population") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")+
  ggtitle("75% penetrance")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$samples[i+1]-done3$samples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$gsamples[i+1] != done3$gsamples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

b <- ggplot(data=done3, aes(x=samples, y=diff_slope)) +
  geom_line(aes(color=factor(gsamples)), size=0.5) +
  geom_point(aes(color=as.factor(gsamples)), size=3) +
       xlab("Number of cases") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=10000, color="darkgrey") +
  scale_x_continuous(trans='log', breaks = c(1250, 10000, 100000))+ labs(col="Population")+
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

########## gnomAD samples

c <- ggplot(data=done2, aes(x=gsamples, y=penetrance, col=as.factor(samples))) +
   #     geom_point(position = position_dodge(width = 1), size = 2) +
        xlab("Number of population participants (log scale)")+
        ylab("Penetrance and 95% CI") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=16), axis.title = element_text(size=16),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=16),
          axis.text.x = element_text(size=16),
          legend.text=element_text(size=16),
          legend.title=element_text(size=16)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = gsamples, y = lci), size = 0.5, color="black", group=done2$gsamples) + 
  geom_point(aes(x = gsamples, y = lci), size = 2) +
  geom_line(aes(x = gsamples, y = uci), size = 0.5, color="black", group=done2$gsamples) + 
    geom_point(aes(x = gsamples, y = uci), size = 2) +
  geom_line(aes(x = gsamples, y = penetrance), size = 0.5, color="red", group=done2$gsamples) + labs(col="Number of cases")   +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 300000,1500000,45000000))+
  geom_vline(xintercept=0, color="black")

#rate of convergence

attach(done2)
done3 <- done2[order(samples),]
detach(done2)

done3$interval <- done3$uci - done3$lci

hmm <- done3[which(done3$gsamples == 75000),]

done3$maxdiff <- rep(hmm$interval, each = 6)

for (i in 1:nrow(done3)) {
  done3$error_diff[i] <- (abs((done3$interval[i]-done3$maxdiff[i])/done3$maxdiff[i])*100)
}

d <- ggplot(data=done3, aes(x=gsamples, y=error_diff)) +
  geom_line(aes(color=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Percentage decrease in uncertainty") +
        theme_minimal() +
        theme(legend.position="top", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  labs(legend="Cases") +
  geom_vline(xintercept=300000, color="darkgrey") +
    scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
  labs(color="Cases") +
  scale_y_continuous(breaks = c(0,20,40,60,80,100,120,140))+
  ylim(0,100) +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=100, color="black")+
    geom_hline(yintercept=0, color="black")

#slope of 2 points
for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- (abs(done3$error_diff[i+1]-done3$error_diff[i])/
     (done3$gsamples[i+1]-done3$gsamples[i])) 
} 

for (i in 1:nrow(done3)) {
done3$diff_slope[i] <- ifelse(done3$samples[i+1] != done3$samples[i], "0", done3$diff_slope[i])
}

done3$diff_slope <- as.numeric(done3$diff_slope)

e <- ggplot(data=done3, aes(x=gsamples, y=diff_slope)) +
  geom_line(aes(color=factor(samples)), size=0.5) +
  geom_point(aes(color=as.factor(samples)), size=3) +
       xlab("Number of population participants") +
        ylab("Rate of convergence") +
        theme_minimal() +
        theme(legend.position="none", axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              panel.grid.minor.y = element_line( size = 0.4),
          panel.grid.major.y = element_line( size = 0.4),
          axis.title.x = element_text(size=14),
          axis.text.x = element_text(size=14),
          legend.text=element_text(size=14),
          legend.title=element_text(size=14)) +
  geom_vline(xintercept=300000, color="darkgrey") +
      scale_x_continuous(trans='log', breaks = c(75000, 1500000,45000000)) +
 labs(col="Cases") +
  geom_vline(xintercept=0, color="black")+
    geom_hline(yintercept=0, color="black")

####
ggarrange(a , b , d, e, ncol = 2, nrow = 2, align = "v")
```

### join four plots for manuscript - suppl figure

```{r}
#install.packages('ggpubr')

library(ggpubr)

ggarrange(a + rremove("xlab"), b + rremove("xlab")+ rremove("ylab") , c, d+ rremove("ylab") , ncol = 2, nrow = 2, align = "v", common.legend = TRUE, legend = "top")
```

### Penetrance Systematic Plots - max, med, min HCM

```{r}
####################################
####################################
#HCM_AF
#min = 0.0001
#median = 0.0003
#max = 0.008
#gnukb_AF
#min = 0.000002
#median = 0.00001
#max = 0.0009

#HCM_AF = 0.008 (max) + gnubk
rm(done)
h_af <- 0.008
g_af <- c(0.000002, 0.00001, 0.0004, 0.0009)
gnoman <- c(100000,200000,300000,400000,500000,600000,700000,800000,900000,1000000)
#70k - 1000K samples
done <- as.integer(unlist(lapply(1:length(g_af), function(x) lapply(1:length(gnoman), function(y) as.matrix(g_af[[x]]) %*%  gnoman[[y]]))))
done <- data.frame(done)
done$done <- gsub("^0", "1", done$done)
done$done <- as.integer(done$done)
done$an <- gnoman
dim(done)
# 40
colnames(done) <- c("g_ac", "g_an")

h_an <- 20000
h_ac <- as.integer(h_af*h_an)
h_ac
#160

done[3:5] <- penetrance(x_D=97, n_D=52660, x_AgD=h_ac, n_AgD=h_an, x_A=done$g_ac, n_A=done$g_an)

done$interval <- done$uci - done$lci
done$group <- rep(c(1,2,3,4), each = 10)
done$samples <- done$g_an/2
done$num <- as.numeric(row.names(done))
done$group <- as.factor(done$group)

done2 <- done[which(done$g_ac >1),]
dim(done2)
#30
done2$gaf <- done2$g_ac/done2$g_an
SoilSciGuylabs <- c("", unique(done2$gaf))

a <- ggplot(data=done2, aes(x=num, y=lci, group=group, color=as.factor(samples))) +
        geom_point(position = position_dodge(width = 1), size=3) +
        xlab("Variant frequency in the population") +
        ggtitle("HCM AF = 0.008 (maximum)") +
        ylab("Penetrance") +
        theme_minimal() +
        theme(legend.position="top", 
              legend.text = element_text(size=14),
              legend.title = element_text(size=14),
              axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              title = element_text(size=14)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = num, y = lci, color=as.factor(samples)), size = 0.5) + 
    geom_line(aes(x = num, y = uci, color=as.factor(samples)), size = 0.5) + 
      geom_line(aes(x = num, y = penetrance, color=as.factor(samples)), size = 0.5) + 
  geom_point(aes(x=num, y=uci, col=as.factor(samples)), size=3) +
    geom_point(aes(x=num, y=penetrance,   group=group), col="black", size=2) +
          scale_x_continuous(breaks = seq(1, 41, by = 10), labels=SoilSciGuylabs) +
  theme(axis.line.y.right = element_line(color = "blue"), 
       axis.ticks.y.right = element_line(color = "blue"),
       axis.title.y.right = element_text(colour = "blue"),
       axis.line.y.left = element_line(color = "black"),
       panel.grid.major.x = element_line(size = 0.5, colour = "grey"), 
       panel.grid.minor.x = element_blank()) +labs(colour="Samples")             

#same but zoom
b <- ggplot(data=done2, aes(x=num, y=lci, group=group, color=as.factor(samples))) +
        geom_point(position = position_dodge(width = 1), size=3) +
        xlab("Variant frequency in the population") +
        ggtitle("HCM AF = 0.008 (maximum), zoom") +
        ylab("Penetrance") +
        theme_minimal() +
        theme(legend.position="top", 
              legend.text = element_text(size=14),
              legend.title = element_text(size=14),
              axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              title = element_text(size=14)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = num, y = lci, color=as.factor(samples)), size = 0.5) + 
    geom_line(aes(x = num, y = uci, color=as.factor(samples)), size = 0.5) + 
      geom_line(aes(x = num, y = penetrance, color=as.factor(samples)), size = 0.5) + 
  geom_point(aes(x=num, y=uci, col=as.factor(samples)), size=3) +
    geom_point(aes(x=num, y=penetrance,   group=group), col="black", size=2) +
          scale_x_continuous(breaks = seq(1, 41, by = 10), labels=SoilSciGuylabs) +
  theme(axis.line.y.right = element_line(color = "blue"), 
       axis.ticks.y.right = element_line(color = "blue"),
       axis.title.y.right = element_text(colour = "blue"),
       axis.line.y.left = element_line(color = "black"),
       panel.grid.major.x = element_line(size = 0.5, colour = "grey"), 
       panel.grid.minor.x = element_blank()) +labs(colour="Samples")   +ylim(0,1)

#HCM_AF = 0.0003 (median) + gnubk
rm(done)
h_af <- 0.0003
g_af <- c(0.000002, 0.00001, 0.0004, 0.0009)
gnoman <- c(100000,200000,300000,400000,500000,600000,700000,800000,900000,1000000)
#70k - 1000K samples
done <- as.integer(unlist(lapply(1:length(g_af), function(x) lapply(1:length(gnoman), function(y) as.matrix(g_af[[x]]) %*%  gnoman[[y]]))))
done <- data.frame(done)
done$done <- gsub("^0", "1", done$done)
done$done <- as.integer(done$done)
done$an <- gnoman
dim(done)
# 40
colnames(done) <- c("g_ac", "g_an")

h_an <- 20000
h_ac <- as.integer(h_af*h_an)

done[3:5] <- penetrance(x_D=97, n_D=52660, x_AgD=h_ac, n_AgD=h_an, x_A=done$g_ac, n_A=done$g_an)

done$interval <- done$uci - done$lci
done$group <- rep(c(1,2,3,4), each = 10)
done$samples <- done$g_an/2
done$num <- as.numeric(row.names(done))
done$group <- as.factor(done$group)

done2 <- done[which(done$g_ac >1),]
dim(done2)
#30

c <- ggplot(data=done2, aes(x=num, y=lci, group=group, color=as.factor(samples))) +
        geom_point(position = position_dodge(width = 1), size=3) +
        xlab("Variant frequency in the population") +
        ggtitle("HCM AF = 0.0003 (median)") +
        ylab("Penetrance") +
        theme_minimal() +
        theme(legend.position="top", 
              legend.text = element_text(size=14),
              legend.title = element_text(size=14),
              axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              title = element_text(size=14)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = num, y = lci, color=as.factor(samples)), size = 0.5) + 
    geom_line(aes(x = num, y = uci, color=as.factor(samples)), size = 0.5) + 
      geom_line(aes(x = num, y = penetrance, color=as.factor(samples)), size = 0.5) + 
  geom_point(aes(x=num, y=uci, col=as.factor(samples)), size=3) +
    geom_point(aes(x=num, y=penetrance,   group=group), col="black", size=2) +
          scale_x_continuous(breaks = seq(1, 41, by = 10), labels=SoilSciGuylabs) +
  theme(axis.line.y.right = element_line(color = "blue"), 
       axis.ticks.y.right = element_line(color = "blue"),
       axis.title.y.right = element_text(colour = "blue"),
       axis.line.y.left = element_line(color = "black"),
       panel.grid.major.x = element_line(size = 0.5, colour = "grey"), 
       panel.grid.minor.x = element_blank()) +labs(colour="Samples")             

#HCM_AF = 0.0001 (min) + gnubk
rm(done)
h_af <- 0.0001
g_af <- c(0.000002, 0.00001, 0.0004, 0.0009)
gnoman <- c(100000,200000,300000,400000,500000,600000,700000,800000,900000,1000000)
#70k - 1000K samples
done <- as.integer(unlist(lapply(1:length(g_af), function(x) lapply(1:length(gnoman), function(y) as.matrix(g_af[[x]]) %*%  gnoman[[y]]))))
done <- data.frame(done)
done$done <- gsub("^0", "1", done$done)
done$done <- as.integer(done$done)
done$an <- gnoman
dim(done)
# 40
colnames(done) <- c("g_ac", "g_an")

h_an <- 20000
h_ac <- as.integer(h_af*h_an)

done[3:5] <- penetrance(x_D=97, n_D=52660, x_AgD=h_ac, n_AgD=h_an, x_A=done$g_ac, n_A=done$g_an)

done$interval <- done$uci - done$lci
done$group <- rep(c(1,2,3,4), each = 10)
done$samples <- done$g_an/2
done$num <- as.numeric(row.names(done))
done$group <- as.factor(done$group)

done2 <- done[which(done$g_ac >1),]

d <- ggplot(data=done2, aes(x=num, y=lci, group=group, color=as.factor(samples))) +
        geom_point(position = position_dodge(width = 1), size=3) +
        xlab("Variant frequency in the population") +
        ggtitle("HCM AF = 0.0001 (minimum)") +
        ylab("Penetrance") +
        theme_minimal() +
        theme(legend.position="top", 
              legend.text = element_text(size=14),
              legend.title = element_text(size=14),
              axis.text = element_text(size=14),
              axis.title = element_text(size=14),
              title = element_text(size=14)) +
        geom_hline(yintercept=1, color ="black", size=0.3)  +
        geom_hline(yintercept=0, color ="black", size=0.3) +
  geom_line(aes(x = num, y = lci, color=as.factor(samples)), size = 0.5) + 
    geom_line(aes(x = num, y = uci, color=as.factor(samples)), size = 0.5) + 
      geom_line(aes(x = num, y = penetrance, color=as.factor(samples)), size = 0.5) + 
  geom_point(aes(x=num, y=uci, col=as.factor(samples)), size=3) +
    geom_point(aes(x=num, y=penetrance,   group=group), col="black", size=2) +
          scale_x_continuous(breaks = seq(1, 41, by = 10), labels=SoilSciGuylabs) +
  theme(axis.line.y.right = element_line(color = "blue"), 
       axis.ticks.y.right = element_line(color = "blue"),
       axis.title.y.right = element_text(colour = "blue"),
       axis.line.y.left = element_line(color = "black"),
       panel.grid.major.x = element_line(size = 0.5, colour = "grey"), 
       panel.grid.minor.x = element_blank()) +labs(colour="Samples")             

ggarrange(a , b , c, d, ncol = 2, nrow = 2, align = "v", common.legend = TRUE)
```

# End 
