Cardiac VariantFX data includes cohort data from seven research and clinical centers [see link]. Individuals selected for sequencing have a confirmed diagnosis of either DCM or HCM, or have been referred for diagnostic sequencing for one of these conditions. Samples within cohorts are considered independent and unrelated. In some cohorts [research cohorts with full data availability: RBH, AHC, NHCS] relatedness has been assessed genetically, in others [diagnostic centres with limited access to primary patient records: GeneDx, LMM , ORGL, NIRGL] analysis is restricted to individuals not known to be related. It was not possible to formally evaluate relatedness between different cohorts.  While some data is taken from hospital cohorts, and other data from diagnostic laboratories, it is not expected that the same patients are observed in both contexts since centres serve largely distinct populations, but this possibility cannot be excluded..

Patients were sequenced using different cardiomyopathy gene panels at different centres. When a cohort is not shown in the variant table, this may be because (1) the gene has not been sequenced in the cohort, or (2) the gene was sequenced in the cohort, but the specific variant of interest was not present. The following table [link: https://github.com/ImperialCardioGenetics/VariantFX-DECIPHER/blob/main/maxANs.csv] contains further information on which genes were sequenced in each cohort.

A value of "NA" indicates that the gene has not been sequenced in the cohort, and no further information is available. A numeric value in the table indicates the number of alleles that were sequenced in that cohort (max AN). Coverage data are not available for every position to confirm whether a site has been adequately sequenced to confidently call the position as homozygous reference, but the data have been considered sufficiently robust to issue a clinical report.
